---
title: "Change Log"
---


proposed 

*  Experiment info
    *  sequencing_prep_plate_info -> library_prep_plate_info 
    *  adding qpcr_parasite_density_info 
*  specimeninfo
    *  change plate_info to storage_plate_info
    *  parasite_density_info -> microscopy_parasite_density_info
*  adding unit field to any values ()

*  adding a table for study design 
    *  adding list of contributors
    *  BioProject_accession
    *  study_name
    *  study_description
    *  study_type 
    *  study_contributors
    *  study_collector_chief_scientist 
    look at BioProject info page that we could align with 


# 2025-03-26
*  modified fields for SpecimenInfo and SequencingInfo to be more aligned to SRA and Mixs standards and for internal consistency
    *  samp_collection_device -> specimen_collection_device 
    *  sample_comments -> specimen_comments
    *  samp_taxon_id -> specimen_taxon_id 
        *  changed specimen_taxon_id from a single value to a list of values in case of mixed species 
    *  added library_layout, library_strategy, library_source, library_selection (needed for SRA submission)
    *  added several Mixs fields that may be of interest to the format as optional fields (travel_out_six_month, drug_usage)
    *  added env_broad_scale, env_local_scale, env_medium as optinal fields (are required by SRA but can be 'not applicable' if missing so aded to PMO as optional)
    *  added seq_platform and seq_instrument_model (both needed for SRA)
*  added PlateInfo class to be used in several locations 
*  added extraction_plate_info sequencing_prep_plate_info to ExperimentInfo as optional fields for storing this info 


# 2025-03-15

*  removed pmo_name
*  major change to the organization of indexes and names
    *  _id will be the integer index into tables 
    *  _name will be a character name identifier for a class 
    *  adding sequencing_info_name, specimen_name, experimental_sample_name, microhaplotype_name to their respective classes 
*  adding pmo_header
    *  info on the PMO file .e.g version etc 
    *  several new classes, including PmoGenerationMethod
*  adding read_counts_by_stage
    *  read counts by each stage 
    *  this led to the creation of several classes to hold this info
*  adding bioinformatics_run_info
    *  info on the pipeline run 
    *  new class of BioinformaticsRunInfo to run info about a run
*  modified BioinformaticsMethoInfo
    *  purpose -> program_description 
*  renaming 
    *  representative_microhaplotype_sequences -> microhaplotype_info
    *  experiment_infos ->  experiment_info
    *  specimen_infos -> specimen_info
    *  sequenceing_infos -> sequenceing_info
    *  taramp_bioinformatics_infos -> bioinformatics_methods_info
    *  removed any mention of taramp
*  adding target_info
    *  a list of the targets in this file to be referred to by panel_info (see below) instead of listed underneath each panel, this allows for the listing of targets that may be shared across panels/reactions 
    *  added class for MarkerOfInterest to list different markers for each target covers e.g. drug_resistance 
*  updating panel_info
    *  modified to be a list of reactions which will have the collection of targets 
    *  added ReactionInfo, listing the reaction targets 
*  modified panel_info/target_info to remove targeted_genome, moved the genome info to the genomic location class instead
*  adding a list of TargetedGenomes 
    *  this list will then be indexed into to list which genomes are being targeted
*  made a large number of the sequencing_info fields optional
*  to list multiple parasite densities for a specimen added a ParasiteDensity class
    *  has method and density
    *  optional list of densities under SpecimenInfo
*  added project name for specimens 
*  a reorganization of microhaplotypes_detected and microhaplotypes_info
  *  microhaplotypes_info
      *  is now a list rather than a dictionary of targets
      *  each list will have a target_id (index into the target infos) and then another list of microhaplotypes 
  *  microhaplotypes_detected
      *  made everything into lists rather than dictionary and changed identifiers to indexes 

