---
title: Format Development 
---

<script src="https://code.jquery.com/jquery-2.2.2.min.js"></script>

<script>
$(document).ready(function() {
	document.querySelectorAll('.downloadLink').forEach(function(e) { e.setAttribute('download', e.text); });
	document.querySelectorAll('.downloadLink').forEach(function(e) { e.innerHTML = '<i class="fa fa-download"></i>  ' + e.text; });
});
</script>

```{r setup, echo=FALSE}
source("../common.R")

#' Check sharing and unqiue values between two vectors
#'
#' @param vectorA the first vector
#' @param vectorB the second vector
#'
#' @returns a list with 4 vectors, "only_in_vectorA" unique to vectorA, "only_in_vectorB" unique to vectorB, "shared_samples" shared between both vectorA and vectorB, "all" all values by combinng vectorA and vectorB 
set_decompose <- function(vectorA, vectorB){
  ret = list()
  # Find unique and shared samples
  ret[["only_in_vectorA"]] <- setdiff(vectorA, vectorB)  # Samples only in vectorA
  ret[["only_in_vectorB"]] <- setdiff(vectorB, vectorA)  # Samples only in vectorB
  ret[["shared_samples"]] <- intersect(vectorA, vectorB) # Samples shared between vectorA and vectorB
  ret[["all"]] <- union(vectorA, vectorB) # All samples between vectorA and vectorB
  return(ret)
}
```

# Sample and sequencing meta data fields 

The meta fields for the sample and sequencing are based on NCBI naming schemes especially from SRA submission portal for human-associated pathogen. These are similar to the [MiXS standards](https://genomicsstandardsconsortium.github.io/mixs/) set by the [Genomics Standards Consortium (GSC)](https://www.gensc.org/)

## Sample 

SRA for sample meta: <https://submit.ncbi.nlm.nih.gov/biosample/template/?package-0=MIGS.eu.human-associated.6.0&action=definition>  
GSC for MIGS eukaryote: <https://genomicsstandardsconsortium.github.io/mixs/0010002/>   

Example file for SRA submission   
```{r}
#| results: asis
#| echo: false

cat(createDownloadLink("MIGS.eu.human-associated.6.0.xlsx"))
```

## Sequencing  

SRA sequencing meta: <https://www.ncbi.nlm.nih.gov/sra/docs/submitportal/#6-sra-metadata>  


SRA 

SRA Biosample attributes descriptions <https://www.ncbi.nlm.nih.gov/biosample/docs/attributes/>

MIMARKS.specimen.human-associated.6.0

<https://submit.ncbi.nlm.nih.gov/biosample/template/?organism-organism_name=&organism-taxonomy_id=&package-0=MIMARKS.specimen&package-1=MIMARKS.specimen.human-associated.6.0&action=definition>  

MIGS.eu.human-associated.6.0

<https://submit.ncbi.nlm.nih.gov/biosample/template/?organism-organism_name=&organism-taxonomy_id=&package-0=MIGS.eu&package-1=MIGS.eu.human-associated.6.0&action=definition>

Required fields 

*  sample_name
*  organism
*  collection_date
*  env_broad_scale
*  env_local_scale
*  env_medium
*  geo_loc_name
*  lat_lon
*  host
*  isol_growth_condt


Additonal fields not in GSC but are found in downloads from SRA 

*  geo_loc_name_country
*  geo_loc_name_country_continent

? adding geo_loc_name_country_region

GSC 

MIGS.eu
<https://genomicsstandardsconsortium.github.io/mixs/0010002/>

MIGS.human-associated
<https://genomicsstandardsconsortium.github.io/mixs/0016003/>

MIGS.eu.human-associated
<https://genomicsstandardsconsortium.github.io/mixs/0010002_0016003/> 

MIMARKS.specimen
<https://genomicsstandardsconsortium.github.io/mixs/0010009/>

MIMARKS.specimen.human-associated
<https://genomicsstandardsconsortium.github.io/mixs/0010009_0016003/> 


Determining difference between MIGS.eu.human-associated and MIMARKS.specimen.human-associated
```{r}
MIGS_eu_human_associated = readxl::read_excel("MIGS.eu.human-associated.6.0.xlsx", skip = 12)
MIMARKS_specimen_human_associated = readxl::read_excel("MIMARKS.specimen.human-associated.6.0.xlsx", skip= 12)

colname_compared = set_decompose(colnames(MIGS_eu_human_associated), colnames(MIMARKS_specimen_human_associated))

list(only_in_MIGS_eu_human_associated = colname_compared$only_in_vectorA, 
     only_in_MIMARKS_specimen_human_associated = colname_compared$only_in_vectorB)

```


Example file for SRA submission 
```{r}
#| results: asis
#| echo: false

cat(createDownloadLink("SRA_metadata.xlsx"))
```

```{r}
PRJNA1180199_SRR31271348_SraRunTable = readr::read_csv("examples_sra_download_info/PRJNA1180199_SRR31271348_SraRunTable.csv")
PRJNA1180199_SRR31271447_SraRunTable = readr::read_csv("examples_sra_download_info/PRJNA1180199_SRR31271447_SraRunTable.csv")
set_decompose(colnames(PRJNA1180199_SRR31271348_SraRunTable), colnames(PRJNA1180199_SRR31271447_SraRunTable))


PRJNA1180199_allRunInfo = readr::read_tsv("examples_sra_download_info/PRJNA1180199_allRunInfo.tsv")
```

# Python implementation 

Linkml generates a datamodel file with all classes: <https://github.com/PlasmoGenEpi/portable-micorhaplotype-object/blob/main/src/plasmo_tar_amp_schema/datamodel/plasmo_tar_amp_schema.py>


# Tools  

Started development of a tool called pmotools. Python implementation can be found here: <https://github.com/PlasmoGenEpi/pmotools-python/tree/develop>




# From GEM meeting

- [ ] adding header with info on each field 
- [x] adding version to header   
- [ ] random access ability  



{{< fa thumbs-up >}}
