<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>PMO fields overview – Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-704a3d686fdb3e6b26ef47c77620618f.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-83b1d0b3b459d34b5b0f98adfe5eb835.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>html{ scroll-behavior: smooth; }</style>
<script src="../site_libs/quarto-diagram/mermaid.min.js"></script><script src="../site_libs/quarto-diagram/mermaid-init.js"></script><link href="../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
<link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">
<li>
    <a class="dropdown-item" href="../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">
<li>
    <a class="dropdown-item" href="../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">
<li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-runner.py</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/pmotools-runner.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-runner.py</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Getting sub info from PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_panel_info_from_pmo.html">
 <span class="dropdown-text">Getting panel info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/build_pmo.html">
 <span class="dropdown-text">Creating a PMO File</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">
<li>
    <a class="dropdown-item" href="../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Contents</h2>
   
  <ul>
<li><a href="#goal" id="toc-goal" class="nav-link active" data-scroll-target="#goal">Goal</a></li>
  <li>
<a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a>
  <ul>
<li>
<a href="#portablemicrohaplotypeobject" id="toc-portablemicrohaplotypeobject" class="nav-link" data-scroll-target="#portablemicrohaplotypeobject">PortableMicrohaplotypeObject</a>
  <ul>
<li><a href="#required" id="toc-required" class="nav-link" data-scroll-target="#required">Required</a></li>
  <li><a href="#optional" id="toc-optional" class="nav-link" data-scroll-target="#optional">Optional</a></li>
  <li><a href="#example" id="toc-example" class="nav-link" data-scroll-target="#example">Example</a></li>
  </ul>
</li>
  <li><a href="#biomethod" id="toc-biomethod" class="nav-link" data-scroll-target="#biomethod">BioMethod</a></li>
  <li><a href="#bioinformaticsmethodinfo" id="toc-bioinformaticsmethodinfo" class="nav-link" data-scroll-target="#bioinformaticsmethodinfo">BioinformaticsMethodInfo</a></li>
  <li><a href="#bioinformaticsruninfo" id="toc-bioinformaticsruninfo" class="nav-link" data-scroll-target="#bioinformaticsruninfo">BioinformaticsRunInfo</a></li>
  <li><a href="#experimentinfo" id="toc-experimentinfo" class="nav-link" data-scroll-target="#experimentinfo">ExperimentInfo</a></li>
  <li><a href="#genomeinfo" id="toc-genomeinfo" class="nav-link" data-scroll-target="#genomeinfo">GenomeInfo</a></li>
  <li><a href="#genomiclocation" id="toc-genomiclocation" class="nav-link" data-scroll-target="#genomiclocation">GenomicLocation</a></li>
  <li><a href="#markerofinterest" id="toc-markerofinterest" class="nav-link" data-scroll-target="#markerofinterest">MarkerOfInterest</a></li>
  <li><a href="#maskinginfo" id="toc-maskinginfo" class="nav-link" data-scroll-target="#maskinginfo">MaskingInfo</a></li>
  <li><a href="#microhaplotypefortarget" id="toc-microhaplotypefortarget" class="nav-link" data-scroll-target="#microhaplotypefortarget">MicrohaplotypeForTarget</a></li>
  <li><a href="#microhaplotypesdetected" id="toc-microhaplotypesdetected" class="nav-link" data-scroll-target="#microhaplotypesdetected">MicrohaplotypesDetected</a></li>
  <li><a href="#microhaplotypesforsample" id="toc-microhaplotypesforsample" class="nav-link" data-scroll-target="#microhaplotypesforsample">MicrohaplotypesForSample</a></li>
  <li><a href="#microhaplotypesfortarget" id="toc-microhaplotypesfortarget" class="nav-link" data-scroll-target="#microhaplotypesfortarget">MicrohaplotypesForTarget</a></li>
  <li><a href="#panelinfo" id="toc-panelinfo" class="nav-link" data-scroll-target="#panelinfo">PanelInfo</a></li>
  <li><a href="#parasitedensity" id="toc-parasitedensity" class="nav-link" data-scroll-target="#parasitedensity">ParasiteDensity</a></li>
  <li><a href="#pmogenerationmethod" id="toc-pmogenerationmethod" class="nav-link" data-scroll-target="#pmogenerationmethod">PmoGenerationMethod</a></li>
  <li><a href="#pmoheader" id="toc-pmoheader" class="nav-link" data-scroll-target="#pmoheader">PmoHeader</a></li>
  <li><a href="#primerinfo" id="toc-primerinfo" class="nav-link" data-scroll-target="#primerinfo">PrimerInfo</a></li>
  <li><a href="#reactioninfo" id="toc-reactioninfo" class="nav-link" data-scroll-target="#reactioninfo">ReactionInfo</a></li>
  <li><a href="#readcountsbystage" id="toc-readcountsbystage" class="nav-link" data-scroll-target="#readcountsbystage">ReadCountsByStage</a></li>
  <li><a href="#readcountsbystageforexperimentalsample" id="toc-readcountsbystageforexperimentalsample" class="nav-link" data-scroll-target="#readcountsbystageforexperimentalsample">ReadCountsByStageForExperimentalSample</a></li>
  <li><a href="#readcountsbystagefortarget" id="toc-readcountsbystagefortarget" class="nav-link" data-scroll-target="#readcountsbystagefortarget">ReadCountsByStageForTarget</a></li>
  <li><a href="#representativemicrohaplotype" id="toc-representativemicrohaplotype" class="nav-link" data-scroll-target="#representativemicrohaplotype">RepresentativeMicrohaplotype</a></li>
  <li><a href="#representativemicrohaplotypes" id="toc-representativemicrohaplotypes" class="nav-link" data-scroll-target="#representativemicrohaplotypes">RepresentativeMicrohaplotypes</a></li>
  <li><a href="#representativemicrohaplotypesfortarget" id="toc-representativemicrohaplotypesfortarget" class="nav-link" data-scroll-target="#representativemicrohaplotypesfortarget">RepresentativeMicrohaplotypesForTarget</a></li>
  <li><a href="#sequencinginfo" id="toc-sequencinginfo" class="nav-link" data-scroll-target="#sequencinginfo">SequencingInfo</a></li>
  <li><a href="#specimeninfo" id="toc-specimeninfo" class="nav-link" data-scroll-target="#specimeninfo">SpecimenInfo</a></li>
  <li><a href="#stagereadcounts" id="toc-stagereadcounts" class="nav-link" data-scroll-target="#stagereadcounts">StageReadCounts</a></li>
  <li><a href="#targetinfo" id="toc-targetinfo" class="nav-link" data-scroll-target="#targetinfo">TargetInfo</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">PMO fields overview</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-body">

    
  
    
  </div>
  


</header><section id="goal" class="level1"><h1>Goal</h1>
<p>Creating fields with efforts to be consistent with <a href="https://genomicsstandardsconsortium.github.io/mixs/">MIxS standards</a></p>
<p>Important aspects to keep in mind are creating an efficient, low-weight, and minimum amount of information about a run without losing any important data. We can build tools around this table to generate certain fields that are important but not necessary always to keep constantly stored in this base class (e.g.&nbsp;SNP/INDEL calls). Also since we are proposing to keep this data in a singular file in <a href="https://en.wikipedia.org/wiki/JSON">JSON format</a> we are not limited to keeping data in a tabular format for organization; output generated from this file can certainly be a table but we don’t have to store things as a table, e.g.&nbsp;certain fields might be a single ID while other fields might be vectors/lists of data. What’s most important is that we agree on what fields are important and what they should store (and what values should be allowable or formatting, e.g.&nbsp;double/floats vs strings vs characters vs POSIX date vs URL etc)</p>
<p>Format is defined by utilizing <a href="https://linkml.io/linkml/">LinkML</a> to generate a general data scheme which creates convenient outputs like <a href="https://json-schema.org/">JSON Schema</a> for validation tools.</p>
<p>Other notable users of LinkML/MIxS <a href="https://github.com/microbiomedata/nmdc-schema">National Microbiome Data Collaborative Schema</a></p>
</section><section id="overview" class="level1 page-columns page-full"><h1>Overview</h1>
<p>Below is an overview of the entire format currently in alpha development and is still under heavy development and optimization so structure and names are subject to undergo significant changes as development continues. Please send questions to <a href="mailto:info@plasmogenepi.org">info@plasmogenepi.org</a></p>
<p>A more informative overview can also be viewed here <a href="https://drive.google.com/file/d/193B1BBLagay_CYhJm4Nee7Fl4T4BFdL_/view?usp=sharing" class="uri">https://drive.google.com/file/d/193B1BBLagay_CYhJm4Nee7Fl4T4BFdL_/view?usp=sharing</a></p>
<p><a href="https://github.com/PlasmoGenEpi/portable-microhaplotype-object" class="uri">https://github.com/PlasmoGenEpi/portable-microhaplotype-object</a></p>
<div class="column-screen">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">erDiagram
PortableMicrohaplotypeObject {

}
ReadCountsByStage {
    integer bioinformatics_run_id  
}
ReadCountsByStageForExperimentalSample {
    integer experiment_sample_id  
    integer total_raw_count  
}
ReadCountsByStageForTarget {
    integer target_id  
}
StageReadCounts {
    integer read_count  
    string stage  
}
PmoHeader {
    string pmo_version  
    string creation_date  
}
PmoGenerationMethod {
    string program_version  
    string program_name  
}
MicrohaplotypesDetected {
    integer bioinformatics_run_id  
}
MicrohaplotypesForSample {
    integer experiment_sample_id  
}
MicrohaplotypesForTarget {
    integer mhaps_target_id  
}
MicrohaplotypeForTarget {
    integer mhap_id  
    integer reads  
    integer umis  
}
BioinformaticsRunInfo {
    integer bioinformatics_methods_id  
    string run_date  
    string bioinformatics_run_name  
}
BioinformaticsMethodInfo {
    string bioinformatics_method_name  
}
BioMethod {
    string program_version  
    string program  
    string program_description  
    stringList additional_argument  
}
RepresentativeMicrohaplotypes {

}
RepresentativeMicrohaplotypesForTarget {
    integer target_id  
}
RepresentativeMicrohaplotype {
    string seq  
    string microhaplotype_name  
    string quality  
    string pseudocigar  
    stringList alt_annotations  
}
MaskingInfo {
    integer seq_start  
    integer seq_segment_size  
    integer replacement_size  
}
GenomeInfo {
    string name  
    string genome_version  
    integer taxon_id  
    string url  
    stringList chromosomes  
    string gff_url  
}
TargetInfo {
    string target_name  
    string gene_name  
    stringList target_attributes  
}
PrimerInfo {
    string seq  
}
GenomicLocation {
    integer genome_id  
    string chrom  
    integer start  
    integer end  
    string strand  
    string ref_seq  
}
PanelInfo {
    string panel_name  
}
ReactionInfo {
    integerList panel_targets  
    string reaction_name  
}
SequencingInfo {
    string sequencing_info_name  
    string seq_instrument  
    string seq_date  
    string nucl_acid_ext  
    string nucl_acid_amp  
    string nucl_acid_ext_date  
    string nucl_acid_amp_date  
    string pcr_cond  
    string lib_screen  
    string lib_layout  
    string lib_kit  
    string seq_center  
}
SpecimenInfo {
    string plate_name  
    string plate_row  
    integer plate_col  
    string specimen_name  
    integer samp_taxon_id  
    integer individual_id  
    integer host_taxon_id  
    stringList alternate_identifiers  
    string specimen_sex  
    string collection_date  
    string date_of_birth  
    string collection_country  
    string geo_admin1  
    string geo_admin2  
    string geo_admin3  
    string lat_lon  
    string collector  
    string samp_store_loc  
    string samp_collect_device  
    string project_name  
    string sample_comments  
}
ParasiteDensity {
    string method  
    double density  
}
ExperimentInfo {
    integer sequencing_info_id  
    string plate_name  
    string plate_row  
    integer plate_col  
    integer specimen_id  
    integer panel_id  
    string accession  
    string experiment_sample_name  
}

PortableMicrohaplotypeObject ||--}| ExperimentInfo : "experiment_info"
PortableMicrohaplotypeObject ||--}| SpecimenInfo : "specimen_info"
PortableMicrohaplotypeObject ||--}| SequencingInfo : "sequencing_info"
PortableMicrohaplotypeObject ||--}| PanelInfo : "panel_info"
PortableMicrohaplotypeObject ||--}| TargetInfo : "target_info"
PortableMicrohaplotypeObject ||--}| GenomeInfo : "targeted_genomes"
PortableMicrohaplotypeObject ||--|| RepresentativeMicrohaplotypes : "microhaplotypes_info"
PortableMicrohaplotypeObject ||--}| BioinformaticsMethodInfo : "bioinformatics_methods_info"
PortableMicrohaplotypeObject ||--}| BioinformaticsRunInfo : "bioinformatics_run_info"
PortableMicrohaplotypeObject ||--}| MicrohaplotypesDetected : "microhaplotypes_detected"
PortableMicrohaplotypeObject ||--|| PmoHeader : "pmo_header"
PortableMicrohaplotypeObject ||--}o ReadCountsByStage : "read_counts_by_stage"
ReadCountsByStage ||--}| ReadCountsByStageForExperimentalSample : "read_counts_by_experimental_sample_by_stage"
ReadCountsByStageForExperimentalSample ||--}o ReadCountsByStageForTarget : "read_counts_for_targets"
ReadCountsByStageForTarget ||--}| StageReadCounts : "stages"
PmoHeader ||--|o PmoGenerationMethod : "generation_method"
MicrohaplotypesDetected ||--}| MicrohaplotypesForSample : "experiment_samples"
MicrohaplotypesForSample ||--}| MicrohaplotypesForTarget : "target_results"
MicrohaplotypesForTarget ||--}| MicrohaplotypeForTarget : "haps"
BioinformaticsMethodInfo ||--|| BioMethod : "demultiplexing_method"
BioinformaticsMethodInfo ||--|| BioMethod : "denoising_method"
BioinformaticsMethodInfo ||--}o BioMethod : "additional_methods"
RepresentativeMicrohaplotypes ||--}| RepresentativeMicrohaplotypesForTarget : "targets"
RepresentativeMicrohaplotypesForTarget ||--}| RepresentativeMicrohaplotype : "microhaplotypes"
RepresentativeMicrohaplotype ||--}o MaskingInfo : "masking"
TargetInfo ||--|o GenomicLocation : "insert_location"
TargetInfo ||--}| PrimerInfo : "forward_primers"
TargetInfo ||--}| PrimerInfo : "reverse_primers"
PrimerInfo ||--|o GenomicLocation : "location"
PanelInfo ||--}| ReactionInfo : "reactions"
SpecimenInfo ||--}o ParasiteDensity : "parasite_density_info"

</pre>
</div>
<p></p>
</div>
</div>
</div>
</div>
<section id="portablemicrohaplotypeobject" class="level2"><h2 class="anchored" data-anchor-id="portablemicrohaplotypeobject">PortableMicrohaplotypeObject</h2>
<p><a href="https://plasmogenepi.github.io/portable-microhaplotype-object/PortableMicrohaplotypeObject/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/PortableMicrohaplotypeObject/</a></p>
<section id="required" class="level3"><h3 class="anchored" data-anchor-id="required">Required</h3>
<ul>
<li>experiment_info (<strong><em>type=<a href="#experimentinfo">ExperimentInfo</a></em></strong>)
<ul>
<li>a list of experiments of all the seq/amp of the specimens within this project<br>
</li>
</ul>
</li>
<li>specimen_info (<strong><em>type=<a href="#specimeninfo">SpecimenInfo</a></em></strong>)
<ul>
<li>a list of specimens of all the samples within this project<br>
</li>
</ul>
</li>
<li>sequencing_info (<strong><em>type=<a href="#sequencinginfo">SequencingInfo</a></em></strong>)
<ul>
<li>a list of sequencing info for this project<br>
</li>
</ul>
</li>
<li>panel_info (<strong><em>type=<a href="#panelinfo">PanelInfo</a></em></strong>)
<ul>
<li>a list of info on the panels<br>
</li>
</ul>
</li>
<li>target_info (<strong><em>type=<a href="#targetinfo">TargetInfo</a></em></strong>)
<ul>
<li>a list of info on the targets<br>
</li>
</ul>
</li>
<li>targeted_genomes (<strong><em>type=<a href="#genomeinfo">GenomeInfo</a></em></strong>)
<ul>
<li>a list of genomes that the targets in TargetInfo refer to<br>
</li>
</ul>
</li>
<li>microhaplotypes_info (<strong><em>type=<a href="#representativemicrohaplotypes">RepresentativeMicrohaplotypes</a></em></strong>)
<ul>
<li>a list of the information on the representative microhaplotypes<br>
</li>
</ul>
</li>
<li>bioinformatics_methods_info (<strong><em>type=<a href="#bioinformaticsmethodinfo">BioinformaticsMethodInfo</a></em></strong>)
<ul>
<li>the bioinformatics pipeline/methods used to generated the amplicon analysis for this project<br>
</li>
</ul>
</li>
<li>bioinformatics_run_info (<strong><em>type=<a href="#bioinformaticsruninfo">BioinformaticsRunInfo</a></em></strong>)
<ul>
<li>the runtime info for the bioinformatics pipeline used to generated the amplicon analysis for this project<br>
</li>
</ul>
</li>
<li>microhaplotypes_detected (<strong><em>type=<a href="#microhaplotypesdetected">MicrohaplotypesDetected</a></em></strong>)
<ul>
<li>the microhaplotypes detected in this projects<br>
</li>
</ul>
</li>
<li>pmo_header (<strong><em>type=<a href="#pmoheader">PmoHeader</a></em></strong>)
<ul>
<li>the PMO information for this file including version etc</li>
</ul>
</li>
</ul></section><section id="optional" class="level3"><h3 class="anchored" data-anchor-id="optional">Optional</h3>
<ul>
<li>read_counts_by_stage (<strong><em>type=<a href="#readcountsbystage">ReadCountsByStage</a></em></strong>)
<ul>
<li>the read counts for different stages of the pipeline</li>
</ul>
</li>
</ul></section><section id="example" class="level3"><h3 class="anchored" data-anchor-id="example">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="biomethod" class="level2"><h2 class="anchored" data-anchor-id="biomethod">BioMethod</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/BioMethod/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/BioMethod/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioMethod" aria-expanded="false" aria-controls="collapseExample"> Show BioMethod fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseBioMethod" class="collapse">
<div class="card card-body">
<section id="required-1" class="level3"><h3 class="anchored" data-anchor-id="required-1">Required</h3>
<ul>
<li>program_version (<strong><em>type=string</em></strong>)
<ul>
<li>the version of generation method, should be in the format of v[MAJOR].[MINOR].[PATCH]<br>
</li>
</ul>
</li>
<li>program (<strong><em>type=string</em></strong>)
<ul>
<li>name of the program used for this portion of the pipeline</li>
</ul>
</li>
</ul></section><section id="optional-1" class="level3"><h3 class="anchored" data-anchor-id="optional-1">Optional</h3>
<ul>
<li>additional_argument (<strong><em>type=array</em></strong>)
<ul>
<li>any additional arguments that differ from the default<br>
</li>
</ul>
</li>
<li>program_description (<strong><em>type=string</em></strong>)
<ul>
<li>a short description of what this method does</li>
</ul>
</li>
</ul></section><section id="example-1" class="level3"><h3 class="anchored" data-anchor-id="example-1">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="bioinformaticsmethodinfo" class="level2"><h2 class="anchored" data-anchor-id="bioinformaticsmethodinfo">BioinformaticsMethodInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsMethodInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsMethodInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioinformaticsMethodInfo" aria-expanded="false" aria-controls="collapseExample"> Show BioinformaticsMethodInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseBioinformaticsMethodInfo" class="collapse">
<div class="card card-body">
<section id="required-2" class="level3"><h3 class="anchored" data-anchor-id="required-2">Required</h3>
<ul>
<li>demultiplexing_method (<strong><em>type=<a href="#biomethod">BioMethod</a></em></strong>)
<ul>
<li>the demultiplexing method used to separate raw reads from barcodes and primer targets<br>
</li>
</ul>
</li>
<li>denoising_method (<strong><em>type=<a href="#biomethod">BioMethod</a></em></strong>)
<ul>
<li>the method used to de-noise and/or cluster the raw reads</li>
</ul>
</li>
</ul></section><section id="optional-2" class="level3"><h3 class="anchored" data-anchor-id="optional-2">Optional</h3>
<ul>
<li>additional_methods (<strong><em>type=<a href="#biomethod">BioMethod</a></em></strong>)
<ul>
<li>any additional methods used to analyze the data<br>
</li>
</ul>
</li>
<li>bioinformatics_method_name (<strong><em>type=string</em></strong>)
<ul>
<li>name of the collection of methods is called, e.g.&nbsp;pipeline</li>
</ul>
</li>
</ul></section><section id="example-2" class="level3"><h3 class="anchored" data-anchor-id="example-2">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="bioinformaticsruninfo" class="level2"><h2 class="anchored" data-anchor-id="bioinformaticsruninfo">BioinformaticsRunInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsRunInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsRunInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioinformaticsRunInfo" aria-expanded="false" aria-controls="collapseExample"> Show BioinformaticsRunInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseBioinformaticsRunInfo" class="collapse">
<div class="card card-body">
<section id="required-3" class="level3"><h3 class="anchored" data-anchor-id="required-3">Required</h3>
<ul>
<li>bioinformatics_methods_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the bioinformatics_methods_info list<br>
</li>
</ul>
</li>
<li>run_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date when the run was done, should be YYYY-MM-DD</li>
</ul>
</li>
</ul></section><section id="optional-3" class="level3"><h3 class="anchored" data-anchor-id="optional-3">Optional</h3>
<ul>
<li>bioinformatics_run_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name to for this run</li>
</ul>
</li>
</ul></section><section id="example-3" class="level3"><h3 class="anchored" data-anchor-id="example-3">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="experimentinfo" class="level2"><h2 class="anchored" data-anchor-id="experimentinfo">ExperimentInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ExperimentInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ExperimentInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseExperimentInfo" aria-expanded="false" aria-controls="collapseExample"> Show ExperimentInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseExperimentInfo" class="collapse">
<div class="card card-body">
<section id="required-4" class="level3"><h3 class="anchored" data-anchor-id="required-4">Required</h3>
<ul>
<li>sequencing_info_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the sequencing_info list<br>
</li>
</ul>
</li>
<li>specimen_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the specimen_info list<br>
</li>
</ul>
</li>
<li>panel_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the panel_info list<br>
</li>
</ul>
</li>
<li>experiment_sample_name (<strong><em>type=string</em></strong>)
<ul>
<li>a unique identifier for this sequence/amplification run on a specimen_name</li>
</ul>
</li>
</ul></section><section id="optional-4" class="level3"><h3 class="anchored" data-anchor-id="optional-4">Optional</h3>
<ul>
<li>accession (<strong><em>type=string</em></strong>)
<ul>
<li>ERA/SRA accession number for the sample if it was submitted<br>
</li>
</ul>
</li>
<li>plate_col (<strong><em>type=integer</em></strong>)
<ul>
<li>the column the specimen was in<br>
</li>
</ul>
</li>
<li>plate_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name of plate the specimen was in<br>
</li>
</ul>
</li>
<li>plate_row (<strong><em>type=string</em></strong>)
<ul>
<li>the row the specimen was in</li>
</ul>
</li>
</ul></section><section id="example-4" class="level3"><h3 class="anchored" data-anchor-id="example-4">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="genomeinfo" class="level2"><h2 class="anchored" data-anchor-id="genomeinfo">GenomeInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/GenomeInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/GenomeInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseGenomeInfo" aria-expanded="false" aria-controls="collapseExample"> Show GenomeInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseGenomeInfo" class="collapse">
<div class="card card-body">
<section id="required-5" class="level3"><h3 class="anchored" data-anchor-id="required-5">Required</h3>
<ul>
<li>name (<strong><em>type=string</em></strong>)
<ul>
<li>name of the genome<br>
</li>
</ul>
</li>
<li>genome_version (<strong><em>type=string</em></strong>)
<ul>
<li>the genome version<br>
</li>
</ul>
</li>
<li>taxon_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the NCBI taxonomy number<br>
</li>
</ul>
</li>
<li>url (<strong><em>type=string</em></strong>)
<ul>
<li>a link to the where this genome file could be downloaded</li>
</ul>
</li>
</ul></section><section id="optional-5" class="level3"><h3 class="anchored" data-anchor-id="optional-5">Optional</h3>
<ul>
<li>chromosomes (<strong><em>type=array</em></strong>)
<ul>
<li>a list of chromosomes found within this genome<br>
</li>
</ul>
</li>
<li>gff_url (<strong><em>type=string</em></strong>)
<ul>
<li>a link to the where this genome’s annotation file could be downloaded</li>
</ul>
</li>
</ul></section><section id="example-5" class="level3"><h3 class="anchored" data-anchor-id="example-5">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="genomiclocation" class="level2"><h2 class="anchored" data-anchor-id="genomiclocation">GenomicLocation</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/GenomicLocation/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/GenomicLocation/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseGenomicLocation" aria-expanded="false" aria-controls="collapseExample"> Show GenomicLocation fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseGenomicLocation" class="collapse">
<div class="card card-body">
<section id="required-6" class="level3"><h3 class="anchored" data-anchor-id="required-6">Required</h3>
<ul>
<li>genome_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index to the genome in the targeted_genomes list that this location refers to<br>
</li>
</ul>
</li>
<li>chrom (<strong><em>type=string</em></strong>)
<ul>
<li>the chromosome name<br>
</li>
</ul>
</li>
<li>start (<strong><em>type=integer</em></strong>)
<ul>
<li>the start of the location, 0-based positioning<br>
</li>
</ul>
</li>
<li>end (<strong><em>type=integer</em></strong>)
<ul>
<li>the end of the location, 0-based positioning</li>
</ul>
</li>
</ul></section><section id="optional-6" class="level3"><h3 class="anchored" data-anchor-id="optional-6">Optional</h3>
<ul>
<li>ref_seq (<strong><em>type=string</em></strong>)
<ul>
<li>the reference sequence of this genomic location<br>
</li>
</ul>
</li>
<li>strand (<strong><em>type=string</em></strong>)
<ul>
<li>which strand the location is, either + for plus strand or - for negative strand</li>
</ul>
</li>
</ul></section><section id="example-6" class="level3"><h3 class="anchored" data-anchor-id="example-6">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="markerofinterest" class="level2"><h2 class="anchored" data-anchor-id="markerofinterest">MarkerOfInterest</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MarkerOfInterest/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MarkerOfInterest/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMarkerOfInterest" aria-expanded="false" aria-controls="collapseExample"> Show MarkerOfInterest fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMarkerOfInterest" class="collapse">
<div class="card card-body">
<section id="required-7" class="level3"><h3 class="anchored" data-anchor-id="required-7">Required</h3>
</section><section id="optional-7" class="level3"><h3 class="anchored" data-anchor-id="optional-7">Optional</h3>
<ul>
<li>associations (<strong><em>type=array</em></strong>)
<ul>
<li>a list of associations with this marker, e.g.&nbsp;SP resistance, etc<br>
</li>
</ul>
</li>
<li>marker_location (<strong><em>type=<a href="#genomiclocation">GenomicLocation</a></em></strong>)
<ul>
<li>the genomic location</li>
</ul>
</li>
</ul></section><section id="example-7" class="level3"><h3 class="anchored" data-anchor-id="example-7">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="maskinginfo" class="level2"><h2 class="anchored" data-anchor-id="maskinginfo">MaskingInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MaskingInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MaskingInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMaskingInfo" aria-expanded="false" aria-controls="collapseExample"> Show MaskingInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMaskingInfo" class="collapse">
<div class="card card-body">
<section id="required-8" class="level3"><h3 class="anchored" data-anchor-id="required-8">Required</h3>
<ul>
<li>seq_start (<strong><em>type=integer</em></strong>)
<ul>
<li>the start of the masking<br>
</li>
</ul>
</li>
<li>seq_segment_size (<strong><em>type=integer</em></strong>)
<ul>
<li>the size of the masking<br>
</li>
</ul>
</li>
<li>replacement_size (<strong><em>type=integer</em></strong>)
<ul>
<li>the size of replacement mask</li>
</ul>
</li>
</ul></section><section id="example-8" class="level3"><h3 class="anchored" data-anchor-id="example-8">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="microhaplotypefortarget" class="level2"><h2 class="anchored" data-anchor-id="microhaplotypefortarget">MicrohaplotypeForTarget</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypeForTarget/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypeForTarget/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypeForTarget" aria-expanded="false" aria-controls="collapseExample"> Show MicrohaplotypeForTarget fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMicrohaplotypeForTarget" class="collapse">
<div class="card card-body">
<section id="required-9" class="level3"><h3 class="anchored" data-anchor-id="required-9">Required</h3>
<ul>
<li>mhap_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index for a microhaplotype for a target in the microhaplotypes_info list, e.g.&nbsp;microhaplotypes_info[mhaps_target_id][mhap_id]<br>
</li>
</ul>
</li>
<li>reads (<strong><em>type=integer</em></strong>)
<ul>
<li>the read count associated with this microhaplotype</li>
</ul>
</li>
</ul></section><section id="optional-8" class="level3"><h3 class="anchored" data-anchor-id="optional-8">Optional</h3>
<ul>
<li>umis (<strong><em>type=integer</em></strong>)
<ul>
<li>the unique molecular identifier (umi) count associated with this microhaplotype</li>
</ul>
</li>
</ul></section><section id="example-9" class="level3"><h3 class="anchored" data-anchor-id="example-9">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="microhaplotypesdetected" class="level2"><h2 class="anchored" data-anchor-id="microhaplotypesdetected">MicrohaplotypesDetected</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesDetected/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesDetected/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesDetected" aria-expanded="false" aria-controls="collapseExample"> Show MicrohaplotypesDetected fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMicrohaplotypesDetected" class="collapse">
<div class="card card-body">
<section id="required-10" class="level3"><h3 class="anchored" data-anchor-id="required-10">Required</h3>
<ul>
<li>bioinformatics_run_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into bioinformatics_run_info list<br>
</li>
</ul>
</li>
<li>experiment_samples (<strong><em>type=<a href="#microhaplotypesforsample">MicrohaplotypesForSample</a></em></strong>)
<ul>
<li>a list of the microhaplotypes detected for a sample by targets</li>
</ul>
</li>
</ul></section><section id="example-10" class="level3"><h3 class="anchored" data-anchor-id="example-10">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="microhaplotypesforsample" class="level2"><h2 class="anchored" data-anchor-id="microhaplotypesforsample">MicrohaplotypesForSample</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForSample/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForSample/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesForSample" aria-expanded="false" aria-controls="collapseExample"> Show MicrohaplotypesForSample fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMicrohaplotypesForSample" class="collapse">
<div class="card card-body">
<section id="required-11" class="level3"><h3 class="anchored" data-anchor-id="required-11">Required</h3>
<ul>
<li>experiment_sample_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the experiment_info list<br>
</li>
</ul>
</li>
<li>target_results (<strong><em>type=<a href="#microhaplotypesfortarget">MicrohaplotypesForTarget</a></em></strong>)
<ul>
<li>a list of the microhaplotypes detected for a list of targets</li>
</ul>
</li>
</ul></section><section id="example-11" class="level3"><h3 class="anchored" data-anchor-id="example-11">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="microhaplotypesfortarget" class="level2"><h2 class="anchored" data-anchor-id="microhaplotypesfortarget">MicrohaplotypesForTarget</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForTarget/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForTarget/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesForTarget" aria-expanded="false" aria-controls="collapseExample"> Show MicrohaplotypesForTarget fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseMicrohaplotypesForTarget" class="collapse">
<div class="card card-body">
<section id="required-12" class="level3"><h3 class="anchored" data-anchor-id="required-12">Required</h3>
<ul>
<li>mhaps_target_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index for a target in the microhaplotypes_info list<br>
</li>
</ul>
</li>
<li>haps (<strong><em>type=<a href="#microhaplotypefortarget">MicrohaplotypeForTarget</a></em></strong>)
<ul>
<li>a list of the microhaplotypes detected for this target</li>
</ul>
</li>
</ul></section><section id="example-12" class="level3"><h3 class="anchored" data-anchor-id="example-12">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="panelinfo" class="level2"><h2 class="anchored" data-anchor-id="panelinfo">PanelInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/PanelInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/PanelInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePanelInfo" aria-expanded="false" aria-controls="collapseExample"> Show PanelInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapsePanelInfo" class="collapse">
<div class="card card-body">
<section id="required-13" class="level3"><h3 class="anchored" data-anchor-id="required-13">Required</h3>
<ul>
<li>reactions (<strong><em>type=<a href="#reactioninfo">ReactionInfo</a></em></strong>)
<ul>
<li>a list of 1 or more reactions that this panel contains, each reactions list the targets that were amplified in that reaction, e.g.&nbsp;pool1, pool2<br>
</li>
</ul>
</li>
<li>panel_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name for the panel</li>
</ul>
</li>
</ul></section><section id="example-13" class="level3"><h3 class="anchored" data-anchor-id="example-13">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="parasitedensity" class="level2"><h2 class="anchored" data-anchor-id="parasitedensity">ParasiteDensity</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ParasiteDensity/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ParasiteDensity/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseParasiteDensity" aria-expanded="false" aria-controls="collapseExample"> Show ParasiteDensity fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseParasiteDensity" class="collapse">
<div class="card card-body">
<section id="required-14" class="level3"><h3 class="anchored" data-anchor-id="required-14">Required</h3>
<ul>
<li>method (<strong><em>type=string</em></strong>)
<ul>
<li>the method of how this density was obtained<br>
</li>
</ul>
</li>
<li>density (<strong><em>type=number</em></strong>)
<ul>
<li>the density in microliters</li>
</ul>
</li>
</ul></section><section id="example-14" class="level3"><h3 class="anchored" data-anchor-id="example-14">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="pmogenerationmethod" class="level2"><h2 class="anchored" data-anchor-id="pmogenerationmethod">PmoGenerationMethod</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/PmoGenerationMethod/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/PmoGenerationMethod/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePmoGenerationMethod" aria-expanded="false" aria-controls="collapseExample"> Show PmoGenerationMethod fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapsePmoGenerationMethod" class="collapse">
<div class="card card-body">
<section id="required-15" class="level3"><h3 class="anchored" data-anchor-id="required-15">Required</h3>
<ul>
<li>program_version (<strong><em>type=string</em></strong>)
<ul>
<li>the version of generation method, should be in the format of v[MAJOR].[MINOR].[PATCH]<br>
</li>
</ul>
</li>
<li>program_name (<strong><em>type=string</em></strong>)
<ul>
<li>the name of the program</li>
</ul>
</li>
</ul></section><section id="example-15" class="level3"><h3 class="anchored" data-anchor-id="example-15">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="pmoheader" class="level2"><h2 class="anchored" data-anchor-id="pmoheader">PmoHeader</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/PmoHeader/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/PmoHeader/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePmoHeader" aria-expanded="false" aria-controls="collapseExample"> Show PmoHeader fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapsePmoHeader" class="collapse">
<div class="card card-body">
<section id="required-16" class="level3"><h3 class="anchored" data-anchor-id="required-16">Required</h3>
<ul>
<li>pmo_version (<strong><em>type=string</em></strong>)
<ul>
<li>the version of the PMO file, should be in the format of v[MAJOR].[MINOR].[PATCH]</li>
</ul>
</li>
</ul></section><section id="optional-9" class="level3"><h3 class="anchored" data-anchor-id="optional-9">Optional</h3>
<ul>
<li>creation_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date of when the PMO file was created or modified, should be YYYY-MM-DD<br>
</li>
</ul>
</li>
<li>generation_method (<strong><em>type=<a href="#pmogenerationmethod">PmoGenerationMethod</a></em></strong>)
<ul>
<li>the generation method to create this PMO</li>
</ul>
</li>
</ul></section><section id="example-16" class="level3"><h3 class="anchored" data-anchor-id="example-16">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="primerinfo" class="level2"><h2 class="anchored" data-anchor-id="primerinfo">PrimerInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/PrimerInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/PrimerInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePrimerInfo" aria-expanded="false" aria-controls="collapseExample"> Show PrimerInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapsePrimerInfo" class="collapse">
<div class="card card-body">
<section id="required-17" class="level3"><h3 class="anchored" data-anchor-id="required-17">Required</h3>
<ul>
<li>seq (<strong><em>type=string</em></strong>)
<ul>
<li>the DNA sequence</li>
</ul>
</li>
</ul></section><section id="optional-10" class="level3"><h3 class="anchored" data-anchor-id="optional-10">Optional</h3>
<ul>
<li>location (<strong><em>type=<a href="#genomiclocation">GenomicLocation</a></em></strong>)
<ul>
<li>what the intended genomic location of the primer is</li>
</ul>
</li>
</ul></section><section id="example-17" class="level3"><h3 class="anchored" data-anchor-id="example-17">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="reactioninfo" class="level2"><h2 class="anchored" data-anchor-id="reactioninfo">ReactionInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ReactionInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ReactionInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReactionInfo" aria-expanded="false" aria-controls="collapseExample"> Show ReactionInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseReactionInfo" class="collapse">
<div class="card card-body">
<section id="required-18" class="level3"><h3 class="anchored" data-anchor-id="required-18">Required</h3>
<ul>
<li>panel_targets (<strong><em>type=array</em></strong>)
<ul>
<li>a list of the target indexes in the target_info list<br>
</li>
</ul>
</li>
<li>reaction_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name for this reaction</li>
</ul>
</li>
</ul></section><section id="example-18" class="level3"><h3 class="anchored" data-anchor-id="example-18">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="readcountsbystage" class="level2"><h2 class="anchored" data-anchor-id="readcountsbystage">ReadCountsByStage</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStage/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStage/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStage" aria-expanded="false" aria-controls="collapseExample"> Show ReadCountsByStage fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseReadCountsByStage" class="collapse">
<div class="card card-body">
<section id="required-19" class="level3"><h3 class="anchored" data-anchor-id="required-19">Required</h3>
<ul>
<li>bioinformatics_run_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into bioinformatics_run_info list<br>
</li>
</ul>
</li>
<li>read_counts_by_experimental_sample_by_stage (<strong><em>type=<a href="#readcountsbystageforexperimentalsample">ReadCountsByStageForExperimentalSample</a></em></strong>)
<ul>
<li>a list by experiment_sample for the counts at each stage</li>
</ul>
</li>
</ul></section><section id="example-19" class="level3"><h3 class="anchored" data-anchor-id="example-19">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="readcountsbystageforexperimentalsample" class="level2"><h2 class="anchored" data-anchor-id="readcountsbystageforexperimentalsample">ReadCountsByStageForExperimentalSample</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForExperimentalSample/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForExperimentalSample/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStageForExperimentalSample" aria-expanded="false" aria-controls="collapseExample"> Show ReadCountsByStageForExperimentalSample fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseReadCountsByStageForExperimentalSample" class="collapse">
<div class="card card-body">
<section id="required-20" class="level3"><h3 class="anchored" data-anchor-id="required-20">Required</h3>
<ul>
<li>experiment_sample_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the experiment_info list<br>
</li>
</ul>
</li>
<li>total_raw_count (<strong><em>type=integer</em></strong>)
<ul>
<li>the raw counts off the sequencing machine that a sample began with</li>
</ul>
</li>
</ul></section><section id="optional-11" class="level3"><h3 class="anchored" data-anchor-id="optional-11">Optional</h3>
<ul>
<li>read_counts_for_targets (<strong><em>type=<a href="#readcountsbystagefortarget">ReadCountsByStageForTarget</a></em></strong>)
<ul>
<li>a list of counts by stage for a target</li>
</ul>
</li>
</ul></section><section id="example-20" class="level3"><h3 class="anchored" data-anchor-id="example-20">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="readcountsbystagefortarget" class="level2"><h2 class="anchored" data-anchor-id="readcountsbystagefortarget">ReadCountsByStageForTarget</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForTarget/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForTarget/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStageForTarget" aria-expanded="false" aria-controls="collapseExample"> Show ReadCountsByStageForTarget fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseReadCountsByStageForTarget" class="collapse">
<div class="card card-body">
<section id="required-21" class="level3"><h3 class="anchored" data-anchor-id="required-21">Required</h3>
<ul>
<li>target_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the target_info list<br>
</li>
</ul>
</li>
<li>stages (<strong><em>type=<a href="#stagereadcounts">StageReadCounts</a></em></strong>)
<ul>
<li>the read counts by each stage</li>
</ul>
</li>
</ul></section><section id="example-21" class="level3"><h3 class="anchored" data-anchor-id="example-21">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="representativemicrohaplotype" class="level2"><h2 class="anchored" data-anchor-id="representativemicrohaplotype">RepresentativeMicrohaplotype</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotype/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotype/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotype" aria-expanded="false" aria-controls="collapseExample"> Show RepresentativeMicrohaplotype fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseRepresentativeMicrohaplotype" class="collapse">
<div class="card card-body">
<section id="required-22" class="level3"><h3 class="anchored" data-anchor-id="required-22">Required</h3>
<ul>
<li>seq (<strong><em>type=string</em></strong>)
<ul>
<li>the DNA sequence</li>
</ul>
</li>
</ul></section><section id="optional-12" class="level3"><h3 class="anchored" data-anchor-id="optional-12">Optional</h3>
<ul>
<li>alt_annotations (<strong><em>type=array</em></strong>)
<ul>
<li>a list of additional annotations associated with this microhaplotype, e.g.&nbsp;wildtype, amino acid changes etc<br>
</li>
</ul>
</li>
<li>masking (<strong><em>type=<a href="#maskinginfo">MaskingInfo</a></em></strong>)
<ul>
<li>masking info for the sequence<br>
</li>
</ul>
</li>
<li>microhaplotype_name (<strong><em>type=string</em></strong>)
<ul>
<li>an optional name for this microhaplotype<br>
</li>
</ul>
</li>
<li>pseudocigar (<strong><em>type=string</em></strong>)
<ul>
<li>the pseudocigar of the haplotype<br>
</li>
</ul>
</li>
<li>quality (<strong><em>type=string</em></strong>)
<ul>
<li>the ansi fastq per base quality score for this sequence, this is optional</li>
</ul>
</li>
</ul></section><section id="example-22" class="level3"><h3 class="anchored" data-anchor-id="example-22">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="representativemicrohaplotypes" class="level2"><h2 class="anchored" data-anchor-id="representativemicrohaplotypes">RepresentativeMicrohaplotypes</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypes/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypes/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotypes" aria-expanded="false" aria-controls="collapseExample"> Show RepresentativeMicrohaplotypes fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseRepresentativeMicrohaplotypes" class="collapse">
<div class="card card-body">
<section id="required-23" class="level3"><h3 class="anchored" data-anchor-id="required-23">Required</h3>
<ul>
<li>targets (<strong><em>type=<a href="#representativemicrohaplotypesfortarget">RepresentativeMicrohaplotypesForTarget</a></em></strong>)
<ul>
<li>a list of the microhaplotype for each targets</li>
</ul>
</li>
</ul></section><section id="example-23" class="level3"><h3 class="anchored" data-anchor-id="example-23">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="representativemicrohaplotypesfortarget" class="level2"><h2 class="anchored" data-anchor-id="representativemicrohaplotypesfortarget">RepresentativeMicrohaplotypesForTarget</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypesForTarget/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypesForTarget/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotypesForTarget" aria-expanded="false" aria-controls="collapseExample"> Show RepresentativeMicrohaplotypesForTarget fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseRepresentativeMicrohaplotypesForTarget" class="collapse">
<div class="card card-body">
<section id="required-24" class="level3"><h3 class="anchored" data-anchor-id="required-24">Required</h3>
<ul>
<li>target_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the index into the target_info list<br>
</li>
</ul>
</li>
<li>microhaplotypes (<strong><em>type=<a href="#representativemicrohaplotype">RepresentativeMicrohaplotype</a></em></strong>)
<ul>
<li>a list of the microhaplotypes detected for a target</li>
</ul>
</li>
</ul></section><section id="example-24" class="level3"><h3 class="anchored" data-anchor-id="example-24">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="sequencinginfo" class="level2"><h2 class="anchored" data-anchor-id="sequencinginfo">SequencingInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/SequencingInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/SequencingInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseSequencingInfo" aria-expanded="false" aria-controls="collapseExample"> Show SequencingInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseSequencingInfo" class="collapse">
<div class="card card-body">
<section id="required-25" class="level3"><h3 class="anchored" data-anchor-id="required-25">Required</h3>
<ul>
<li>sequencing_info_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name of for the sequencing done, e.g.&nbsp;batch1<br>
</li>
</ul>
</li>
<li>seq_instrument (<strong><em>type=string</em></strong>)
<ul>
<li>the sequencing instrument used to sequence the run, e.g.&nbsp;ILLUMINA, Illumina MiSeq<br>
</li>
</ul>
</li>
<li>seq_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date of sequencing, should be YYYY-MM or YYYY-MM-DD</li>
</ul>
</li>
</ul></section><section id="optional-13" class="level3"><h3 class="anchored" data-anchor-id="optional-13">Optional</h3>
<ul>
<li>lib_kit (<strong><em>type=string</em></strong>)
<ul>
<li>Name, version, and applicable cell or cycle numbers for the kit used to prepare libraries and load cells or chips for sequencing. If possible, include a part number, e.g.&nbsp;MiSeq Reagent Kit v3 (150-cycle), MS-102-3001<br>
</li>
</ul>
</li>
<li>lib_layout (<strong><em>type=string</em></strong>)
<ul>
<li>Specify the configuration of reads, e.g.&nbsp;paired-end<br>
</li>
</ul>
</li>
<li>lib_screen (<strong><em>type=string</em></strong>)
<ul>
<li>Describe enrichment, screening, or normalization methods applied during amplification or library preparation, e.g.&nbsp;size selection 390bp, diluted to 1 ng DNA/sample<br>
</li>
</ul>
</li>
<li>nucl_acid_amp (<strong><em>type=string</em></strong>)
<ul>
<li>Link to a reference or kit that describes the enzymatic amplification of nucleic acids<br>
</li>
</ul>
</li>
<li>nucl_acid_amp_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date of the nucleoacid amplification<br>
</li>
</ul>
</li>
<li>nucl_acid_ext (<strong><em>type=string</em></strong>)
<ul>
<li>Link to a reference or kit that describes the recovery of nucleic acids from the sample<br>
</li>
</ul>
</li>
<li>nucl_acid_ext_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date of the nucleoacid extraction<br>
</li>
</ul>
</li>
<li>pcr_cond (<strong><em>type=string</em></strong>)
<ul>
<li>the method/conditions for PCR, List PCR cycles used to amplify the target<br>
</li>
</ul>
</li>
<li>seq_center (<strong><em>type=string</em></strong>)
<ul>
<li>Name of facility where sequencing was performed (lab, core facility, or company)</li>
</ul>
</li>
</ul></section><section id="example-25" class="level3"><h3 class="anchored" data-anchor-id="example-25">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="specimeninfo" class="level2"><h2 class="anchored" data-anchor-id="specimeninfo">SpecimenInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/SpecimenInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/SpecimenInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseSpecimenInfo" aria-expanded="false" aria-controls="collapseExample"> Show SpecimenInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseSpecimenInfo" class="collapse">
<div class="card card-body">
<section id="required-26" class="level3"><h3 class="anchored" data-anchor-id="required-26">Required</h3>
<ul>
<li>specimen_name (<strong><em>type=string</em></strong>)
<ul>
<li>an identifier for the specimen<br>
</li>
</ul>
</li>
<li>samp_taxon_id (<strong><em>type=integer</em></strong>)
<ul>
<li>the NCBI taxonomy number of the organism of interest<br>
</li>
</ul>
</li>
<li>collection_date (<strong><em>type=string</em></strong>)
<ul>
<li>the date of the sample collection<br>
</li>
</ul>
</li>
<li>collection_country (<strong><em>type=string</em></strong>)
<ul>
<li>the name of country collected in, would be the same as admin level 0<br>
</li>
</ul>
</li>
<li>collector (<strong><em>type=string</em></strong>)
<ul>
<li>the name of the primary person managing the specimen<br>
</li>
</ul>
</li>
<li>samp_store_loc (<strong><em>type=string</em></strong>)
<ul>
<li>the sample store site, address or facility name<br>
</li>
</ul>
</li>
<li>samp_collect_device (<strong><em>type=string</em></strong>)
<ul>
<li>the way the sample was collected, e.g.&nbsp;whole blood, dried blood spot, etc<br>
</li>
</ul>
</li>
<li>project_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name of the project under which the sample is organized</li>
</ul>
</li>
</ul></section><section id="optional-14" class="level3"><h3 class="anchored" data-anchor-id="optional-14">Optional</h3>
<ul>
<li>alternate_identifiers (<strong><em>type=array</em></strong>)
<ul>
<li>a list of optional alternative names for the samples<br>
</li>
</ul>
</li>
<li>date_of_birth (<strong><em>type=string</em></strong>)
<ul>
<li>if specimen is from a person, the date of birth of that person<br>
</li>
</ul>
</li>
<li>geo_admin1 (<strong><em>type=string</em></strong>)
<ul>
<li>geographical admin level 1, the secondary large demarcation of a nation (nation = admin level 0)<br>
</li>
</ul>
</li>
<li>geo_admin2 (<strong><em>type=string</em></strong>)
<ul>
<li>geographical admin level 2, the third large demarcation of a nation (nation = admin level 0)<br>
</li>
</ul>
</li>
<li>geo_admin3 (<strong><em>type=string</em></strong>)
<ul>
<li>geographical admin level 3, the third large demarcation of a nation (nation = admin level 0)<br>
</li>
</ul>
</li>
<li>host_taxon_id (<strong><em>type=integer</em></strong>)
<ul>
<li>optional the NCBI taxonomy number of the host of the organism<br>
</li>
</ul>
</li>
<li>individual_id (<strong><em>type=integer</em></strong>)
<ul>
<li>an identifier for the individual a specimen was collected from<br>
</li>
</ul>
</li>
<li>lat_lon (<strong><em>type=string</em></strong>)
<ul>
<li>the latitude and longitude of the collection site of the specimen<br>
</li>
</ul>
</li>
<li>parasite_density_info (<strong><em>type=<a href="#parasitedensity">ParasiteDensity</a></em></strong>)
<ul>
<li>one or more parasite densities in microliters for this specimen<br>
</li>
</ul>
</li>
<li>plate_col (<strong><em>type=integer</em></strong>)
<ul>
<li>the column the specimen was in<br>
</li>
</ul>
</li>
<li>plate_name (<strong><em>type=string</em></strong>)
<ul>
<li>a name of plate the specimen was in<br>
</li>
</ul>
</li>
<li>plate_row (<strong><em>type=string</em></strong>)
<ul>
<li>the row the specimen was in<br>
</li>
</ul>
</li>
<li>sample_comments (<strong><em>type=string</em></strong>)
<ul>
<li>any additional comments about the sample<br>
</li>
</ul>
</li>
<li>specimen_sex (<strong><em>type=string</em></strong>)
<ul>
<li>if specimen is from a person, the sex listed for that person</li>
</ul>
</li>
</ul></section><section id="example-26" class="level3"><h3 class="anchored" data-anchor-id="example-26">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="stagereadcounts" class="level2"><h2 class="anchored" data-anchor-id="stagereadcounts">StageReadCounts</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/StageReadCounts/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/StageReadCounts/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseStageReadCounts" aria-expanded="false" aria-controls="collapseExample"> Show StageReadCounts fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseStageReadCounts" class="collapse">
<div class="card card-body">
<section id="required-27" class="level3"><h3 class="anchored" data-anchor-id="required-27">Required</h3>
<ul>
<li>read_count (<strong><em>type=integer</em></strong>)
<ul>
<li>the read counts<br>
</li>
</ul>
</li>
<li>stage (<strong><em>type=string</em></strong>)
<ul>
<li>the stage of the pipeline, e.g.&nbsp;demultiplexed, denoised, etc</li>
</ul>
</li>
</ul></section><section id="example-27" class="level3"><h3 class="anchored" data-anchor-id="example-27">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
</section><section id="targetinfo" class="level2"><h2 class="anchored" data-anchor-id="targetinfo">TargetInfo</h2>
<a href="https://plasmogenepi.github.io/portable-microhaplotype-object/TargetInfo/" class="uri">https://plasmogenepi.github.io/portable-microhaplotype-object/TargetInfo/</a><br><a class="btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseTargetInfo" aria-expanded="false" aria-controls="collapseExample"> Show TargetInfo fields <i class="fa-solid fa-caret-right"></i> </a>
<div id="collapseTargetInfo" class="collapse">
<div class="card card-body">
<section id="required-28" class="level3"><h3 class="anchored" data-anchor-id="required-28">Required</h3>
<ul>
<li>target_name (<strong><em>type=string</em></strong>)
<ul>
<li>an identifier for this target<br>
</li>
</ul>
</li>
<li>forward_primers (<strong><em>type=<a href="#primerinfo">PrimerInfo</a></em></strong>)
<ul>
<li>A list of forward primers associated with this target<br>
</li>
</ul>
</li>
<li>reverse_primers (<strong><em>type=<a href="#primerinfo">PrimerInfo</a></em></strong>)
<ul>
<li>A list of reverse primers associated with this target</li>
</ul>
</li>
</ul></section><section id="optional-15" class="level3"><h3 class="anchored" data-anchor-id="optional-15">Optional</h3>
<ul>
<li>gene_name (<strong><em>type=string</em></strong>)
<ul>
<li>an identifier of the gene, if any, is being covered with this targeted<br>
</li>
</ul>
</li>
<li>insert_location (<strong><em>type=<a href="#genomiclocation">GenomicLocation</a></em></strong>)
<ul>
<li>the intended genomic location of the insert of the amplicon (the location between the end of the forward primer and the beginning of the reverse primer)<br>
</li>
</ul>
</li>
<li>target_attributes (<strong><em>type=array</em></strong>)
<ul>
<li>a list of classification type for the primer target</li>
</ul>
</li>
</ul></section><section id="example-28" class="level3"><h3 class="anchored" data-anchor-id="example-28">Example</h3>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</div>
</div>
<p><br></p>
<p><i class="fa-solid fa-dna" aria-label="dna"></i></p>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb30" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> PMO fields overview </span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># format: docx</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, echo=FALSE}</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="in">source("../common.R")</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="fu"># Goal  </span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>Creating fields with efforts to be consistent with <span class="co">[</span><span class="ot">MIxS standards</span><span class="co">](https://genomicsstandardsconsortium.github.io/mixs/)</span> </span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>Important aspects to keep in mind are creating an efficient, low-weight, and minimum amount of information about a run without losing any important data. We can build tools around this table to generate certain fields that are important but not necessary always to keep constantly stored in this base class (e.g. SNP/INDEL calls). Also since we are proposing to keep this data in a singular file in <span class="co">[</span><span class="ot">JSON format</span><span class="co">](https://en.wikipedia.org/wiki/JSON)</span> we are not limited to keeping data in a tabular format for organization; output generated from this file can certainly be a table but we don’t have to store things as a table, e.g. certain fields might be a single ID while other fields might be vectors/lists of data. What’s most important is that we agree on what fields are important and what they should store (and what values should be allowable or formatting, e.g. double/floats vs strings vs characters vs POSIX date vs URL etc)</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>Format is defined by utilizing <span class="co">[</span><span class="ot">LinkML</span><span class="co">](https://linkml.io/linkml/)</span> to generate a general data scheme which creates convenient outputs like <span class="co">[</span><span class="ot">JSON Schema</span><span class="co">](https://json-schema.org/)</span> for validation tools. </span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>Other notable users of LinkML/MIxS <span class="co">[</span><span class="ot">National Microbiome Data Collaborative Schema</span><span class="co">](https://github.com/microbiomedata/nmdc-schema)</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a><span class="fu"># Overview   </span></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>Below is an overview of the entire format currently in alpha development and is still under heavy development and optimization so structure and names are subject to undergo significant changes as development continues. Please send questions to <span class="co">[</span><span class="ot">info@plasmogenepi.org</span><span class="co">](mailto:info@plasmogenepi.org)</span></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>A more informative overview can also be viewed here <span class="ot">&lt;https://drive.google.com/file/d/193B1BBLagay_CYhJm4Nee7Fl4T4BFdL_/view?usp=sharing&gt;</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://github.com/PlasmoGenEpi/portable-microhaplotype-object&gt;</span></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>:::{.column-screen}</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a><span class="in">```{mermaid}</span></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a><span class="in">%%| fig-align: center </span></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a><span class="in">erDiagram</span></span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject {</span></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStage {</span></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a><span class="in">    integer bioinformatics_run_id  </span></span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStageForExperimentalSample {</span></span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a><span class="in">    integer experiment_sample_id  </span></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a><span class="in">    integer total_raw_count  </span></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStageForTarget {</span></span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a><span class="in">    integer target_id  </span></span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a><span class="in">StageReadCounts {</span></span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a><span class="in">    integer read_count  </span></span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a><span class="in">    string stage  </span></span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a><span class="in">PmoHeader {</span></span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a><span class="in">    string pmo_version  </span></span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a><span class="in">    string creation_date  </span></span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a><span class="in">PmoGenerationMethod {</span></span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a><span class="in">    string program_version  </span></span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a><span class="in">    string program_name  </span></span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesDetected {</span></span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a><span class="in">    integer bioinformatics_run_id  </span></span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesForSample {</span></span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a><span class="in">    integer experiment_sample_id  </span></span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesForTarget {</span></span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a><span class="in">    integer mhaps_target_id  </span></span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypeForTarget {</span></span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a><span class="in">    integer mhap_id  </span></span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a><span class="in">    integer reads  </span></span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a><span class="in">    integer umis  </span></span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a><span class="in">BioinformaticsRunInfo {</span></span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a><span class="in">    integer bioinformatics_methods_id  </span></span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a><span class="in">    string run_date  </span></span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a><span class="in">    string bioinformatics_run_name  </span></span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a><span class="in">BioinformaticsMethodInfo {</span></span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a><span class="in">    string bioinformatics_method_name  </span></span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a><span class="in">BioMethod {</span></span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a><span class="in">    string program_version  </span></span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a><span class="in">    string program  </span></span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a><span class="in">    string program_description  </span></span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a><span class="in">    stringList additional_argument  </span></span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-91"><a href="#cb30-91" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotypes {</span></span>
<span id="cb30-92"><a href="#cb30-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-93"><a href="#cb30-93" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-94"><a href="#cb30-94" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotypesForTarget {</span></span>
<span id="cb30-95"><a href="#cb30-95" aria-hidden="true" tabindex="-1"></a><span class="in">    integer target_id  </span></span>
<span id="cb30-96"><a href="#cb30-96" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-97"><a href="#cb30-97" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotype {</span></span>
<span id="cb30-98"><a href="#cb30-98" aria-hidden="true" tabindex="-1"></a><span class="in">    string seq  </span></span>
<span id="cb30-99"><a href="#cb30-99" aria-hidden="true" tabindex="-1"></a><span class="in">    string microhaplotype_name  </span></span>
<span id="cb30-100"><a href="#cb30-100" aria-hidden="true" tabindex="-1"></a><span class="in">    string quality  </span></span>
<span id="cb30-101"><a href="#cb30-101" aria-hidden="true" tabindex="-1"></a><span class="in">    string pseudocigar  </span></span>
<span id="cb30-102"><a href="#cb30-102" aria-hidden="true" tabindex="-1"></a><span class="in">    stringList alt_annotations  </span></span>
<span id="cb30-103"><a href="#cb30-103" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-104"><a href="#cb30-104" aria-hidden="true" tabindex="-1"></a><span class="in">MaskingInfo {</span></span>
<span id="cb30-105"><a href="#cb30-105" aria-hidden="true" tabindex="-1"></a><span class="in">    integer seq_start  </span></span>
<span id="cb30-106"><a href="#cb30-106" aria-hidden="true" tabindex="-1"></a><span class="in">    integer seq_segment_size  </span></span>
<span id="cb30-107"><a href="#cb30-107" aria-hidden="true" tabindex="-1"></a><span class="in">    integer replacement_size  </span></span>
<span id="cb30-108"><a href="#cb30-108" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-109"><a href="#cb30-109" aria-hidden="true" tabindex="-1"></a><span class="in">GenomeInfo {</span></span>
<span id="cb30-110"><a href="#cb30-110" aria-hidden="true" tabindex="-1"></a><span class="in">    string name  </span></span>
<span id="cb30-111"><a href="#cb30-111" aria-hidden="true" tabindex="-1"></a><span class="in">    string genome_version  </span></span>
<span id="cb30-112"><a href="#cb30-112" aria-hidden="true" tabindex="-1"></a><span class="in">    integer taxon_id  </span></span>
<span id="cb30-113"><a href="#cb30-113" aria-hidden="true" tabindex="-1"></a><span class="in">    string url  </span></span>
<span id="cb30-114"><a href="#cb30-114" aria-hidden="true" tabindex="-1"></a><span class="in">    stringList chromosomes  </span></span>
<span id="cb30-115"><a href="#cb30-115" aria-hidden="true" tabindex="-1"></a><span class="in">    string gff_url  </span></span>
<span id="cb30-116"><a href="#cb30-116" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-117"><a href="#cb30-117" aria-hidden="true" tabindex="-1"></a><span class="in">TargetInfo {</span></span>
<span id="cb30-118"><a href="#cb30-118" aria-hidden="true" tabindex="-1"></a><span class="in">    string target_name  </span></span>
<span id="cb30-119"><a href="#cb30-119" aria-hidden="true" tabindex="-1"></a><span class="in">    string gene_name  </span></span>
<span id="cb30-120"><a href="#cb30-120" aria-hidden="true" tabindex="-1"></a><span class="in">    stringList target_attributes  </span></span>
<span id="cb30-121"><a href="#cb30-121" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-122"><a href="#cb30-122" aria-hidden="true" tabindex="-1"></a><span class="in">PrimerInfo {</span></span>
<span id="cb30-123"><a href="#cb30-123" aria-hidden="true" tabindex="-1"></a><span class="in">    string seq  </span></span>
<span id="cb30-124"><a href="#cb30-124" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-125"><a href="#cb30-125" aria-hidden="true" tabindex="-1"></a><span class="in">GenomicLocation {</span></span>
<span id="cb30-126"><a href="#cb30-126" aria-hidden="true" tabindex="-1"></a><span class="in">    integer genome_id  </span></span>
<span id="cb30-127"><a href="#cb30-127" aria-hidden="true" tabindex="-1"></a><span class="in">    string chrom  </span></span>
<span id="cb30-128"><a href="#cb30-128" aria-hidden="true" tabindex="-1"></a><span class="in">    integer start  </span></span>
<span id="cb30-129"><a href="#cb30-129" aria-hidden="true" tabindex="-1"></a><span class="in">    integer end  </span></span>
<span id="cb30-130"><a href="#cb30-130" aria-hidden="true" tabindex="-1"></a><span class="in">    string strand  </span></span>
<span id="cb30-131"><a href="#cb30-131" aria-hidden="true" tabindex="-1"></a><span class="in">    string ref_seq  </span></span>
<span id="cb30-132"><a href="#cb30-132" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-133"><a href="#cb30-133" aria-hidden="true" tabindex="-1"></a><span class="in">PanelInfo {</span></span>
<span id="cb30-134"><a href="#cb30-134" aria-hidden="true" tabindex="-1"></a><span class="in">    string panel_name  </span></span>
<span id="cb30-135"><a href="#cb30-135" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-136"><a href="#cb30-136" aria-hidden="true" tabindex="-1"></a><span class="in">ReactionInfo {</span></span>
<span id="cb30-137"><a href="#cb30-137" aria-hidden="true" tabindex="-1"></a><span class="in">    integerList panel_targets  </span></span>
<span id="cb30-138"><a href="#cb30-138" aria-hidden="true" tabindex="-1"></a><span class="in">    string reaction_name  </span></span>
<span id="cb30-139"><a href="#cb30-139" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-140"><a href="#cb30-140" aria-hidden="true" tabindex="-1"></a><span class="in">SequencingInfo {</span></span>
<span id="cb30-141"><a href="#cb30-141" aria-hidden="true" tabindex="-1"></a><span class="in">    string sequencing_info_name  </span></span>
<span id="cb30-142"><a href="#cb30-142" aria-hidden="true" tabindex="-1"></a><span class="in">    string seq_instrument  </span></span>
<span id="cb30-143"><a href="#cb30-143" aria-hidden="true" tabindex="-1"></a><span class="in">    string seq_date  </span></span>
<span id="cb30-144"><a href="#cb30-144" aria-hidden="true" tabindex="-1"></a><span class="in">    string nucl_acid_ext  </span></span>
<span id="cb30-145"><a href="#cb30-145" aria-hidden="true" tabindex="-1"></a><span class="in">    string nucl_acid_amp  </span></span>
<span id="cb30-146"><a href="#cb30-146" aria-hidden="true" tabindex="-1"></a><span class="in">    string nucl_acid_ext_date  </span></span>
<span id="cb30-147"><a href="#cb30-147" aria-hidden="true" tabindex="-1"></a><span class="in">    string nucl_acid_amp_date  </span></span>
<span id="cb30-148"><a href="#cb30-148" aria-hidden="true" tabindex="-1"></a><span class="in">    string pcr_cond  </span></span>
<span id="cb30-149"><a href="#cb30-149" aria-hidden="true" tabindex="-1"></a><span class="in">    string lib_screen  </span></span>
<span id="cb30-150"><a href="#cb30-150" aria-hidden="true" tabindex="-1"></a><span class="in">    string lib_layout  </span></span>
<span id="cb30-151"><a href="#cb30-151" aria-hidden="true" tabindex="-1"></a><span class="in">    string lib_kit  </span></span>
<span id="cb30-152"><a href="#cb30-152" aria-hidden="true" tabindex="-1"></a><span class="in">    string seq_center  </span></span>
<span id="cb30-153"><a href="#cb30-153" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-154"><a href="#cb30-154" aria-hidden="true" tabindex="-1"></a><span class="in">SpecimenInfo {</span></span>
<span id="cb30-155"><a href="#cb30-155" aria-hidden="true" tabindex="-1"></a><span class="in">    string plate_name  </span></span>
<span id="cb30-156"><a href="#cb30-156" aria-hidden="true" tabindex="-1"></a><span class="in">    string plate_row  </span></span>
<span id="cb30-157"><a href="#cb30-157" aria-hidden="true" tabindex="-1"></a><span class="in">    integer plate_col  </span></span>
<span id="cb30-158"><a href="#cb30-158" aria-hidden="true" tabindex="-1"></a><span class="in">    string specimen_name  </span></span>
<span id="cb30-159"><a href="#cb30-159" aria-hidden="true" tabindex="-1"></a><span class="in">    integer samp_taxon_id  </span></span>
<span id="cb30-160"><a href="#cb30-160" aria-hidden="true" tabindex="-1"></a><span class="in">    integer individual_id  </span></span>
<span id="cb30-161"><a href="#cb30-161" aria-hidden="true" tabindex="-1"></a><span class="in">    integer host_taxon_id  </span></span>
<span id="cb30-162"><a href="#cb30-162" aria-hidden="true" tabindex="-1"></a><span class="in">    stringList alternate_identifiers  </span></span>
<span id="cb30-163"><a href="#cb30-163" aria-hidden="true" tabindex="-1"></a><span class="in">    string specimen_sex  </span></span>
<span id="cb30-164"><a href="#cb30-164" aria-hidden="true" tabindex="-1"></a><span class="in">    string collection_date  </span></span>
<span id="cb30-165"><a href="#cb30-165" aria-hidden="true" tabindex="-1"></a><span class="in">    string date_of_birth  </span></span>
<span id="cb30-166"><a href="#cb30-166" aria-hidden="true" tabindex="-1"></a><span class="in">    string collection_country  </span></span>
<span id="cb30-167"><a href="#cb30-167" aria-hidden="true" tabindex="-1"></a><span class="in">    string geo_admin1  </span></span>
<span id="cb30-168"><a href="#cb30-168" aria-hidden="true" tabindex="-1"></a><span class="in">    string geo_admin2  </span></span>
<span id="cb30-169"><a href="#cb30-169" aria-hidden="true" tabindex="-1"></a><span class="in">    string geo_admin3  </span></span>
<span id="cb30-170"><a href="#cb30-170" aria-hidden="true" tabindex="-1"></a><span class="in">    string lat_lon  </span></span>
<span id="cb30-171"><a href="#cb30-171" aria-hidden="true" tabindex="-1"></a><span class="in">    string collector  </span></span>
<span id="cb30-172"><a href="#cb30-172" aria-hidden="true" tabindex="-1"></a><span class="in">    string samp_store_loc  </span></span>
<span id="cb30-173"><a href="#cb30-173" aria-hidden="true" tabindex="-1"></a><span class="in">    string samp_collect_device  </span></span>
<span id="cb30-174"><a href="#cb30-174" aria-hidden="true" tabindex="-1"></a><span class="in">    string project_name  </span></span>
<span id="cb30-175"><a href="#cb30-175" aria-hidden="true" tabindex="-1"></a><span class="in">    string sample_comments  </span></span>
<span id="cb30-176"><a href="#cb30-176" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-177"><a href="#cb30-177" aria-hidden="true" tabindex="-1"></a><span class="in">ParasiteDensity {</span></span>
<span id="cb30-178"><a href="#cb30-178" aria-hidden="true" tabindex="-1"></a><span class="in">    string method  </span></span>
<span id="cb30-179"><a href="#cb30-179" aria-hidden="true" tabindex="-1"></a><span class="in">    double density  </span></span>
<span id="cb30-180"><a href="#cb30-180" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-181"><a href="#cb30-181" aria-hidden="true" tabindex="-1"></a><span class="in">ExperimentInfo {</span></span>
<span id="cb30-182"><a href="#cb30-182" aria-hidden="true" tabindex="-1"></a><span class="in">    integer sequencing_info_id  </span></span>
<span id="cb30-183"><a href="#cb30-183" aria-hidden="true" tabindex="-1"></a><span class="in">    string plate_name  </span></span>
<span id="cb30-184"><a href="#cb30-184" aria-hidden="true" tabindex="-1"></a><span class="in">    string plate_row  </span></span>
<span id="cb30-185"><a href="#cb30-185" aria-hidden="true" tabindex="-1"></a><span class="in">    integer plate_col  </span></span>
<span id="cb30-186"><a href="#cb30-186" aria-hidden="true" tabindex="-1"></a><span class="in">    integer specimen_id  </span></span>
<span id="cb30-187"><a href="#cb30-187" aria-hidden="true" tabindex="-1"></a><span class="in">    integer panel_id  </span></span>
<span id="cb30-188"><a href="#cb30-188" aria-hidden="true" tabindex="-1"></a><span class="in">    string accession  </span></span>
<span id="cb30-189"><a href="#cb30-189" aria-hidden="true" tabindex="-1"></a><span class="in">    string experiment_sample_name  </span></span>
<span id="cb30-190"><a href="#cb30-190" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb30-191"><a href="#cb30-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-192"><a href="#cb30-192" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| ExperimentInfo : "experiment_info"</span></span>
<span id="cb30-193"><a href="#cb30-193" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| SpecimenInfo : "specimen_info"</span></span>
<span id="cb30-194"><a href="#cb30-194" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| SequencingInfo : "sequencing_info"</span></span>
<span id="cb30-195"><a href="#cb30-195" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| PanelInfo : "panel_info"</span></span>
<span id="cb30-196"><a href="#cb30-196" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| TargetInfo : "target_info"</span></span>
<span id="cb30-197"><a href="#cb30-197" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| GenomeInfo : "targeted_genomes"</span></span>
<span id="cb30-198"><a href="#cb30-198" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--|| RepresentativeMicrohaplotypes : "microhaplotypes_info"</span></span>
<span id="cb30-199"><a href="#cb30-199" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| BioinformaticsMethodInfo : "bioinformatics_methods_info"</span></span>
<span id="cb30-200"><a href="#cb30-200" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| BioinformaticsRunInfo : "bioinformatics_run_info"</span></span>
<span id="cb30-201"><a href="#cb30-201" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}| MicrohaplotypesDetected : "microhaplotypes_detected"</span></span>
<span id="cb30-202"><a href="#cb30-202" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--|| PmoHeader : "pmo_header"</span></span>
<span id="cb30-203"><a href="#cb30-203" aria-hidden="true" tabindex="-1"></a><span class="in">PortableMicrohaplotypeObject ||--}o ReadCountsByStage : "read_counts_by_stage"</span></span>
<span id="cb30-204"><a href="#cb30-204" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStage ||--}| ReadCountsByStageForExperimentalSample : "read_counts_by_experimental_sample_by_stage"</span></span>
<span id="cb30-205"><a href="#cb30-205" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStageForExperimentalSample ||--}o ReadCountsByStageForTarget : "read_counts_for_targets"</span></span>
<span id="cb30-206"><a href="#cb30-206" aria-hidden="true" tabindex="-1"></a><span class="in">ReadCountsByStageForTarget ||--}| StageReadCounts : "stages"</span></span>
<span id="cb30-207"><a href="#cb30-207" aria-hidden="true" tabindex="-1"></a><span class="in">PmoHeader ||--|o PmoGenerationMethod : "generation_method"</span></span>
<span id="cb30-208"><a href="#cb30-208" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesDetected ||--}| MicrohaplotypesForSample : "experiment_samples"</span></span>
<span id="cb30-209"><a href="#cb30-209" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesForSample ||--}| MicrohaplotypesForTarget : "target_results"</span></span>
<span id="cb30-210"><a href="#cb30-210" aria-hidden="true" tabindex="-1"></a><span class="in">MicrohaplotypesForTarget ||--}| MicrohaplotypeForTarget : "haps"</span></span>
<span id="cb30-211"><a href="#cb30-211" aria-hidden="true" tabindex="-1"></a><span class="in">BioinformaticsMethodInfo ||--|| BioMethod : "demultiplexing_method"</span></span>
<span id="cb30-212"><a href="#cb30-212" aria-hidden="true" tabindex="-1"></a><span class="in">BioinformaticsMethodInfo ||--|| BioMethod : "denoising_method"</span></span>
<span id="cb30-213"><a href="#cb30-213" aria-hidden="true" tabindex="-1"></a><span class="in">BioinformaticsMethodInfo ||--}o BioMethod : "additional_methods"</span></span>
<span id="cb30-214"><a href="#cb30-214" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotypes ||--}| RepresentativeMicrohaplotypesForTarget : "targets"</span></span>
<span id="cb30-215"><a href="#cb30-215" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotypesForTarget ||--}| RepresentativeMicrohaplotype : "microhaplotypes"</span></span>
<span id="cb30-216"><a href="#cb30-216" aria-hidden="true" tabindex="-1"></a><span class="in">RepresentativeMicrohaplotype ||--}o MaskingInfo : "masking"</span></span>
<span id="cb30-217"><a href="#cb30-217" aria-hidden="true" tabindex="-1"></a><span class="in">TargetInfo ||--|o GenomicLocation : "insert_location"</span></span>
<span id="cb30-218"><a href="#cb30-218" aria-hidden="true" tabindex="-1"></a><span class="in">TargetInfo ||--}| PrimerInfo : "forward_primers"</span></span>
<span id="cb30-219"><a href="#cb30-219" aria-hidden="true" tabindex="-1"></a><span class="in">TargetInfo ||--}| PrimerInfo : "reverse_primers"</span></span>
<span id="cb30-220"><a href="#cb30-220" aria-hidden="true" tabindex="-1"></a><span class="in">PrimerInfo ||--|o GenomicLocation : "location"</span></span>
<span id="cb30-221"><a href="#cb30-221" aria-hidden="true" tabindex="-1"></a><span class="in">PanelInfo ||--}| ReactionInfo : "reactions"</span></span>
<span id="cb30-222"><a href="#cb30-222" aria-hidden="true" tabindex="-1"></a><span class="in">SpecimenInfo ||--}o ParasiteDensity : "parasite_density_info"</span></span>
<span id="cb30-223"><a href="#cb30-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-224"><a href="#cb30-224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-225"><a href="#cb30-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-226"><a href="#cb30-226" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb30-227"><a href="#cb30-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-228"><a href="#cb30-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-229"><a href="#cb30-229" aria-hidden="true" tabindex="-1"></a><span class="fu">## PortableMicrohaplotypeObject  </span></span>
<span id="cb30-230"><a href="#cb30-230" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/PortableMicrohaplotypeObject/&gt;</span>  </span>
<span id="cb30-231"><a href="#cb30-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-232"><a href="#cb30-232" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-233"><a href="#cb30-233" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>experiment_info (***type=[ExperimentInfo](#experimentinfo)***)  </span>
<span id="cb30-234"><a href="#cb30-234" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of experiments of all the seq/amp of the specimens within this project  </span>
<span id="cb30-235"><a href="#cb30-235" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>specimen_info (***type=[SpecimenInfo](#specimeninfo)***)  </span>
<span id="cb30-236"><a href="#cb30-236" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of specimens of all the samples within this project  </span>
<span id="cb30-237"><a href="#cb30-237" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>sequencing_info (***type=[SequencingInfo](#sequencinginfo)***)  </span>
<span id="cb30-238"><a href="#cb30-238" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of sequencing info for this project  </span>
<span id="cb30-239"><a href="#cb30-239" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>panel_info (***type=[PanelInfo](#panelinfo)***)  </span>
<span id="cb30-240"><a href="#cb30-240" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of info on the panels  </span>
<span id="cb30-241"><a href="#cb30-241" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_info (***type=[TargetInfo](#targetinfo)***)  </span>
<span id="cb30-242"><a href="#cb30-242" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of info on the targets  </span>
<span id="cb30-243"><a href="#cb30-243" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>targeted_genomes (***type=[GenomeInfo](#genomeinfo)***)  </span>
<span id="cb30-244"><a href="#cb30-244" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of genomes that the targets in TargetInfo refer to  </span>
<span id="cb30-245"><a href="#cb30-245" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>microhaplotypes_info (***type=[RepresentativeMicrohaplotypes](#representativemicrohaplotypes)***)  </span>
<span id="cb30-246"><a href="#cb30-246" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the information on the representative microhaplotypes  </span>
<span id="cb30-247"><a href="#cb30-247" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_methods_info (***type=[BioinformaticsMethodInfo](#bioinformaticsmethodinfo)***)  </span>
<span id="cb30-248"><a href="#cb30-248" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the bioinformatics pipeline/methods used to generated the amplicon analysis for this project  </span>
<span id="cb30-249"><a href="#cb30-249" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_run_info (***type=[BioinformaticsRunInfo](#bioinformaticsruninfo)***)  </span>
<span id="cb30-250"><a href="#cb30-250" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the runtime info for the bioinformatics pipeline used to generated the amplicon analysis for this project  </span>
<span id="cb30-251"><a href="#cb30-251" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>microhaplotypes_detected (***type=[MicrohaplotypesDetected](#microhaplotypesdetected)***)  </span>
<span id="cb30-252"><a href="#cb30-252" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the microhaplotypes detected in this projects  </span>
<span id="cb30-253"><a href="#cb30-253" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>pmo_header (***type=[PmoHeader](#pmoheader)***)  </span>
<span id="cb30-254"><a href="#cb30-254" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the PMO information for this file including version etc  </span>
<span id="cb30-255"><a href="#cb30-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-256"><a href="#cb30-256" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-257"><a href="#cb30-257" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>read_counts_by_stage (***type=[ReadCountsByStage](#readcountsbystage)***)  </span>
<span id="cb30-258"><a href="#cb30-258" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the read counts for different stages of the pipeline  </span>
<span id="cb30-259"><a href="#cb30-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-260"><a href="#cb30-260" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-263"><a href="#cb30-263" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-264"><a href="#cb30-264" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-265"><a href="#cb30-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-266"><a href="#cb30-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-267"><a href="#cb30-267" aria-hidden="true" tabindex="-1"></a><span class="fu">## BioMethod  </span></span>
<span id="cb30-268"><a href="#cb30-268" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/BioMethod/&gt;</span>  </span>
<span id="cb30-269"><a href="#cb30-269" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioMethod" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-270"><a href="#cb30-270" aria-hidden="true" tabindex="-1"></a>  Show BioMethod fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-271"><a href="#cb30-271" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-272"><a href="#cb30-272" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseBioMethod"&gt;</span>
<span id="cb30-273"><a href="#cb30-273" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-274"><a href="#cb30-274" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-275"><a href="#cb30-275" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>program_version (***type=string***)  </span>
<span id="cb30-276"><a href="#cb30-276" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the version of generation method, should be in the format of v<span class="co">[</span><span class="ot">MAJOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">MINOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">PATCH</span><span class="co">]</span>  </span>
<span id="cb30-277"><a href="#cb30-277" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>program (***type=string***)  </span>
<span id="cb30-278"><a href="#cb30-278" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>name of the program used for this portion of the pipeline  </span>
<span id="cb30-279"><a href="#cb30-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-280"><a href="#cb30-280" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-281"><a href="#cb30-281" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>additional_argument (***type=array***)  </span>
<span id="cb30-282"><a href="#cb30-282" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>any additional arguments that differ from the default  </span>
<span id="cb30-283"><a href="#cb30-283" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>program_description (***type=string***)  </span>
<span id="cb30-284"><a href="#cb30-284" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a short description of what this method does  </span>
<span id="cb30-285"><a href="#cb30-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-286"><a href="#cb30-286" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-289"><a href="#cb30-289" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-290"><a href="#cb30-290" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-291"><a href="#cb30-291" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-292"><a href="#cb30-292" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-293"><a href="#cb30-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-294"><a href="#cb30-294" aria-hidden="true" tabindex="-1"></a><span class="fu">## BioinformaticsMethodInfo  </span></span>
<span id="cb30-295"><a href="#cb30-295" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsMethodInfo/&gt;</span>  </span>
<span id="cb30-296"><a href="#cb30-296" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioinformaticsMethodInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-297"><a href="#cb30-297" aria-hidden="true" tabindex="-1"></a>  Show BioinformaticsMethodInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-298"><a href="#cb30-298" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-299"><a href="#cb30-299" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseBioinformaticsMethodInfo"&gt;</span>
<span id="cb30-300"><a href="#cb30-300" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-301"><a href="#cb30-301" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-302"><a href="#cb30-302" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>demultiplexing_method (***type=[BioMethod](#biomethod)***)  </span>
<span id="cb30-303"><a href="#cb30-303" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the demultiplexing method used to separate raw reads from barcodes and primer targets  </span>
<span id="cb30-304"><a href="#cb30-304" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>denoising_method (***type=[BioMethod](#biomethod)***)  </span>
<span id="cb30-305"><a href="#cb30-305" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the method used to de-noise and/or cluster the raw reads  </span>
<span id="cb30-306"><a href="#cb30-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-307"><a href="#cb30-307" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-308"><a href="#cb30-308" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>additional_methods (***type=[BioMethod](#biomethod)***)  </span>
<span id="cb30-309"><a href="#cb30-309" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>any additional methods used to analyze the data  </span>
<span id="cb30-310"><a href="#cb30-310" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_method_name (***type=string***)  </span>
<span id="cb30-311"><a href="#cb30-311" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>name of the collection of methods is called, e.g. pipeline   </span>
<span id="cb30-312"><a href="#cb30-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-313"><a href="#cb30-313" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-316"><a href="#cb30-316" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-317"><a href="#cb30-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-318"><a href="#cb30-318" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-319"><a href="#cb30-319" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-320"><a href="#cb30-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-321"><a href="#cb30-321" aria-hidden="true" tabindex="-1"></a><span class="fu">## BioinformaticsRunInfo  </span></span>
<span id="cb30-322"><a href="#cb30-322" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/BioinformaticsRunInfo/&gt;</span>  </span>
<span id="cb30-323"><a href="#cb30-323" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseBioinformaticsRunInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-324"><a href="#cb30-324" aria-hidden="true" tabindex="-1"></a>  Show BioinformaticsRunInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-325"><a href="#cb30-325" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-326"><a href="#cb30-326" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseBioinformaticsRunInfo"&gt;</span>
<span id="cb30-327"><a href="#cb30-327" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-328"><a href="#cb30-328" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-329"><a href="#cb30-329" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_methods_id (***type=integer***)  </span>
<span id="cb30-330"><a href="#cb30-330" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the bioinformatics_methods_info list  </span>
<span id="cb30-331"><a href="#cb30-331" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>run_date (***type=string***)  </span>
<span id="cb30-332"><a href="#cb30-332" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date when the run was done, should be YYYY-MM-DD  </span>
<span id="cb30-333"><a href="#cb30-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-334"><a href="#cb30-334" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-335"><a href="#cb30-335" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_run_name (***type=string***)  </span>
<span id="cb30-336"><a href="#cb30-336" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name to for this run  </span>
<span id="cb30-337"><a href="#cb30-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-338"><a href="#cb30-338" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-341"><a href="#cb30-341" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-342"><a href="#cb30-342" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-343"><a href="#cb30-343" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-344"><a href="#cb30-344" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-345"><a href="#cb30-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-346"><a href="#cb30-346" aria-hidden="true" tabindex="-1"></a><span class="fu">## ExperimentInfo  </span></span>
<span id="cb30-347"><a href="#cb30-347" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ExperimentInfo/&gt;</span>  </span>
<span id="cb30-348"><a href="#cb30-348" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseExperimentInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-349"><a href="#cb30-349" aria-hidden="true" tabindex="-1"></a>  Show ExperimentInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-350"><a href="#cb30-350" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-351"><a href="#cb30-351" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseExperimentInfo"&gt;</span>
<span id="cb30-352"><a href="#cb30-352" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-353"><a href="#cb30-353" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-354"><a href="#cb30-354" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>sequencing_info_id (***type=integer***)  </span>
<span id="cb30-355"><a href="#cb30-355" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the sequencing_info list  </span>
<span id="cb30-356"><a href="#cb30-356" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>specimen_id (***type=integer***)  </span>
<span id="cb30-357"><a href="#cb30-357" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the specimen_info list  </span>
<span id="cb30-358"><a href="#cb30-358" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>panel_id (***type=integer***)  </span>
<span id="cb30-359"><a href="#cb30-359" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the panel_info list  </span>
<span id="cb30-360"><a href="#cb30-360" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>experiment_sample_name (***type=string***)  </span>
<span id="cb30-361"><a href="#cb30-361" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a unique identifier for this sequence/amplification run on a specimen_name  </span>
<span id="cb30-362"><a href="#cb30-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-363"><a href="#cb30-363" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-364"><a href="#cb30-364" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>accession (***type=string***)  </span>
<span id="cb30-365"><a href="#cb30-365" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>ERA/SRA accession number for the sample if it was submitted  </span>
<span id="cb30-366"><a href="#cb30-366" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_col (***type=integer***)  </span>
<span id="cb30-367"><a href="#cb30-367" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the column the specimen was in  </span>
<span id="cb30-368"><a href="#cb30-368" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_name (***type=string***)  </span>
<span id="cb30-369"><a href="#cb30-369" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name of plate the specimen was in  </span>
<span id="cb30-370"><a href="#cb30-370" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_row (***type=string***)  </span>
<span id="cb30-371"><a href="#cb30-371" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the row the specimen was in  </span>
<span id="cb30-372"><a href="#cb30-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-373"><a href="#cb30-373" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-376"><a href="#cb30-376" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-377"><a href="#cb30-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-378"><a href="#cb30-378" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-379"><a href="#cb30-379" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-380"><a href="#cb30-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-381"><a href="#cb30-381" aria-hidden="true" tabindex="-1"></a><span class="fu">## GenomeInfo  </span></span>
<span id="cb30-382"><a href="#cb30-382" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/GenomeInfo/&gt;</span>  </span>
<span id="cb30-383"><a href="#cb30-383" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseGenomeInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-384"><a href="#cb30-384" aria-hidden="true" tabindex="-1"></a>  Show GenomeInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-385"><a href="#cb30-385" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-386"><a href="#cb30-386" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseGenomeInfo"&gt;</span>
<span id="cb30-387"><a href="#cb30-387" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-388"><a href="#cb30-388" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-389"><a href="#cb30-389" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>name (***type=string***)  </span>
<span id="cb30-390"><a href="#cb30-390" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>name of the genome  </span>
<span id="cb30-391"><a href="#cb30-391" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>genome_version (***type=string***)  </span>
<span id="cb30-392"><a href="#cb30-392" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the genome version  </span>
<span id="cb30-393"><a href="#cb30-393" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>taxon_id (***type=integer***)  </span>
<span id="cb30-394"><a href="#cb30-394" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the NCBI taxonomy number  </span>
<span id="cb30-395"><a href="#cb30-395" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>url (***type=string***)  </span>
<span id="cb30-396"><a href="#cb30-396" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a link to the where this genome file could be downloaded  </span>
<span id="cb30-397"><a href="#cb30-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-398"><a href="#cb30-398" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-399"><a href="#cb30-399" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>chromosomes (***type=array***)  </span>
<span id="cb30-400"><a href="#cb30-400" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of chromosomes found within this genome  </span>
<span id="cb30-401"><a href="#cb30-401" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>gff_url (***type=string***)  </span>
<span id="cb30-402"><a href="#cb30-402" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a link to the where this genome's annotation file could be downloaded  </span>
<span id="cb30-403"><a href="#cb30-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-404"><a href="#cb30-404" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-407"><a href="#cb30-407" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-408"><a href="#cb30-408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-409"><a href="#cb30-409" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-410"><a href="#cb30-410" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-411"><a href="#cb30-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-412"><a href="#cb30-412" aria-hidden="true" tabindex="-1"></a><span class="fu">## GenomicLocation  </span></span>
<span id="cb30-413"><a href="#cb30-413" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/GenomicLocation/&gt;</span>  </span>
<span id="cb30-414"><a href="#cb30-414" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseGenomicLocation" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-415"><a href="#cb30-415" aria-hidden="true" tabindex="-1"></a>  Show GenomicLocation fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-416"><a href="#cb30-416" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-417"><a href="#cb30-417" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseGenomicLocation"&gt;</span>
<span id="cb30-418"><a href="#cb30-418" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-419"><a href="#cb30-419" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-420"><a href="#cb30-420" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>genome_id (***type=integer***)  </span>
<span id="cb30-421"><a href="#cb30-421" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index to the genome in the targeted_genomes list that this location refers to   </span>
<span id="cb30-422"><a href="#cb30-422" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>chrom (***type=string***)  </span>
<span id="cb30-423"><a href="#cb30-423" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the chromosome name  </span>
<span id="cb30-424"><a href="#cb30-424" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>start (***type=integer***)  </span>
<span id="cb30-425"><a href="#cb30-425" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the start of the location, 0-based positioning  </span>
<span id="cb30-426"><a href="#cb30-426" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>end (***type=integer***)  </span>
<span id="cb30-427"><a href="#cb30-427" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the end of the location, 0-based positioning  </span>
<span id="cb30-428"><a href="#cb30-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-429"><a href="#cb30-429" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-430"><a href="#cb30-430" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>ref_seq (***type=string***)  </span>
<span id="cb30-431"><a href="#cb30-431" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the reference sequence of this genomic location  </span>
<span id="cb30-432"><a href="#cb30-432" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>strand (***type=string***)  </span>
<span id="cb30-433"><a href="#cb30-433" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>which strand the location is, either + for plus strand or - for negative strand  </span>
<span id="cb30-434"><a href="#cb30-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-435"><a href="#cb30-435" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-438"><a href="#cb30-438" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-439"><a href="#cb30-439" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-440"><a href="#cb30-440" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-441"><a href="#cb30-441" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-442"><a href="#cb30-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-443"><a href="#cb30-443" aria-hidden="true" tabindex="-1"></a><span class="fu">## MarkerOfInterest  </span></span>
<span id="cb30-444"><a href="#cb30-444" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MarkerOfInterest/&gt;</span>  </span>
<span id="cb30-445"><a href="#cb30-445" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMarkerOfInterest" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-446"><a href="#cb30-446" aria-hidden="true" tabindex="-1"></a>  Show MarkerOfInterest fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-447"><a href="#cb30-447" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-448"><a href="#cb30-448" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMarkerOfInterest"&gt;</span>
<span id="cb30-449"><a href="#cb30-449" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-450"><a href="#cb30-450" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-451"><a href="#cb30-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-452"><a href="#cb30-452" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-453"><a href="#cb30-453" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>associations (***type=array***)  </span>
<span id="cb30-454"><a href="#cb30-454" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of associations with this marker, e.g. SP resistance, etc  </span>
<span id="cb30-455"><a href="#cb30-455" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>marker_location (***type=[GenomicLocation](#genomiclocation)***)  </span>
<span id="cb30-456"><a href="#cb30-456" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the genomic location  </span>
<span id="cb30-457"><a href="#cb30-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-458"><a href="#cb30-458" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-461"><a href="#cb30-461" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-462"><a href="#cb30-462" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-463"><a href="#cb30-463" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-464"><a href="#cb30-464" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-465"><a href="#cb30-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-466"><a href="#cb30-466" aria-hidden="true" tabindex="-1"></a><span class="fu">## MaskingInfo  </span></span>
<span id="cb30-467"><a href="#cb30-467" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MaskingInfo/&gt;</span>  </span>
<span id="cb30-468"><a href="#cb30-468" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMaskingInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-469"><a href="#cb30-469" aria-hidden="true" tabindex="-1"></a>  Show MaskingInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-470"><a href="#cb30-470" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-471"><a href="#cb30-471" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMaskingInfo"&gt;</span>
<span id="cb30-472"><a href="#cb30-472" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-473"><a href="#cb30-473" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-474"><a href="#cb30-474" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq_start (***type=integer***)  </span>
<span id="cb30-475"><a href="#cb30-475" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the start of the masking  </span>
<span id="cb30-476"><a href="#cb30-476" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq_segment_size (***type=integer***)  </span>
<span id="cb30-477"><a href="#cb30-477" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the size of the masking  </span>
<span id="cb30-478"><a href="#cb30-478" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>replacement_size (***type=integer***)  </span>
<span id="cb30-479"><a href="#cb30-479" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the size of replacement mask  </span>
<span id="cb30-480"><a href="#cb30-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-481"><a href="#cb30-481" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-484"><a href="#cb30-484" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-485"><a href="#cb30-485" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-486"><a href="#cb30-486" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-487"><a href="#cb30-487" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-488"><a href="#cb30-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-489"><a href="#cb30-489" aria-hidden="true" tabindex="-1"></a><span class="fu">## MicrohaplotypeForTarget  </span></span>
<span id="cb30-490"><a href="#cb30-490" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypeForTarget/&gt;</span>  </span>
<span id="cb30-491"><a href="#cb30-491" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypeForTarget" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-492"><a href="#cb30-492" aria-hidden="true" tabindex="-1"></a>  Show MicrohaplotypeForTarget fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-493"><a href="#cb30-493" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-494"><a href="#cb30-494" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMicrohaplotypeForTarget"&gt;</span>
<span id="cb30-495"><a href="#cb30-495" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-496"><a href="#cb30-496" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-497"><a href="#cb30-497" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>mhap_id (***type=integer***)  </span>
<span id="cb30-498"><a href="#cb30-498" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index for a microhaplotype for a target in the microhaplotypes_info list, e.g. microhaplotypes_info<span class="co">[</span><span class="ot">mhaps_target_id</span><span class="co">][mhap_id]</span>  </span>
<span id="cb30-499"><a href="#cb30-499" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>reads (***type=integer***)  </span>
<span id="cb30-500"><a href="#cb30-500" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the read count associated with this microhaplotype  </span>
<span id="cb30-501"><a href="#cb30-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-502"><a href="#cb30-502" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-503"><a href="#cb30-503" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>umis (***type=integer***)  </span>
<span id="cb30-504"><a href="#cb30-504" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the unique molecular identifier (umi) count associated with this microhaplotype  </span>
<span id="cb30-505"><a href="#cb30-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-506"><a href="#cb30-506" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-509"><a href="#cb30-509" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-510"><a href="#cb30-510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-511"><a href="#cb30-511" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-512"><a href="#cb30-512" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-513"><a href="#cb30-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-514"><a href="#cb30-514" aria-hidden="true" tabindex="-1"></a><span class="fu">## MicrohaplotypesDetected  </span></span>
<span id="cb30-515"><a href="#cb30-515" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesDetected/&gt;</span>  </span>
<span id="cb30-516"><a href="#cb30-516" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesDetected" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-517"><a href="#cb30-517" aria-hidden="true" tabindex="-1"></a>  Show MicrohaplotypesDetected fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-518"><a href="#cb30-518" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-519"><a href="#cb30-519" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMicrohaplotypesDetected"&gt;</span>
<span id="cb30-520"><a href="#cb30-520" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-521"><a href="#cb30-521" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-522"><a href="#cb30-522" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_run_id (***type=integer***)  </span>
<span id="cb30-523"><a href="#cb30-523" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into bioinformatics_run_info list  </span>
<span id="cb30-524"><a href="#cb30-524" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>experiment_samples (***type=[MicrohaplotypesForSample](#microhaplotypesforsample)***)  </span>
<span id="cb30-525"><a href="#cb30-525" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the microhaplotypes detected for a sample by targets   </span>
<span id="cb30-526"><a href="#cb30-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-527"><a href="#cb30-527" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-530"><a href="#cb30-530" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-531"><a href="#cb30-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-532"><a href="#cb30-532" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-533"><a href="#cb30-533" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-534"><a href="#cb30-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-535"><a href="#cb30-535" aria-hidden="true" tabindex="-1"></a><span class="fu">## MicrohaplotypesForSample  </span></span>
<span id="cb30-536"><a href="#cb30-536" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForSample/&gt;</span>  </span>
<span id="cb30-537"><a href="#cb30-537" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesForSample" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-538"><a href="#cb30-538" aria-hidden="true" tabindex="-1"></a>  Show MicrohaplotypesForSample fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-539"><a href="#cb30-539" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-540"><a href="#cb30-540" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMicrohaplotypesForSample"&gt;</span>
<span id="cb30-541"><a href="#cb30-541" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-542"><a href="#cb30-542" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-543"><a href="#cb30-543" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>experiment_sample_id (***type=integer***)  </span>
<span id="cb30-544"><a href="#cb30-544" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the experiment_info list  </span>
<span id="cb30-545"><a href="#cb30-545" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_results (***type=[MicrohaplotypesForTarget](#microhaplotypesfortarget)***)  </span>
<span id="cb30-546"><a href="#cb30-546" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the microhaplotypes detected for a list of targets  </span>
<span id="cb30-547"><a href="#cb30-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-548"><a href="#cb30-548" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-551"><a href="#cb30-551" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-552"><a href="#cb30-552" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-553"><a href="#cb30-553" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-554"><a href="#cb30-554" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-555"><a href="#cb30-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-556"><a href="#cb30-556" aria-hidden="true" tabindex="-1"></a><span class="fu">## MicrohaplotypesForTarget  </span></span>
<span id="cb30-557"><a href="#cb30-557" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/MicrohaplotypesForTarget/&gt;</span>  </span>
<span id="cb30-558"><a href="#cb30-558" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseMicrohaplotypesForTarget" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-559"><a href="#cb30-559" aria-hidden="true" tabindex="-1"></a>  Show MicrohaplotypesForTarget fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-560"><a href="#cb30-560" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-561"><a href="#cb30-561" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseMicrohaplotypesForTarget"&gt;</span>
<span id="cb30-562"><a href="#cb30-562" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-563"><a href="#cb30-563" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-564"><a href="#cb30-564" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>mhaps_target_id (***type=integer***)  </span>
<span id="cb30-565"><a href="#cb30-565" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index for a target in the microhaplotypes_info list  </span>
<span id="cb30-566"><a href="#cb30-566" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>haps (***type=[MicrohaplotypeForTarget](#microhaplotypefortarget)***)  </span>
<span id="cb30-567"><a href="#cb30-567" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the microhaplotypes detected for this target  </span>
<span id="cb30-568"><a href="#cb30-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-569"><a href="#cb30-569" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-572"><a href="#cb30-572" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-573"><a href="#cb30-573" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-574"><a href="#cb30-574" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-575"><a href="#cb30-575" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-576"><a href="#cb30-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-577"><a href="#cb30-577" aria-hidden="true" tabindex="-1"></a><span class="fu">## PanelInfo  </span></span>
<span id="cb30-578"><a href="#cb30-578" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/PanelInfo/&gt;</span>  </span>
<span id="cb30-579"><a href="#cb30-579" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePanelInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-580"><a href="#cb30-580" aria-hidden="true" tabindex="-1"></a>  Show PanelInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-581"><a href="#cb30-581" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-582"><a href="#cb30-582" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapsePanelInfo"&gt;</span>
<span id="cb30-583"><a href="#cb30-583" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-584"><a href="#cb30-584" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-585"><a href="#cb30-585" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>reactions (***type=[ReactionInfo](#reactioninfo)***)  </span>
<span id="cb30-586"><a href="#cb30-586" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of 1 or more reactions that this panel contains, each reactions list the targets that were amplified in that reaction, e.g. pool1, pool2  </span>
<span id="cb30-587"><a href="#cb30-587" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>panel_name (***type=string***)  </span>
<span id="cb30-588"><a href="#cb30-588" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name for the panel  </span>
<span id="cb30-589"><a href="#cb30-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-590"><a href="#cb30-590" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-593"><a href="#cb30-593" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-594"><a href="#cb30-594" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-595"><a href="#cb30-595" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-596"><a href="#cb30-596" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-597"><a href="#cb30-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-598"><a href="#cb30-598" aria-hidden="true" tabindex="-1"></a><span class="fu">## ParasiteDensity  </span></span>
<span id="cb30-599"><a href="#cb30-599" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ParasiteDensity/&gt;</span>  </span>
<span id="cb30-600"><a href="#cb30-600" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseParasiteDensity" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-601"><a href="#cb30-601" aria-hidden="true" tabindex="-1"></a>  Show ParasiteDensity fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-602"><a href="#cb30-602" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-603"><a href="#cb30-603" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseParasiteDensity"&gt;</span>
<span id="cb30-604"><a href="#cb30-604" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-605"><a href="#cb30-605" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-606"><a href="#cb30-606" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>method (***type=string***)  </span>
<span id="cb30-607"><a href="#cb30-607" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the method of how this density was obtained  </span>
<span id="cb30-608"><a href="#cb30-608" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>density (***type=number***)  </span>
<span id="cb30-609"><a href="#cb30-609" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the density in microliters  </span>
<span id="cb30-610"><a href="#cb30-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-611"><a href="#cb30-611" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-614"><a href="#cb30-614" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-615"><a href="#cb30-615" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-616"><a href="#cb30-616" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-617"><a href="#cb30-617" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-618"><a href="#cb30-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-619"><a href="#cb30-619" aria-hidden="true" tabindex="-1"></a><span class="fu">## PmoGenerationMethod  </span></span>
<span id="cb30-620"><a href="#cb30-620" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/PmoGenerationMethod/&gt;</span>  </span>
<span id="cb30-621"><a href="#cb30-621" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePmoGenerationMethod" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-622"><a href="#cb30-622" aria-hidden="true" tabindex="-1"></a>  Show PmoGenerationMethod fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-623"><a href="#cb30-623" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-624"><a href="#cb30-624" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapsePmoGenerationMethod"&gt;</span>
<span id="cb30-625"><a href="#cb30-625" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-626"><a href="#cb30-626" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-627"><a href="#cb30-627" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>program_version (***type=string***)  </span>
<span id="cb30-628"><a href="#cb30-628" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the version of generation method, should be in the format of v<span class="co">[</span><span class="ot">MAJOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">MINOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">PATCH</span><span class="co">]</span>  </span>
<span id="cb30-629"><a href="#cb30-629" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>program_name (***type=string***)  </span>
<span id="cb30-630"><a href="#cb30-630" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the name of the program  </span>
<span id="cb30-631"><a href="#cb30-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-632"><a href="#cb30-632" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-635"><a href="#cb30-635" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-636"><a href="#cb30-636" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-637"><a href="#cb30-637" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-638"><a href="#cb30-638" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-639"><a href="#cb30-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-640"><a href="#cb30-640" aria-hidden="true" tabindex="-1"></a><span class="fu">## PmoHeader  </span></span>
<span id="cb30-641"><a href="#cb30-641" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/PmoHeader/&gt;</span>  </span>
<span id="cb30-642"><a href="#cb30-642" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePmoHeader" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-643"><a href="#cb30-643" aria-hidden="true" tabindex="-1"></a>  Show PmoHeader fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-644"><a href="#cb30-644" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-645"><a href="#cb30-645" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapsePmoHeader"&gt;</span>
<span id="cb30-646"><a href="#cb30-646" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-647"><a href="#cb30-647" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-648"><a href="#cb30-648" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>pmo_version (***type=string***)  </span>
<span id="cb30-649"><a href="#cb30-649" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the version of the PMO file, should be in the format of v<span class="co">[</span><span class="ot">MAJOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">MINOR</span><span class="co">]</span>.<span class="co">[</span><span class="ot">PATCH</span><span class="co">]</span>  </span>
<span id="cb30-650"><a href="#cb30-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-651"><a href="#cb30-651" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-652"><a href="#cb30-652" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>creation_date (***type=string***)  </span>
<span id="cb30-653"><a href="#cb30-653" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date of when the PMO file was created or modified, should be YYYY-MM-DD  </span>
<span id="cb30-654"><a href="#cb30-654" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>generation_method (***type=[PmoGenerationMethod](#pmogenerationmethod)***)  </span>
<span id="cb30-655"><a href="#cb30-655" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the generation method to create this PMO   </span>
<span id="cb30-656"><a href="#cb30-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-657"><a href="#cb30-657" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-660"><a href="#cb30-660" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-661"><a href="#cb30-661" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-662"><a href="#cb30-662" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-663"><a href="#cb30-663" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-664"><a href="#cb30-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-665"><a href="#cb30-665" aria-hidden="true" tabindex="-1"></a><span class="fu">## PrimerInfo  </span></span>
<span id="cb30-666"><a href="#cb30-666" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/PrimerInfo/&gt;</span>  </span>
<span id="cb30-667"><a href="#cb30-667" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapsePrimerInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-668"><a href="#cb30-668" aria-hidden="true" tabindex="-1"></a>  Show PrimerInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-669"><a href="#cb30-669" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-670"><a href="#cb30-670" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapsePrimerInfo"&gt;</span>
<span id="cb30-671"><a href="#cb30-671" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-672"><a href="#cb30-672" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-673"><a href="#cb30-673" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq (***type=string***)  </span>
<span id="cb30-674"><a href="#cb30-674" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the DNA sequence  </span>
<span id="cb30-675"><a href="#cb30-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-676"><a href="#cb30-676" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-677"><a href="#cb30-677" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>location (***type=[GenomicLocation](#genomiclocation)***)  </span>
<span id="cb30-678"><a href="#cb30-678" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>what the intended genomic location of the primer is  </span>
<span id="cb30-679"><a href="#cb30-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-680"><a href="#cb30-680" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-683"><a href="#cb30-683" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-684"><a href="#cb30-684" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-685"><a href="#cb30-685" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-686"><a href="#cb30-686" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-687"><a href="#cb30-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-688"><a href="#cb30-688" aria-hidden="true" tabindex="-1"></a><span class="fu">## ReactionInfo  </span></span>
<span id="cb30-689"><a href="#cb30-689" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ReactionInfo/&gt;</span>  </span>
<span id="cb30-690"><a href="#cb30-690" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReactionInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-691"><a href="#cb30-691" aria-hidden="true" tabindex="-1"></a>  Show ReactionInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-692"><a href="#cb30-692" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-693"><a href="#cb30-693" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseReactionInfo"&gt;</span>
<span id="cb30-694"><a href="#cb30-694" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-695"><a href="#cb30-695" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-696"><a href="#cb30-696" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>panel_targets (***type=array***)  </span>
<span id="cb30-697"><a href="#cb30-697" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the target indexes in the target_info list  </span>
<span id="cb30-698"><a href="#cb30-698" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>reaction_name (***type=string***)  </span>
<span id="cb30-699"><a href="#cb30-699" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name for this reaction  </span>
<span id="cb30-700"><a href="#cb30-700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-701"><a href="#cb30-701" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-704"><a href="#cb30-704" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-705"><a href="#cb30-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-706"><a href="#cb30-706" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-707"><a href="#cb30-707" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-708"><a href="#cb30-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-709"><a href="#cb30-709" aria-hidden="true" tabindex="-1"></a><span class="fu">## ReadCountsByStage  </span></span>
<span id="cb30-710"><a href="#cb30-710" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStage/&gt;</span>  </span>
<span id="cb30-711"><a href="#cb30-711" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStage" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-712"><a href="#cb30-712" aria-hidden="true" tabindex="-1"></a>  Show ReadCountsByStage fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-713"><a href="#cb30-713" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-714"><a href="#cb30-714" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseReadCountsByStage"&gt;</span>
<span id="cb30-715"><a href="#cb30-715" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-716"><a href="#cb30-716" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-717"><a href="#cb30-717" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>bioinformatics_run_id (***type=integer***)  </span>
<span id="cb30-718"><a href="#cb30-718" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into bioinformatics_run_info list  </span>
<span id="cb30-719"><a href="#cb30-719" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>read_counts_by_experimental_sample_by_stage (***type=[ReadCountsByStageForExperimentalSample](#readcountsbystageforexperimentalsample)***)  </span>
<span id="cb30-720"><a href="#cb30-720" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list by experiment_sample for the counts at each stage  </span>
<span id="cb30-721"><a href="#cb30-721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-722"><a href="#cb30-722" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-725"><a href="#cb30-725" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-726"><a href="#cb30-726" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-727"><a href="#cb30-727" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-728"><a href="#cb30-728" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-729"><a href="#cb30-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-730"><a href="#cb30-730" aria-hidden="true" tabindex="-1"></a><span class="fu">## ReadCountsByStageForExperimentalSample  </span></span>
<span id="cb30-731"><a href="#cb30-731" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForExperimentalSample/&gt;</span>  </span>
<span id="cb30-732"><a href="#cb30-732" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStageForExperimentalSample" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-733"><a href="#cb30-733" aria-hidden="true" tabindex="-1"></a>  Show ReadCountsByStageForExperimentalSample fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-734"><a href="#cb30-734" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-735"><a href="#cb30-735" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseReadCountsByStageForExperimentalSample"&gt;</span>
<span id="cb30-736"><a href="#cb30-736" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-737"><a href="#cb30-737" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-738"><a href="#cb30-738" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>experiment_sample_id (***type=integer***)  </span>
<span id="cb30-739"><a href="#cb30-739" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the experiment_info list  </span>
<span id="cb30-740"><a href="#cb30-740" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>total_raw_count (***type=integer***)  </span>
<span id="cb30-741"><a href="#cb30-741" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the raw counts off the sequencing machine that a sample began with  </span>
<span id="cb30-742"><a href="#cb30-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-743"><a href="#cb30-743" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-744"><a href="#cb30-744" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>read_counts_for_targets (***type=[ReadCountsByStageForTarget](#readcountsbystagefortarget)***)  </span>
<span id="cb30-745"><a href="#cb30-745" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of counts by stage for a target  </span>
<span id="cb30-746"><a href="#cb30-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-747"><a href="#cb30-747" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-750"><a href="#cb30-750" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-751"><a href="#cb30-751" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-752"><a href="#cb30-752" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-753"><a href="#cb30-753" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-754"><a href="#cb30-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-755"><a href="#cb30-755" aria-hidden="true" tabindex="-1"></a><span class="fu">## ReadCountsByStageForTarget  </span></span>
<span id="cb30-756"><a href="#cb30-756" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/ReadCountsByStageForTarget/&gt;</span>  </span>
<span id="cb30-757"><a href="#cb30-757" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseReadCountsByStageForTarget" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-758"><a href="#cb30-758" aria-hidden="true" tabindex="-1"></a>  Show ReadCountsByStageForTarget fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-759"><a href="#cb30-759" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-760"><a href="#cb30-760" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseReadCountsByStageForTarget"&gt;</span>
<span id="cb30-761"><a href="#cb30-761" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-762"><a href="#cb30-762" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-763"><a href="#cb30-763" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_id (***type=integer***)  </span>
<span id="cb30-764"><a href="#cb30-764" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the target_info list  </span>
<span id="cb30-765"><a href="#cb30-765" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>stages (***type=[StageReadCounts](#stagereadcounts)***)  </span>
<span id="cb30-766"><a href="#cb30-766" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the read counts by each stage  </span>
<span id="cb30-767"><a href="#cb30-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-768"><a href="#cb30-768" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-771"><a href="#cb30-771" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-772"><a href="#cb30-772" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-773"><a href="#cb30-773" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-774"><a href="#cb30-774" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-775"><a href="#cb30-775" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-776"><a href="#cb30-776" aria-hidden="true" tabindex="-1"></a><span class="fu">## RepresentativeMicrohaplotype  </span></span>
<span id="cb30-777"><a href="#cb30-777" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotype/&gt;</span>  </span>
<span id="cb30-778"><a href="#cb30-778" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotype" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-779"><a href="#cb30-779" aria-hidden="true" tabindex="-1"></a>  Show RepresentativeMicrohaplotype fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-780"><a href="#cb30-780" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-781"><a href="#cb30-781" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseRepresentativeMicrohaplotype"&gt;</span>
<span id="cb30-782"><a href="#cb30-782" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-783"><a href="#cb30-783" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-784"><a href="#cb30-784" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq (***type=string***)  </span>
<span id="cb30-785"><a href="#cb30-785" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the DNA sequence  </span>
<span id="cb30-786"><a href="#cb30-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-787"><a href="#cb30-787" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-788"><a href="#cb30-788" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>alt_annotations (***type=array***)  </span>
<span id="cb30-789"><a href="#cb30-789" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of additional annotations associated with this microhaplotype, e.g. wildtype, amino acid changes etc  </span>
<span id="cb30-790"><a href="#cb30-790" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>masking (***type=[MaskingInfo](#maskinginfo)***)  </span>
<span id="cb30-791"><a href="#cb30-791" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>masking info for the sequence  </span>
<span id="cb30-792"><a href="#cb30-792" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>microhaplotype_name (***type=string***)  </span>
<span id="cb30-793"><a href="#cb30-793" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>an optional name for this microhaplotype  </span>
<span id="cb30-794"><a href="#cb30-794" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>pseudocigar (***type=string***)  </span>
<span id="cb30-795"><a href="#cb30-795" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the pseudocigar of the haplotype  </span>
<span id="cb30-796"><a href="#cb30-796" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>quality (***type=string***)  </span>
<span id="cb30-797"><a href="#cb30-797" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the ansi fastq per base quality score for this sequence, this is optional  </span>
<span id="cb30-798"><a href="#cb30-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-799"><a href="#cb30-799" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-802"><a href="#cb30-802" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-803"><a href="#cb30-803" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-804"><a href="#cb30-804" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-805"><a href="#cb30-805" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-806"><a href="#cb30-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-807"><a href="#cb30-807" aria-hidden="true" tabindex="-1"></a><span class="fu">## RepresentativeMicrohaplotypes  </span></span>
<span id="cb30-808"><a href="#cb30-808" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypes/&gt;</span>  </span>
<span id="cb30-809"><a href="#cb30-809" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotypes" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-810"><a href="#cb30-810" aria-hidden="true" tabindex="-1"></a>  Show RepresentativeMicrohaplotypes fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-811"><a href="#cb30-811" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-812"><a href="#cb30-812" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseRepresentativeMicrohaplotypes"&gt;</span>
<span id="cb30-813"><a href="#cb30-813" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-814"><a href="#cb30-814" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-815"><a href="#cb30-815" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>targets (***type=[RepresentativeMicrohaplotypesForTarget](#representativemicrohaplotypesfortarget)***)  </span>
<span id="cb30-816"><a href="#cb30-816" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the microhaplotype for each targets   </span>
<span id="cb30-817"><a href="#cb30-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-818"><a href="#cb30-818" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-821"><a href="#cb30-821" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-822"><a href="#cb30-822" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-823"><a href="#cb30-823" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-824"><a href="#cb30-824" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-825"><a href="#cb30-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-826"><a href="#cb30-826" aria-hidden="true" tabindex="-1"></a><span class="fu">## RepresentativeMicrohaplotypesForTarget  </span></span>
<span id="cb30-827"><a href="#cb30-827" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/RepresentativeMicrohaplotypesForTarget/&gt;</span>  </span>
<span id="cb30-828"><a href="#cb30-828" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseRepresentativeMicrohaplotypesForTarget" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-829"><a href="#cb30-829" aria-hidden="true" tabindex="-1"></a>  Show RepresentativeMicrohaplotypesForTarget fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-830"><a href="#cb30-830" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-831"><a href="#cb30-831" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseRepresentativeMicrohaplotypesForTarget"&gt;</span>
<span id="cb30-832"><a href="#cb30-832" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-833"><a href="#cb30-833" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-834"><a href="#cb30-834" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_id (***type=integer***)  </span>
<span id="cb30-835"><a href="#cb30-835" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the index into the target_info list  </span>
<span id="cb30-836"><a href="#cb30-836" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>microhaplotypes (***type=[RepresentativeMicrohaplotype](#representativemicrohaplotype)***)  </span>
<span id="cb30-837"><a href="#cb30-837" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of the microhaplotypes detected for a target   </span>
<span id="cb30-838"><a href="#cb30-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-839"><a href="#cb30-839" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-842"><a href="#cb30-842" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-843"><a href="#cb30-843" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-844"><a href="#cb30-844" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-845"><a href="#cb30-845" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-846"><a href="#cb30-846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-847"><a href="#cb30-847" aria-hidden="true" tabindex="-1"></a><span class="fu">## SequencingInfo  </span></span>
<span id="cb30-848"><a href="#cb30-848" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/SequencingInfo/&gt;</span>  </span>
<span id="cb30-849"><a href="#cb30-849" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseSequencingInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-850"><a href="#cb30-850" aria-hidden="true" tabindex="-1"></a>  Show SequencingInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-851"><a href="#cb30-851" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-852"><a href="#cb30-852" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseSequencingInfo"&gt;</span>
<span id="cb30-853"><a href="#cb30-853" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-854"><a href="#cb30-854" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-855"><a href="#cb30-855" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>sequencing_info_name (***type=string***)  </span>
<span id="cb30-856"><a href="#cb30-856" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name of for the sequencing done, e.g. batch1  </span>
<span id="cb30-857"><a href="#cb30-857" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq_instrument (***type=string***)  </span>
<span id="cb30-858"><a href="#cb30-858" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the sequencing instrument used to sequence the run, e.g. ILLUMINA, Illumina MiSeq  </span>
<span id="cb30-859"><a href="#cb30-859" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq_date (***type=string***)  </span>
<span id="cb30-860"><a href="#cb30-860" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date of sequencing, should be YYYY-MM or YYYY-MM-DD  </span>
<span id="cb30-861"><a href="#cb30-861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-862"><a href="#cb30-862" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-863"><a href="#cb30-863" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>lib_kit (***type=string***)  </span>
<span id="cb30-864"><a href="#cb30-864" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Name, version, and applicable cell or cycle numbers for the kit used to prepare libraries and load cells or chips for sequencing. If possible, include a part number, e.g. MiSeq Reagent Kit v3 (150-cycle), MS-102-3001  </span>
<span id="cb30-865"><a href="#cb30-865" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>lib_layout (***type=string***)  </span>
<span id="cb30-866"><a href="#cb30-866" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Specify the configuration of reads, e.g. paired-end  </span>
<span id="cb30-867"><a href="#cb30-867" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>lib_screen (***type=string***)  </span>
<span id="cb30-868"><a href="#cb30-868" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Describe enrichment, screening, or normalization methods applied during amplification or library preparation, e.g. size selection 390bp, diluted to 1 ng DNA/sample  </span>
<span id="cb30-869"><a href="#cb30-869" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>nucl_acid_amp (***type=string***)  </span>
<span id="cb30-870"><a href="#cb30-870" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Link to a reference or kit that describes the enzymatic amplification of nucleic acids  </span>
<span id="cb30-871"><a href="#cb30-871" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>nucl_acid_amp_date (***type=string***)  </span>
<span id="cb30-872"><a href="#cb30-872" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date of the nucleoacid amplification  </span>
<span id="cb30-873"><a href="#cb30-873" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>nucl_acid_ext (***type=string***)  </span>
<span id="cb30-874"><a href="#cb30-874" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Link to a reference or kit that describes the recovery of nucleic acids from the sample  </span>
<span id="cb30-875"><a href="#cb30-875" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>nucl_acid_ext_date (***type=string***)  </span>
<span id="cb30-876"><a href="#cb30-876" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date of the nucleoacid extraction  </span>
<span id="cb30-877"><a href="#cb30-877" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>pcr_cond (***type=string***)  </span>
<span id="cb30-878"><a href="#cb30-878" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the method/conditions for PCR, List PCR cycles used to amplify the target  </span>
<span id="cb30-879"><a href="#cb30-879" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>seq_center (***type=string***)  </span>
<span id="cb30-880"><a href="#cb30-880" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>Name of facility where sequencing was performed (lab, core facility, or company)  </span>
<span id="cb30-881"><a href="#cb30-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-882"><a href="#cb30-882" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-885"><a href="#cb30-885" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-886"><a href="#cb30-886" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-887"><a href="#cb30-887" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-888"><a href="#cb30-888" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-889"><a href="#cb30-889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-890"><a href="#cb30-890" aria-hidden="true" tabindex="-1"></a><span class="fu">## SpecimenInfo  </span></span>
<span id="cb30-891"><a href="#cb30-891" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/SpecimenInfo/&gt;</span>  </span>
<span id="cb30-892"><a href="#cb30-892" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseSpecimenInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-893"><a href="#cb30-893" aria-hidden="true" tabindex="-1"></a>  Show SpecimenInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-894"><a href="#cb30-894" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-895"><a href="#cb30-895" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseSpecimenInfo"&gt;</span>
<span id="cb30-896"><a href="#cb30-896" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-897"><a href="#cb30-897" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-898"><a href="#cb30-898" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>specimen_name (***type=string***)  </span>
<span id="cb30-899"><a href="#cb30-899" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>an identifier for the specimen  </span>
<span id="cb30-900"><a href="#cb30-900" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>samp_taxon_id (***type=integer***)  </span>
<span id="cb30-901"><a href="#cb30-901" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the NCBI taxonomy number of the organism of interest  </span>
<span id="cb30-902"><a href="#cb30-902" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>collection_date (***type=string***)  </span>
<span id="cb30-903"><a href="#cb30-903" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the date of the sample collection  </span>
<span id="cb30-904"><a href="#cb30-904" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>collection_country (***type=string***)  </span>
<span id="cb30-905"><a href="#cb30-905" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the name of country collected in, would be the same as admin level 0  </span>
<span id="cb30-906"><a href="#cb30-906" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>collector (***type=string***)  </span>
<span id="cb30-907"><a href="#cb30-907" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the name of the primary person managing the specimen  </span>
<span id="cb30-908"><a href="#cb30-908" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>samp_store_loc (***type=string***)  </span>
<span id="cb30-909"><a href="#cb30-909" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the sample store site, address or facility name  </span>
<span id="cb30-910"><a href="#cb30-910" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>samp_collect_device (***type=string***)  </span>
<span id="cb30-911"><a href="#cb30-911" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the way the sample was collected, e.g. whole blood, dried blood spot, etc  </span>
<span id="cb30-912"><a href="#cb30-912" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>project_name (***type=string***)  </span>
<span id="cb30-913"><a href="#cb30-913" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name of the project under which the sample is organized  </span>
<span id="cb30-914"><a href="#cb30-914" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-915"><a href="#cb30-915" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-916"><a href="#cb30-916" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>alternate_identifiers (***type=array***)  </span>
<span id="cb30-917"><a href="#cb30-917" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of optional alternative names for the samples  </span>
<span id="cb30-918"><a href="#cb30-918" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>date_of_birth (***type=string***)  </span>
<span id="cb30-919"><a href="#cb30-919" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>if specimen is from a person, the date of birth of that person  </span>
<span id="cb30-920"><a href="#cb30-920" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>geo_admin1 (***type=string***)  </span>
<span id="cb30-921"><a href="#cb30-921" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>geographical admin level 1, the secondary large demarcation of a nation (nation = admin level 0)  </span>
<span id="cb30-922"><a href="#cb30-922" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>geo_admin2 (***type=string***)  </span>
<span id="cb30-923"><a href="#cb30-923" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>geographical admin level 2, the third large demarcation of a nation (nation = admin level 0)  </span>
<span id="cb30-924"><a href="#cb30-924" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>geo_admin3 (***type=string***)  </span>
<span id="cb30-925"><a href="#cb30-925" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>geographical admin level 3, the third large demarcation of a nation (nation = admin level 0)  </span>
<span id="cb30-926"><a href="#cb30-926" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>host_taxon_id (***type=integer***)  </span>
<span id="cb30-927"><a href="#cb30-927" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>optional the NCBI taxonomy number of the host of the organism  </span>
<span id="cb30-928"><a href="#cb30-928" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>individual_id (***type=integer***)  </span>
<span id="cb30-929"><a href="#cb30-929" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>an identifier for the individual a specimen was collected from  </span>
<span id="cb30-930"><a href="#cb30-930" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>lat_lon (***type=string***)  </span>
<span id="cb30-931"><a href="#cb30-931" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the latitude and longitude of the collection site of the specimen  </span>
<span id="cb30-932"><a href="#cb30-932" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>parasite_density_info (***type=[ParasiteDensity](#parasitedensity)***)  </span>
<span id="cb30-933"><a href="#cb30-933" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>one or more parasite densities in microliters for this specimen  </span>
<span id="cb30-934"><a href="#cb30-934" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_col (***type=integer***)  </span>
<span id="cb30-935"><a href="#cb30-935" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the column the specimen was in  </span>
<span id="cb30-936"><a href="#cb30-936" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_name (***type=string***)  </span>
<span id="cb30-937"><a href="#cb30-937" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a name of plate the specimen was in  </span>
<span id="cb30-938"><a href="#cb30-938" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>plate_row (***type=string***)  </span>
<span id="cb30-939"><a href="#cb30-939" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the row the specimen was in  </span>
<span id="cb30-940"><a href="#cb30-940" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>sample_comments (***type=string***)  </span>
<span id="cb30-941"><a href="#cb30-941" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>any additional comments about the sample  </span>
<span id="cb30-942"><a href="#cb30-942" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>specimen_sex (***type=string***)  </span>
<span id="cb30-943"><a href="#cb30-943" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>if specimen is from a person, the sex listed for that person  </span>
<span id="cb30-944"><a href="#cb30-944" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-945"><a href="#cb30-945" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-948"><a href="#cb30-948" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-949"><a href="#cb30-949" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-950"><a href="#cb30-950" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-951"><a href="#cb30-951" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-952"><a href="#cb30-952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-953"><a href="#cb30-953" aria-hidden="true" tabindex="-1"></a><span class="fu">## StageReadCounts  </span></span>
<span id="cb30-954"><a href="#cb30-954" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/StageReadCounts/&gt;</span>  </span>
<span id="cb30-955"><a href="#cb30-955" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseStageReadCounts" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-956"><a href="#cb30-956" aria-hidden="true" tabindex="-1"></a>  Show StageReadCounts fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-957"><a href="#cb30-957" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-958"><a href="#cb30-958" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseStageReadCounts"&gt;</span>
<span id="cb30-959"><a href="#cb30-959" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-960"><a href="#cb30-960" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-961"><a href="#cb30-961" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>read_count (***type=integer***)  </span>
<span id="cb30-962"><a href="#cb30-962" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the read counts  </span>
<span id="cb30-963"><a href="#cb30-963" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>stage (***type=string***)  </span>
<span id="cb30-964"><a href="#cb30-964" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the stage of the pipeline, e.g. demultiplexed, denoised, etc  </span>
<span id="cb30-965"><a href="#cb30-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-966"><a href="#cb30-966" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-969"><a href="#cb30-969" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-970"><a href="#cb30-970" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-971"><a href="#cb30-971" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-972"><a href="#cb30-972" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-973"><a href="#cb30-973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-974"><a href="#cb30-974" aria-hidden="true" tabindex="-1"></a><span class="fu">## TargetInfo  </span></span>
<span id="cb30-975"><a href="#cb30-975" aria-hidden="true" tabindex="-1"></a><span class="ot">&lt;https://plasmogenepi.github.io/portable-microhaplotype-object/TargetInfo/&gt;</span>  </span>
<span id="cb30-976"><a href="#cb30-976" aria-hidden="true" tabindex="-1"></a>&lt;a class = "btn btn-light" data-bs-toggle="collapse" data-bs-target="#collapseTargetInfo" aria-expanded="false" aria-controls="collapseExample"&gt;</span>
<span id="cb30-977"><a href="#cb30-977" aria-hidden="true" tabindex="-1"></a>  Show TargetInfo fields &lt;i class="fa-solid fa-caret-right"&gt;&lt;/i&gt;</span>
<span id="cb30-978"><a href="#cb30-978" aria-hidden="true" tabindex="-1"></a>&lt;/a&gt;</span>
<span id="cb30-979"><a href="#cb30-979" aria-hidden="true" tabindex="-1"></a>&lt;div class="collapse" id="collapseTargetInfo"&gt;</span>
<span id="cb30-980"><a href="#cb30-980" aria-hidden="true" tabindex="-1"></a>  &lt;div class="card card-body"&gt;</span>
<span id="cb30-981"><a href="#cb30-981" aria-hidden="true" tabindex="-1"></a><span class="fu">### Required  </span></span>
<span id="cb30-982"><a href="#cb30-982" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_name (***type=string***)  </span>
<span id="cb30-983"><a href="#cb30-983" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>an identifier for this target  </span>
<span id="cb30-984"><a href="#cb30-984" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>forward_primers (***type=[PrimerInfo](#primerinfo)***)  </span>
<span id="cb30-985"><a href="#cb30-985" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>A list of forward primers associated with this target  </span>
<span id="cb30-986"><a href="#cb30-986" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>reverse_primers (***type=[PrimerInfo](#primerinfo)***)  </span>
<span id="cb30-987"><a href="#cb30-987" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>A list of reverse primers associated with this target  </span>
<span id="cb30-988"><a href="#cb30-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-989"><a href="#cb30-989" aria-hidden="true" tabindex="-1"></a><span class="fu">### Optional  </span></span>
<span id="cb30-990"><a href="#cb30-990" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>gene_name (***type=string***)  </span>
<span id="cb30-991"><a href="#cb30-991" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>an identifier of the gene, if any, is being covered with this targeted  </span>
<span id="cb30-992"><a href="#cb30-992" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>insert_location (***type=[GenomicLocation](#genomiclocation)***)  </span>
<span id="cb30-993"><a href="#cb30-993" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>the intended genomic location of the insert of the amplicon (the location between the end of the forward primer and the beginning of the reverse primer)  </span>
<span id="cb30-994"><a href="#cb30-994" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>target_attributes (***type=array***)  </span>
<span id="cb30-995"><a href="#cb30-995" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>a list of classification type for the primer target  </span>
<span id="cb30-996"><a href="#cb30-996" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-997"><a href="#cb30-997" aria-hidden="true" tabindex="-1"></a><span class="fu">### Example  </span></span>
<span id="cb30-1000"><a href="#cb30-1000" aria-hidden="true" tabindex="-1"></a><span class="in">```{json}</span></span>
<span id="cb30-1001"><a href="#cb30-1001" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb30-1002"><a href="#cb30-1002" aria-hidden="true" tabindex="-1"></a>  &lt;/div&gt;</span>
<span id="cb30-1003"><a href="#cb30-1003" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;br&gt;</span>
<span id="cb30-1004"><a href="#cb30-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-1005"><a href="#cb30-1005" aria-hidden="true" tabindex="-1"></a>{{&lt; fa dna &gt;}}</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>