<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Creating a PMO File â€“ Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script><script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-704a3d686fdb3e6b26ef47c77620618f.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-83b1d0b3b459d34b5b0f98adfe5eb835.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">    
        <li>
    <a class="dropdown-item" href="../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">    
        <li>
    <a class="dropdown-item" href="../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">    
        <li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-runner.py</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/pmotools-runner.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-runner.py</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Getting sub info from PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_panel_info_from_pmo.html">
 <span class="dropdown-text">Getting panel info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/build_pmo.html">
 <span class="dropdown-text">Creating a PMO File</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">    
        <li>
    <a class="dropdown-item" href="../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#creating-pmo" id="toc-creating-pmo" class="nav-link" data-scroll-target="#creating-pmo">Creating PMO</a>
  <ul>
  <li><a href="#panel-information" id="toc-panel-information" class="nav-link" data-scroll-target="#panel-information">Panel Information</a></li>
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata">Metadata</a>
  <ul>
  <li><a href="#specimen-level-metadata" id="toc-specimen-level-metadata" class="nav-link" data-scroll-target="#specimen-level-metadata">Specimen Level Metadata</a></li>
  <li><a href="#experiment-level-metadata" id="toc-experiment-level-metadata" class="nav-link" data-scroll-target="#experiment-level-metadata">Experiment Level Metadata</a></li>
  </ul></li>
  <li><a href="#microhaplotype-information" id="toc-microhaplotype-information" class="nav-link" data-scroll-target="#microhaplotype-information">Microhaplotype Information</a></li>
  <li><a href="#demultiplexed-experiment-samples" id="toc-demultiplexed-experiment-samples" class="nav-link" data-scroll-target="#demultiplexed-experiment-samples">Demultiplexed Experiment Samples</a></li>
  <li><a href="#sequencing-info" id="toc-sequencing-info" class="nav-link" data-scroll-target="#sequencing-info">Sequencing info</a></li>
  <li><a href="#bioinformatics-info" id="toc-bioinformatics-info" class="nav-link" data-scroll-target="#bioinformatics-info">Bioinformatics Info</a></li>
  </ul></li>
  <li><a href="#compose-pmo" id="toc-compose-pmo" class="nav-link" data-scroll-target="#compose-pmo">Compose PMO</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Creating a PMO File</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-page-left">

    
  
    
  </div>
  


</header>


<p>In this tutorial we will go through the steps to build a PMO, utilising the functions within the pmo-tools package.</p>
<p>For more information on any of the fields mentioned please see the <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html">documentation</a>.</p>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<p>First we will import the functions that we will need to run this notebook.</p>
<div id="73420f3c-3f26-43ff-ba13-16a2d6fc2e25" class="cell" data-execution_count="1">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.json_convertors.microhaplotype_table_to_pmo_dict <span class="im">import</span> microhaplotype_table_to_pmo_dict</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.json_convertors.metatable_to_json_meta <span class="im">import</span> experiment_info_table_to_json, specimen_info_table_to_json</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.json_convertors.panel_information_to_pmo_dict <span class="im">import</span> panel_info_table_to_pmo_dict</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.json_convertors.demultiplexed_targets_to_pmo_dict <span class="im">import</span> demultiplexed_targets_to_pmo_dict</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="563be5a0-241b-47d7-888b-8e71f2089c59" class="cell" data-execution_count="2">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> json </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here we define a function that will be used to print a few lines from the data we will be creating.</p>
<div id="6a3abe73-54a8-4032-b63f-1fb71f2cdeee" class="cell" data-execution_count="3">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> print_json_head(<span class="bu">dict</span>, n<span class="op">=</span><span class="dv">10</span>):</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>    json_object <span class="op">=</span> json.dumps(<span class="bu">dict</span>, indent<span class="op">=</span><span class="dv">4</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i, l <span class="kw">in</span> <span class="bu">enumerate</span>(json_object.split(<span class="st">'</span><span class="ch">\n</span><span class="st">'</span>)):</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i <span class="op">&gt;=</span> n:</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>            <span class="cf">break</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(l)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="creating-pmo" class="level2">
<h2 class="anchored" data-anchor-id="creating-pmo">Creating PMO</h2>
<p>To create the full PMO we will need a few sets of information. These include:</p>
<ul>
<li><strong>Panel Information</strong> : A table including data on the targets that make up the panel.</li>
<li><strong>Allele table</strong> : A table containing the alleles called for each of the samples for each of the targets and the reads associated.</li>
<li><strong>Demultiplexed reads</strong> : A table containing the raw reads for each sample, for each target after demultiplexing, before any filtering.</li>
<li><strong>Experimental metadata</strong> : Information on the sequencing run, for example where each sample was located on the plate.</li>
<li><strong>Specimen Information</strong> : metadata on the biological samples</li>
</ul>
<p>We will specify the paths to the example data we will use below, but if you would like to try and use your own data then replace the following paths:</p>
<div id="ff38e5db-369f-4ad2-940a-e8497e449dd4" class="cell" data-execution_count="4">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>panel_information_path <span class="op">=</span> <span class="st">'example_data/mad4hatter_panel_info_example.tsv'</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>allele_table_path <span class="op">=</span> <span class="st">'example_data/mad4hatter_allele_data_example.txt'</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>demultiplexed_reads_path <span class="op">=</span> <span class="st">'example_data/mad4hatter_amplicon_coverage.txt'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="panel-information" class="level3">
<h3 class="anchored" data-anchor-id="panel-information">Panel Information</h3>
<p>First we will work on putting the <strong>panel information</strong> into PMO format. Although labs may store this information in a variety of ways and this process may seem cumbersome, you will only have to do this once for each panel that you work with.</p>
<p>The panel information consists of 2 parts; The <strong>panel_targets</strong> (information on the targets) and the <strong>target_genome</strong> (information on the reference genome being targeted).</p>
<p>To include details of the reference genome we need the following information.</p>
<ul>
<li><strong>name</strong> : name of the genome</li>
<li><strong>version</strong> : the genome version</li>
<li><strong>taxon_id</strong> : the NCBI taxonomy number</li>
<li><strong>url</strong> : a link to the where this genome file could be downloaded</li>
</ul>
<p>Optionally, you can also include a link to genomes annotation file, as we have below. Below is an example of compiling this information into the json format manually:</p>
<div id="26f6e55e-32b8-47a9-9133-98ed63880a2c" class="cell" data-execution_count="5">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>target_genome_info <span class="op">=</span> {</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>            <span class="st">"gff_url"</span> : <span class="st">"https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/gff/data/PlasmoDB-65_Pfalciparum3D7.gff"</span>,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>            <span class="st">"name"</span> : <span class="st">"3D7"</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">"taxon_id"</span> : <span class="dv">5833</span>,</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">"url"</span> : <span class="st">"https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/fasta/data/PlasmoDB-65_Pfalciparum3D7_Genome.fasta"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>            <span class="st">"version"</span> : <span class="st">"2020-09-01"</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>        }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Fields that are required to define the <strong>target information</strong> are â€¦</p>
<ul>
<li><strong>target_id</strong> : a unique identifier for each of the targets</li>
<li><strong>forward primer seq</strong> : The sequence for the forward primer associated with this target</li>
<li><strong>reverse primer sequence</strong> : The sequence for the reverse primer associated with this target</li>
</ul>
<p><strong>Note: in the case that you have multiple primers to target the same region, please include these on separate lines in the table with the same target_id.</strong></p>
<p>Optionally you can also include location information for the primers. To include this information you will need to include in the table:</p>
<ul>
<li><strong>chrom</strong> : the chromosome name</li>
<li><strong>start</strong> : the start of the location, 0-based positioning</li>
<li><strong>end</strong> : the end of the location, 0-based positioning</li>
</ul>
<p>For more information on optional fields that can be included, check the <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#targetinfo">documentation</a>.</p>
<p>Here we show how to take panel information that is used to run the MAD4HatTeR pipeline and convert it to PMO.</p>
<div id="24877407-d3ad-4b4d-88ce-ddd6582699fa" class="cell" data-execution_count="6">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info <span class="op">=</span> pd.read_csv(panel_information_path, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info.head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="6">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">amplicon</th>
<th data-quarto-table-cell-role="th">amplicon_start</th>
<th data-quarto-table-cell-role="th">amplicon_end</th>
<th data-quarto-table-cell-role="th">ampInsert_start</th>
<th data-quarto-table-cell-role="th">ampInsert_end</th>
<th data-quarto-table-cell-role="th">rev_primer</th>
<th data-quarto-table-cell-role="th">amplicon_length</th>
<th data-quarto-table-cell-role="th">ampInsert_length</th>
<th data-quarto-table-cell-role="th">fwd_primer</th>
<th data-quarto-table-cell-role="th">target_type</th>
<th data-quarto-table-cell-role="th">strand</th>
<th data-quarto-table-cell-role="th">gene_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>145388</td>
<td>145662</td>
<td>145421</td>
<td>145630</td>
<td>AAAATGTCCAATATGTCAAGGTATATTAAAGT</td>
<td>274</td>
<td>209</td>
<td>CCTGAGTTTTAAGTGAATGAATATATTTTTGTT</td>
<td>diversity</td>
<td>+</td>
<td>PF3D7_0103300</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Pf3D7_01_v3-162867-163115-1A</td>
<td>162867</td>
<td>163115</td>
<td>162889</td>
<td>163092</td>
<td>TGTGTGCTTTGTCGTTGATTCAT</td>
<td>248</td>
<td>203</td>
<td>TACTACCGATCATCAAGCCGAA</td>
<td>diversity</td>
<td>+</td>
<td>PF3D7_0103600</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Pf3D7_01_v3-181512-181761-1A</td>
<td>181512</td>
<td>181761</td>
<td>181545</td>
<td>181729</td>
<td>TAGTTTAAATCTATACTTGTCTCACCTGAACA</td>
<td>249</td>
<td>184</td>
<td>CTTTTCATATTTGTCTATTAGCTTTTTCAAACC</td>
<td>diversity</td>
<td>+</td>
<td>PF3D7_0104100</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>Pf3D7_01_v3-455794-456054-1A</td>
<td>455794</td>
<td>456054</td>
<td>455827</td>
<td>456021</td>
<td>GTGTTTCATTATTTTAGACACATTCAGGAATTT</td>
<td>260</td>
<td>194</td>
<td>ACAATGTAGAACAATATATAAAACTGGAAAAGA</td>
<td>diversity</td>
<td>+</td>
<td>NaN</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>Pf3D7_01_v3-528859-529104-1A</td>
<td>528859</td>
<td>529104</td>
<td>528890</td>
<td>529073</td>
<td>AATCATTTTATCCCACTTATTTATCTCGTCT</td>
<td>245</td>
<td>183</td>
<td>CTTAGTTTAGATTTGCCTACAATATTTGCAC</td>
<td>diversity</td>
<td>+</td>
<td>PF3D7_0113800</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>We can use the <code>panel_info_table_to_pmo_dict</code> function to convert this into the correct format for PMO.</p>
<div id="56ec7047-74b5-4087-a9b9-88253384a9c6" class="cell" data-execution_count="7">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(panel_info_table_to_pmo_dict.__doc__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Convert a dataframe containing panel information into dictionary of targets and reference information


    :param target_table: The dataframe containing the target information
    :param panel_id: the panel ID assigned to the panel
    :param genome_info: A dictionary containing the genome information
    :param target_id_col: the name of the column containing the target IDs
    :param forward_primers_seq_col: the name of the column containing the sequence of the forward primer
    :param reverse_primers_seq_col: the name of the column containing the sequence of the reverse primer
    :param forward_primers_start_col (Optional): the name of the column containing the 0-based start coordinate of the forward primer
    :param forward_primers_end_col (Optional): the name of the column containing the 0-based end coordinate of the forward primer
    :param reverse_primers_start_col (Optional): the name of the column containing the 0-based start coordinate of the reverse primer
    :param reverse_primers_end_col (Optional): the name of the column containing the 0-based end coordinate of the reverse primer
    :param insert_start_col (Optional): the name of the column containing the 0-based start coordinate of the insert
    :param insert_end_col (Optional): the name of the column containing the 0-based end coordinate of the insert
    :param chrom_col (Optional): the name of the column containing the chromosome for the target
    :param gene_id_col (Optional): the name of the column containing the gene id
    :param strand_col (Optional): the name of the column containing the strand for the target
    :param target_type_col (Optional): A classification type for the target
    :param additional_target_info_cols (Optional): dictionary of optional additional columns to add to the target information dictionary. Keys are column names and values are the type.
    :return: a dict of the panel information
    </code></pre>
</div>
</div>
<p>We will use this first just to include the most basic required information.</p>
<div id="39469bf5-5239-4fa0-9003-38e1bb678d84" class="cell" data-scrolled="true" data-execution_count="8">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>panel_information_pmo <span class="op">=</span> panel_info_table_to_pmo_dict(</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    madhatter_panel_info,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"mad4hatter_poolsD1R1R2"</span>,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    target_genome_info,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    target_id_col<span class="op">=</span><span class="st">"amplicon"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    forward_primers_seq_col<span class="op">=</span><span class="st">"fwd_primer"</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    reverse_primers_seq_col<span class="op">=</span><span class="st">"rev_primer"</span>,</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Letâ€™s take a look at the first 30 rows of the information we put togetherâ€¦</p>
<div id="6a6559ef-e75a-48e0-bc67-cda4a20b4239" class="cell" data-execution_count="9">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>print_json_head(panel_information_pmo,<span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{
    "panel_info": {
        "mad4hatter_poolsD1R1R2": {
            "panel_id": "mad4hatter_poolsD1R1R2",
            "target_genome": {
                "gff_url": "https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/gff/data/PlasmoDB-65_Pfalciparum3D7.gff",
                "name": "3D7",
                "taxon_id": 5833,
                "url": "https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/fasta/data/PlasmoDB-65_Pfalciparum3D7_Genome.fasta",
                "version": "2020-09-01"
            },
            "targets": {
                "Pf3D7_01_v3-145388-145662-1A": {
                    "target_id": "Pf3D7_01_v3-145388-145662-1A",
                    "forward_primers": [
                        {
                            "seq": "CCTGAGTTTTAAGTGAATGAATATATTTTTGTT"
                        }
                    ],
                    "reverse_primers": [
                        {
                            "seq": "AAAATGTCCAATATGTCAAGGTATATTAAAGT"
                        }
                    ]
                },
                "Pf3D7_01_v3-162867-163115-1A": {
                    "target_id": "Pf3D7_01_v3-162867-163115-1A",
                    "forward_primers": [
                        {
                            "seq": "TACTACCGATCATCAAGCCGAA"</code></pre>
</div>
</div>
<p>Optionally we can include some more information. You can see that some of the fields in the table above donâ€™t directly match the optional fields. Therefore, we must first wrangle the data slightly to fit the requirements.</p>
<p><strong>Note: You may not have to apply all of the following steps to your panel information, this is just an example and is specific to the MAD4HatTeR panel information</strong>.</p>
<p>The chromosome for each target is stored within the locus name, so we extract that and put it in itâ€™s own column below.</p>
<div id="b9e814e5-bd8c-4e99-b09e-5e7873055445" class="cell" data-execution_count="10">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'chrom'</span>] <span class="op">=</span> [<span class="bu">chr</span>[<span class="dv">0</span>] <span class="cf">for</span> <span class="bu">chr</span> <span class="kw">in</span> madhatter_panel_info.amplicon.<span class="bu">str</span>.split(<span class="st">'-'</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Next we need to generate 0-based coordinates of the location the primers are targetting. The panel information we have only includes the full target start and end (including the primer) and is 1-based, so we do the conversion as follows.</p>
<div id="2474ca4e-a31e-4ed2-a7af-27a201ebfd2d" class="cell" data-execution_count="11">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create 0-based coordinate for start of forward primer</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'fwd_primer_start_0_based'</span>] <span class="op">=</span> madhatter_panel_info.amplicon_start<span class="op">-</span><span class="dv">1</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the length of the forward primer, add this to the primer start coordinate to get the end coordinate </span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'fwd_primer_len'</span>] <span class="op">=</span> [<span class="bu">len</span>(p) <span class="cf">for</span> p <span class="kw">in</span> madhatter_panel_info.fwd_primer]</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'fwd_primer_end_0_based'</span>] <span class="op">=</span> madhatter_panel_info.fwd_primer_start_0_based<span class="op">+</span>madhatter_panel_info.fwd_primer_len</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the length of the reverse primer. Subtract this from the end coordinate of the target to get the start coordinate of the reverse primer</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'rev_primer_len'</span>] <span class="op">=</span> [<span class="bu">len</span>(p) <span class="cf">for</span> p <span class="kw">in</span> madhatter_panel_info.rev_primer]</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'rev_primer_start_0_based'</span>] <span class="op">=</span> madhatter_panel_info.amplicon_end<span class="op">-</span>madhatter_panel_info.rev_primer_len</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co"># The 0-based reverse primer end would be the same as the amplicon end </span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'rev_primer_end_0_based'</span>] <span class="op">=</span> madhatter_panel_info.amplicon_end</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the MAD4HatTeR pipeline, we trim one base from each end of the amplicon insert because the base following a primer is often erroneous. We can create insert coordinates with this adjustment, as shown below. If you choose not to apply this trimming step, you can instead use the coordinate at the end of the forward primer and the beginning of the reverse primer to define the start and end of the insert.</p>
<div id="e2892899-4ab8-4907-b269-3e0c3f30d2ae" class="cell" data-execution_count="12">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'insert_start_0_based'</span>] <span class="op">=</span> madhatter_panel_info.fwd_primer_end_0_based<span class="op">+</span><span class="dv">1</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>madhatter_panel_info[<span class="st">'insert_end_0_based'</span>] <span class="op">=</span> madhatter_panel_info.rev_primer_start_0_based<span class="op">-</span><span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we can create panel information to go into PMO with all of the optional fields we just created.</p>
<div id="0af9db2f-35e0-4505-b3f1-c9b6074b85e3" class="cell" data-scrolled="true" data-execution_count="13">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>panel_information_pmo <span class="op">=</span> panel_info_table_to_pmo_dict(</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>    madhatter_panel_info, </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"mad4hatter_poolsD1R1R2"</span>, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    target_genome_info, </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>    target_id_col<span class="op">=</span><span class="st">"amplicon"</span>,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>    forward_primers_seq_col<span class="op">=</span><span class="st">"fwd_primer"</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    reverse_primers_seq_col<span class="op">=</span><span class="st">"rev_primer"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    forward_primers_start_col<span class="op">=</span><span class="st">"fwd_primer_start_0_based"</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>    forward_primers_end_col<span class="op">=</span><span class="st">"fwd_primer_end_0_based"</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    reverse_primers_start_col<span class="op">=</span><span class="st">"rev_primer_start_0_based"</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>    reverse_primers_end_col<span class="op">=</span><span class="st">"rev_primer_end_0_based"</span>,</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    insert_start_col<span class="op">=</span><span class="st">"insert_start_0_based"</span>,</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>    insert_end_col<span class="op">=</span><span class="st">"insert_end_0_based"</span>,</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    chrom_col<span class="op">=</span><span class="st">"chrom"</span>,</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    strand_col<span class="op">=</span><span class="st">"strand"</span>,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>    gene_id_col<span class="op">=</span><span class="st">"gene_id"</span>,</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>    target_type_col<span class="op">=</span><span class="st">"target_type"</span>,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>You can also add on your own custom fields using the additional_target_info_cols parameter. Below we add on the amplicon insert length information. If there is a field that you want to add and think others would find useful please <a href="https://www.plasmogenepi.org/DataStandards">contact us</a> and we can add it in. This way we can make sure to keep ontologies consistent!</p>
<div id="e71fd00f-269b-4284-8f9d-78a7a9152d71" class="cell" data-execution_count="14">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>panel_information_pmo <span class="op">=</span> panel_info_table_to_pmo_dict(</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    madhatter_panel_info,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"mad4hatter_poolsD1R1R2"</span>,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    target_genome_info,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    target_id_col<span class="op">=</span><span class="st">"amplicon"</span>,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    forward_primers_seq_col<span class="op">=</span><span class="st">"fwd_primer"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    reverse_primers_seq_col<span class="op">=</span><span class="st">"rev_primer"</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    forward_primers_start_col<span class="op">=</span><span class="st">"fwd_primer_start_0_based"</span>,</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    forward_primers_end_col<span class="op">=</span><span class="st">"fwd_primer_end_0_based"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    reverse_primers_start_col<span class="op">=</span><span class="st">"rev_primer_start_0_based"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    reverse_primers_end_col<span class="op">=</span><span class="st">"rev_primer_end_0_based"</span>,</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    insert_start_col<span class="op">=</span><span class="st">"insert_start_0_based"</span>,</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    insert_end_col<span class="op">=</span><span class="st">"insert_end_0_based"</span>,</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    chrom_col<span class="op">=</span><span class="st">"chrom"</span>,</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    strand_col<span class="op">=</span><span class="st">"strand"</span>,</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    gene_id_col<span class="op">=</span><span class="st">"gene_id"</span>,</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    target_type_col<span class="op">=</span><span class="st">"target_type"</span>,</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    additional_target_info_cols<span class="op">=</span>[<span class="st">"ampInsert_length"</span>]</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Letâ€™s have a look at this now with the extra information added to the panel information</p>
<div id="522a81b7-ee41-4a10-b068-e66d71ee7331" class="cell" data-execution_count="15">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>print_json_head(panel_information_pmo,<span class="dv">46</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{
    "panel_info": {
        "mad4hatter_poolsD1R1R2": {
            "panel_id": "mad4hatter_poolsD1R1R2",
            "target_genome": {
                "gff_url": "https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/gff/data/PlasmoDB-65_Pfalciparum3D7.gff",
                "name": "3D7",
                "taxon_id": 5833,
                "url": "https://plasmodb.org/common/downloads/release-65/Pfalciparum3D7/fasta/data/PlasmoDB-65_Pfalciparum3D7_Genome.fasta",
                "version": "2020-09-01"
            },
            "targets": {
                "Pf3D7_01_v3-145388-145662-1A": {
                    "target_id": "Pf3D7_01_v3-145388-145662-1A",
                    "forward_primers": [
                        {
                            "seq": "CCTGAGTTTTAAGTGAATGAATATATTTTTGTT",
                            "location": {
                                "chrom": "Pf3D7_01_v3",
                                "end": 145420,
                                "start": 145387,
                                "strand": "+"
                            }
                        }
                    ],
                    "reverse_primers": [
                        {
                            "seq": "AAAATGTCCAATATGTCAAGGTATATTAAAGT",
                            "location": {
                                "chrom": "Pf3D7_01_v3",
                                "end": 145662,
                                "start": 145630,
                                "strand": "+"
                            }
                        }
                    ],
                    "gene_id": "PF3D7_0103300",
                    "target_type": "diversity",
                    "ampInsert_length": 209,
                    "insert_location": {
                        "chrom": "Pf3D7_01_v3",
                        "start": 145421,
                        "end": 145629,
                        "strand": "+"
                    }
                },</code></pre>
</div>
</div>
</section>
<section id="metadata" class="level3">
<h3 class="anchored" data-anchor-id="metadata">Metadata</h3>
<p>This section will compile metadata on two levels:</p>
<ul>
<li>Specimen Level: Information about specimen that was collected.</li>
<li>Experiment Level: Information about the sequencing or amplification runs performed on a specimen.</li>
</ul>
<p>Itâ€™s important to note that a single specimen may be linked to multiple experiments.</p>
<p>In our example this is stored in one table, but may be stored in multiple places for you.</p>
<div id="751ea9c5-6c8e-4f77-8dcc-9d0e6595d9bf" class="cell" data-execution_count="16">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="op">=</span> pd.read_excel(<span class="st">'example_data/mad4hatter_metadata_example.xlsx'</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>metadata.head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="16">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">specimen_id</th>
<th data-quarto-table-cell-role="th">collection_date</th>
<th data-quarto-table-cell-role="th">collection_country</th>
<th data-quarto-table-cell-role="th">samp_collect_device</th>
<th data-quarto-table-cell-role="th">lat_lon</th>
<th data-quarto-table-cell-role="th">collector</th>
<th data-quarto-table-cell-role="th">geo_admin3</th>
<th data-quarto-table-cell-role="th">host_taxon_id</th>
<th data-quarto-table-cell-role="th">project_name</th>
<th data-quarto-table-cell-role="th">samp_store_loc</th>
<th data-quarto-table-cell-role="th">samp_taxon_id</th>
<th data-quarto-table-cell-role="th">experiment_sample_id</th>
<th data-quarto-table-cell-role="th">panel_id</th>
<th data-quarto-table-cell-role="th">plate_name</th>
<th data-quarto-table-cell-role="th">plate_row</th>
<th data-quarto-table-cell-role="th">plate_col</th>
<th data-quarto-table-cell-role="th">sequencing_info_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SAMN38241219</td>
<td>2019-01</td>
<td>Mozambique</td>
<td>dried blood spot</td>
<td>25.58,32.35</td>
<td>Brokhattingen, Nanna</td>
<td>Maputo</td>
<td>1758</td>
<td>PRJNA1040019</td>
<td>UCSF Greenhouse Lab</td>
<td>5833</td>
<td>SRR26819135</td>
<td>Mad4hatter</td>
<td>plate1</td>
<td>A</td>
<td>1</td>
<td>run1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SAMN38241215</td>
<td>2017-02</td>
<td>Mozambique</td>
<td>dried blood spot</td>
<td>25.58,32.35</td>
<td>Brokhattingen, Nanna</td>
<td>Maputo</td>
<td>1758</td>
<td>PRJNA1040019</td>
<td>UCSF Greenhouse Lab</td>
<td>5833</td>
<td>SRR26819139</td>
<td>Mad4hatter</td>
<td>plate1</td>
<td>A</td>
<td>2</td>
<td>run1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SAMN38241214</td>
<td>2016-05</td>
<td>Mozambique</td>
<td>dried blood spot</td>
<td>25.58,32.35</td>
<td>Brokhattingen, Nanna</td>
<td>Maputo</td>
<td>1758</td>
<td>PRJNA1040019</td>
<td>UCSF Greenhouse Lab</td>
<td>5833</td>
<td>SRR26819141</td>
<td>Mad4hatter</td>
<td>plate1</td>
<td>A</td>
<td>3</td>
<td>run1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SAMN38241052</td>
<td>2015-05</td>
<td>Mozambique</td>
<td>dried blood spot</td>
<td>25.58,32.35</td>
<td>Brokhattingen, Nanna</td>
<td>Maputo</td>
<td>1758</td>
<td>PRJNA1040019</td>
<td>UCSF Greenhouse Lab</td>
<td>5833</td>
<td>SRR26819151</td>
<td>Mad4hatter</td>
<td>plate1</td>
<td>A</td>
<td>4</td>
<td>run1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SAMN38241112</td>
<td>2019-06</td>
<td>Mozambique</td>
<td>dried blood spot</td>
<td>25.58,32.35</td>
<td>Brokhattingen, Nanna</td>
<td>Maputo</td>
<td>1758</td>
<td>PRJNA1040019</td>
<td>UCSF Greenhouse Lab</td>
<td>5833</td>
<td>SRR26819200</td>
<td>Mad4hatter</td>
<td>plate1</td>
<td>A</td>
<td>5</td>
<td>run1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<section id="specimen-level-metadata" class="level4">
<h4 class="anchored" data-anchor-id="specimen-level-metadata">Specimen Level Metadata</h4>
<p>Now we put together the specimen level metadata. This is the metadata associated with the sample collected from the host. For more information on this section see the <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#specimeninfo">documentation</a>.</p>
<div id="26203fd9-e278-422e-bee8-f9887d11c23c" class="cell" data-execution_count="17">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(specimen_info_table_to_json.__doc__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Converts a DataFrame containing specimen information into JSON.

    :param contents (pd.DataFrame): The input DataFrame containing experiment data.
    :param specimen_id_col (str): The column name for specimen sample IDs. Default: specimen_id
    :param samp_taxon_id (int): NCBI taxonomy number of the organism. Default: samp_taxon_id
    :param collection_date (string): Date of the sample collection. Default: collection_date
    :param collection_country (string): Name of country collected in (admin level 0). Default : collection_country
    :param collector (string): Name of the primary person managing the specimen. Default: collector
    :param samp_store_loc (string): Sample storage site. Default: samp_store_loc
    :param samp_collect_device (string): The way the sample was collected. Default : samp_collect_device
    :param project_name (string): Name of the project. Default : project_name
    :param alternate_identifiers (Optional[str]): List of optional alternative names for the samples
    :param geo_admin1 (Optional[str]): Geographical admin level 1
    :param geo_admin2 (Optional[str]): Geographical admin level 2
    :param geo_admin3 (Optional[str]): Geographical admin level 3
    :param host_taxon_id (Optional[int]): NCBI taxonomy number of the host
    :param individual_id (Optional[str]): ID for the individual a specimen was collected from
    :param lat_lon (Optional[str]): Latitude and longitude of the collection site
    :param parasite_density (Optional[float]): The parasite density
    :param plate_col (Optional[int]): Column the specimen was in in the plate
    :param plate_name (Optional[str]): Name of plate the specimen was in
    :param plate_row (Optional[str]): Row the specimen was in in the plate
    :param sample_comments (Optional[str]): Additional comments about the sample
    :param additional_specimen_cols (Optional[List[str], None]]): Additional column names to include

    :return: JSON format where keys are `specimen_id` and values are corresponding row data.
    </code></pre>
</div>
</div>
<div id="d732cd5e-24e2-430c-bba1-4d8f2ed2a2f5" class="cell" data-execution_count="18">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>specimen_info_json <span class="op">=</span> specimen_info_table_to_json(metadata, geo_admin3<span class="op">=</span><span class="st">'geo_admin3'</span>,host_taxon_id<span class="op">=</span><span class="st">'host_taxon_id'</span>, lat_lon<span class="op">=</span><span class="st">'lat_lon'</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>print_json_head(specimen_info_json, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{
    "SAMN38241219": {
        "specimen_id": "SAMN38241219",
        "samp_taxon_id": 5833,
        "collection_date": "2019-01",
        "collection_country": "Mozambique",
        "collector": "Brokhattingen, Nanna",
        "samp_store_loc": "UCSF Greenhouse Lab",
        "samp_collect_device": "dried blood spot",
        "project_name": "PRJNA1040019",
        "geo_admin3": "Maputo",
        "host_taxon_id": 1758,
        "lat_lon": "25.58,32.35"
    },
    "SAMN38241215": {
        "specimen_id": "SAMN38241215",
        "samp_taxon_id": 5833,
        "collection_date": "2017-02",
        "collection_country": "Mozambique",
        "collector": "Brokhattingen, Nanna",</code></pre>
</div>
</div>
</section>
<section id="experiment-level-metadata" class="level4">
<h4 class="anchored" data-anchor-id="experiment-level-metadata">Experiment Level Metadata</h4>
<p>This section shows how to put together the experiment level metadata. More information on this table can be found [here](pd.read_excel(â€˜example_data/mad4hatter_experiment_info_table_example.xlsxâ€™).</p>
<div id="f22f1ddb-6587-4182-b315-c61aa8456b95" class="cell" data-execution_count="19">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(experiment_info_table_to_json.__doc__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Converts a DataFrame containing experiment information into JSON.

    :param contents (pd.DataFrame): Input DataFrame containing experiment data.
    :param experiment_sample_id_col (str): Column name for experiment sample IDs. Default: experiment_sample_id
    :param sequencing_info_id (str): Column name for sequencing information IDs. Default: sequencing_info_id
    :param specimen_id (str): Column name for specimen IDs. Default: specimen_id
    :param panel_id (str): Column name for panel IDs. Default: panel_id
    :param accession (Optional[str]): Column name for accession information.
    :param plate_col (Optional[int]): Column index for plate information.
    :param plate_name (Optional[str]): Column name for plate names.
    :param plate_row (Optional[str]): Column name for plate rows.
    :param additional_experiment_cols (Optional[List[str], None]]): Additional column names to include.

    :return: JSON format where keys are `experiment_sample_id` and values are corresponding row data.
    </code></pre>
</div>
</div>
<div id="420e8468-5b7d-4eaf-acd5-a58eb7921c35" class="cell" data-execution_count="20">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>experiment_info_json <span class="op">=</span> experiment_info_table_to_json(metadata, plate_name<span class="op">=</span><span class="st">'plate_name'</span>, plate_col<span class="op">=</span><span class="st">'plate_col'</span>,plate_row<span class="op">=</span><span class="st">'plate_row'</span>, additional_experiment_cols<span class="op">=</span>[<span class="st">'collection_date'</span>,<span class="st">'collection_country'</span>])</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>print_json_head(experiment_info_json, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{
    "SRR26819135": {
        "experiment_sample_id": "SRR26819135",
        "sequencing_info_id": "run1",
        "specimen_id": "SAMN38241219",
        "panel_id": "Mad4hatter",
        "plate_col": 1,
        "plate_name": "plate1",
        "plate_row": "A",
        "collection_date": "2019-01",
        "collection_country": "Mozambique"
    },
    "SRR26819139": {
        "experiment_sample_id": "SRR26819139",
        "sequencing_info_id": "run1",
        "specimen_id": "SAMN38241215",
        "panel_id": "Mad4hatter",
        "plate_col": 2,
        "plate_name": "plate1",
        "plate_row": "A",</code></pre>
</div>
</div>
</section>
</section>
<section id="microhaplotype-information" class="level3">
<h3 class="anchored" data-anchor-id="microhaplotype-information">Microhaplotype Information</h3>
<p>Next, weâ€™ll organize the microhaplotype information into the required format.</p>
<p>This involves two components that we will generate from one table(click on the links to find out more information about each part):</p>
<ul>
<li>The <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#representativemicrohaplotypesequences">representative microhaplotype details</a>: A summary of all of unique microhaplotypes called within the population you have included in your PMO for each target. Each unique microhaplotype will be assigned a short ID within PMO to improve the scalability of the format.</li>
<li>The <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#microhaplotypesdetected">detected microhaplotypes</a>: Microhaplotypes called for each sample for each target and the associated reads. This will be linked to the above table using the generated microhaplotype ID instead of the full microhaplotype sequence.</li>
</ul>
<p>First we will load an example allele table that may be similar to something you have from your own microhaplotype pipeline. This table includes a sampleID, the target, and the ASV and number of reads detected for each of these.</p>
<div id="cfe02fb2-580f-4013-8a72-41e5f1f7c063" class="cell" data-execution_count="21">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>example_allele_table <span class="op">=</span> pd.read_csv(allele_table_path, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>example_allele_table.head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="21">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">SampleID</th>
<th data-quarto-table-cell-role="th">Locus</th>
<th data-quarto-table-cell-role="th">ASV</th>
<th data-quarto-table-cell-role="th">Reads</th>
<th data-quarto-table-cell-role="th">Allele</th>
<th data-quarto-table-cell-role="th">PseudoCIGAR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR26819553</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>GATATGTTTAAATATATGATTCTCGAAAAAACTTTTTTTATTTTTT...</td>
<td>13</td>
<td>Pf3D7_01_v3-145388-145662-1A.1</td>
<td>25+25N169+8N188+9N</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR26819207</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>GATATGTTTAAATATATGATTCTCGAAAAAACTTTTTTTATTTTTT...</td>
<td>4</td>
<td>Pf3D7_01_v3-145388-145662-1A.2</td>
<td>25+25N94A139T169+8N188+9N</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR26819545</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>GATATGTTTAAATATATGATTCTCGAAAAAACTTTTTTTATTTTTT...</td>
<td>22</td>
<td>Pf3D7_01_v3-145388-145662-1A.1</td>
<td>25+25N169+8N188+9N</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR26819527</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>GATATGTTTAAATATATGATTCTCGAAAAAACTTTTTTTATTTTTT...</td>
<td>1</td>
<td>Pf3D7_01_v3-145388-145662-1A.2</td>
<td>25+25N94A139T169+8N188+9N</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR26819214</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>GATATGTTTAAATATATGATTCTCGAAAAAACTTTTTTTATTTTTT...</td>
<td>14</td>
<td>Pf3D7_01_v3-145388-145662-1A.3</td>
<td>25+25N139T169+8N188+9N</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Letâ€™s have a look at the function we will use to create this part of PMO <code>microhaplotype_table_to_pmo_dict</code></p>
<div id="c43520aa-6afb-4b93-961e-2256a7c6a024" class="cell" data-execution_count="22">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(microhaplotype_table_to_pmo_dict.__doc__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Convert a dataframe of a microhaplotype calls into a dictionary containing a dictionary for the haplotypes_detected and a dictionary for the representative_haplotype_sequences.

    :param contents: The dataframe containing microhaplotype calls
    :param bioinfo_id: the bioinformatics ID of the microhaplotype table
    :param sampleID_col: the name of the column containing the sample IDs
    :param locus_col: the name of the column containing the locus IDs
    :param mhap_col: the name of the column containing the microhaplotype sequence
    :param reads_col: the name of the column containing the reads counts
    :param additional_hap_detected_cols: optional additional columns to add to the microhaplotype detected dictionary, the key is the pandas column and the value is what to name it in the output
    :return: a dict of both the haplotypes_detected and representative_haplotype_sequences
    </code></pre>
</div>
</div>
<p>We can see that we need a dataframe with columns for <strong>sample IDs</strong>, <strong>locus names</strong>, <strong>microhaplotype sequences</strong>, and their corresponding <strong>read counts</strong>. We also need to supply a unique <strong>bioinformatics ID</strong>. Including this ID allows us to store results from multiple bioinformatics pipelines run on the same sequencing data in a unified format if necessary.</p>
<p>Here we set a bioinformatics ID, so we can use the same one when generating other tables later on.</p>
<div id="d3a4809a-4f15-4fba-a4e0-c53ba9ed8711" class="cell" data-execution_count="23">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>bioinfo_id <span class="op">=</span> <span class="st">"Mozambique2018-MAD4HatTeR"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The function has default column names that align with the standard output from DADA2. However, since weâ€™re using MAD4HatTeR data, which has slightly different column headers, weâ€™ll need to specify these column names explicitly in the function.</p>
<div id="dac31235-eca9-4d9c-a31c-38f4538bb2e3" class="cell" data-execution_count="24">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>microhaplotype_info <span class="op">=</span> microhaplotype_table_to_pmo_dict(</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>    example_allele_table,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>    sampleID_col<span class="op">=</span><span class="st">"SampleID"</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>    locus_col<span class="op">=</span><span class="st">"Locus"</span>,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>    mhap_col<span class="op">=</span><span class="st">"ASV"</span>,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>    reads_col<span class="op">=</span><span class="st">"Reads"</span>,</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>    bioinfo_id<span class="op">=</span>bioinfo_id,</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="demultiplexed-experiment-samples" class="level3">
<h3 class="anchored" data-anchor-id="demultiplexed-experiment-samples">Demultiplexed Experiment Samples</h3>
<p>We also include information on the demultiplexed reads for each Sample for each target using a function called <code>demultiplexed_targets_to_pmo_dict</code>.</p>
<div id="41d2294e-6863-4536-9e44-f559d5042134" class="cell" data-execution_count="25">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(demultiplexed_targets_to_pmo_dict.__doc__)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Convert a dataframe of microhaplotype calls into a dictionary for detected haplotypes 
    and representative haplotype sequences.

    :param contents: DataFrame containing demultiplexed sample information
    :param bioinfo_id: Bioinformatics ID of the demultiplexed targets
    :param sampleID_col: Name of the column containing sample IDs
    :param target_id_col: Name of the column containing locus IDs
    :param read_count_col: Name of the column containing read counts
    :param additional_hap_detected_cols: Optional columns to include in the output,
                                         with keys as column names and values as their output names
    :return: JSON string containing the processed data
    </code></pre>
</div>
</div>
<div id="da6ad600-25f0-4501-a1dc-48f7e706efdc" class="cell" data-execution_count="26">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>amplicon_coverage <span class="op">=</span> pd.read_csv(demultiplexed_reads_path, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>amplicon_coverage.head()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="26">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">SampleID</th>
<th data-quarto-table-cell-role="th">Locus</th>
<th data-quarto-table-cell-role="th">Reads</th>
<th data-quarto-table-cell-role="th">OutputDada2</th>
<th data-quarto-table-cell-role="th">OutputPostprocessing</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>SRR26819135</td>
<td>Pf3D7_01_v3-145388-145662-1A</td>
<td>54</td>
<td>54</td>
<td>54</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>SRR26819135</td>
<td>Pf3D7_01_v3-162867-163115-1A</td>
<td>400</td>
<td>398</td>
<td>398</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>SRR26819135</td>
<td>Pf3D7_01_v3-181512-181761-1A</td>
<td>266</td>
<td>266</td>
<td>266</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>SRR26819135</td>
<td>Pf3D7_01_v3-455794-456054-1A</td>
<td>81</td>
<td>80</td>
<td>80</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>SRR26819135</td>
<td>Pf3D7_01_v3-528859-529104-1A</td>
<td>485</td>
<td>485</td>
<td>485</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Note that we use the same <code>bioinfo_id</code> that we set above.</p>
<div id="da4ccf13-3415-466f-8c67-0df901dde215" class="cell" data-execution_count="27">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>demultiplexed_targets_pmo <span class="op">=</span> demultiplexed_targets_to_pmo_dict(amplicon_coverage, bioinfo_id,  sampleID_col <span class="op">=</span> <span class="st">'SampleID'</span>, target_id_col<span class="op">=</span><span class="st">'Locus'</span>,read_count_col <span class="op">=</span><span class="st">'Reads'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="eab4daab-d48c-49ac-848d-7c7160dee8d1" class="cell" data-execution_count="28">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>print_json_head(demultiplexed_targets_pmo, <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>{
    "target_demultiplexed_experiment_samples": {
        "Mozambique2018-MAD4HatTeR": {
            "demultiplexed_experiment_samples": {
                "SRR26819135": {
                    "demultiplexed_targets": {
                        "experiment_sample_id": "SRR26819135",
                        "Pf3D7_01_v3-145388-145662-1A": {
                            "raw_read_count": 54,
                            "target_id": "Pf3D7_01_v3-145388-145662-1A"
                        },
                        "Pf3D7_01_v3-162867-163115-1A": {
                            "raw_read_count": 400,
                            "target_id": "Pf3D7_01_v3-162867-163115-1A"
                        },</code></pre>
</div>
</div>
</section>
<section id="sequencing-info" class="level3">
<h3 class="anchored" data-anchor-id="sequencing-info">Sequencing info</h3>
<p>PMO includes details of the sequencing run. Below is an example and more information on the required fields can be found <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#sequencinginfo">here</a></p>
<div id="17825a8d-c6b9-44f5-b461-98cc6adb78d8" class="cell" data-execution_count="29">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>sequencing_infos <span class="op">=</span>{</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mozambique2018"</span> : </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>        {</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>            <span class="st">"lib_kit"</span> : <span class="st">"TruSeq i5/i7 barcode primers"</span>,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>            <span class="st">"lib_layout"</span> : <span class="st">"paired-end"</span>,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>            <span class="st">"lib_screen"</span> : <span class="st">"40 ÂµL reaction containing 10 ÂµL of bead purified digested product, 18Î¼L of nuclease-free water, 8Î¼L of 5X secondary PCR master mix, and 5 ÂµL of 10 ÂµM TruSeq i5/i7 barcode primers"</span>,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>            <span class="st">"nucl_acid_amp"</span> : <span class="st">"https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/"</span>,</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">"nucl_acid_date"</span> : <span class="st">"2019-07-15"</span>,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">"nucl_acid_ext"</span> : <span class="st">"https://www.paragongenomics.com/targeted-sequencing/amplicon-sequencing/cleanplex-ngs-amplicon-sequencing/"</span>,</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>            <span class="st">"pcr_cond"</span> : <span class="st">"10 min at 95Â°C, 13 cycles for high density samples (or 15 cycles for low density samples) of 15 sec at 98Â°C and 75 sec at 60Â°C"</span>,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>            <span class="st">"seq_center"</span> : <span class="st">"UCSF"</span>,</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>            <span class="st">"seq_date"</span> : <span class="st">"2019-07-15"</span>,</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>            <span class="st">"seq_instrument"</span> : <span class="st">"NextSeq 550 instrument"</span>,</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>            <span class="st">"sequencing_info_id"</span> : <span class="st">"run1"</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>    }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="bioinformatics-info" class="level3">
<h3 class="anchored" data-anchor-id="bioinformatics-info">Bioinformatics Info</h3>
<p>Now we manually enter some information on the bioinformatics run. More information on the fields can be found <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverview.html#tarampbioinformaticsinfo">here</a>. Below is an example from the MAD4HatTeR pipeline.</p>
<div id="4530da0e-70da-44b8-8264-90e20eb754e1" class="cell" data-execution_count="30">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>taramp_bioinformatics_infos <span class="op">=</span> {</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    bioinfo_id : </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"demultiplexing_method"</span> : </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>        {</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>            <span class="st">"program"</span> : <span class="st">"Cutadapt extractorPairedEnd"</span>,</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>            <span class="st">"purpose"</span> : <span class="st">"Takes raw paired-end reads and demultiplexes on primers and does QC filtering"</span>,</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">"version"</span> : <span class="st">"v4.4"</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"denoising_method"</span> : </span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>        {</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>            <span class="st">"program"</span> : <span class="st">"DADA2"</span>,</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>            <span class="st">"purpose"</span> : <span class="st">"Takes sequences per sample per target and clusters them"</span>,</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>            <span class="st">"version"</span> : <span class="st">"v3.16"</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tar_amp_bioinformatics_info_id"</span> : bioinfo_id</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="compose-pmo" class="level1">
<h1>Compose PMO</h1>
<p>To create our final PMO we will put together all of the parts we have created</p>
<div id="0aa8909b-93a8-4488-81ca-d062cd03db83" class="cell" data-execution_count="48">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Put together the information we h</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>format_pmo <span class="op">=</span> {</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"experiment_infos"</span>: experiment_info_json,  </span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"sequencing_infos"</span>: sequencing_infos, </span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"specimen_infos"</span>: specimen_info_json, </span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"taramp_bioinformatics_infos"</span>: taramp_bioinformatics_infos, </span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>    <span class="op">**</span>microhaplotype_info, </span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>    <span class="op">**</span>panel_information_pmo,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="op">**</span>demultiplexed_targets_pmo,</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Finally we output this to a file</p>
<div id="42f0fbe9-dd90-47f5-bcc1-73d094d58311" class="cell" data-execution_count="32">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Write to a JSON file</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>output_file <span class="op">=</span> <span class="st">"example_pmo.json"</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> <span class="bu">open</span>(output_file, <span class="st">"w"</span>) <span class="im">as</span> f:</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    json.dump(format_pmo, f, indent<span class="op">=</span><span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Thatâ€™s it! You have put together a PMO, congratulations.</p>
<p>Next time you should be able to reuse multiple parts with minor tweaks. See the rest of the documentation for ways that you can work with your new PMO file.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>