<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Extracting allele tables using pmotools-python – Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-49a824ff064150fa59632921ab93e189.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6a2c4b793c0becaeb94cc5bf146c1484.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>html{ scroll-behavior: smooth; }</style>
<link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">
<li>
    <a class="dropdown-item" href="../format/DevelopmentOfFormat.html">
 <span class="dropdown-text">Development of Format</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatOverviewAdvanced.html">
 <span class="dropdown-text">Format Overview For Developers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">
<li>
    <a class="dropdown-item" href="../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">
<li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-python</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/pmotools-python.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Getting sub info from PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_panel_info_from_pmo.html">
 <span class="dropdown-text">Getting panel info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Validating PMO files</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/validating_pmos.html">
 <span class="dropdown-text">Validating PMOs pmotools-python</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/build_pmo.html">
 <span class="dropdown-text">Creating a PMO File</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">
<li>
    <a class="dropdown-item" href="../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li class="dropdown-header"><span class="menuTitle">Documentation</span></li>
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/PMO_Docs">
 <span class="dropdown-text">Documentation Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/PMO_Docs/issues">
 <span class="dropdown-text">Comment or Report an issue for Documentation</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-python</span></li>
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/pmotools-python/tree/develop">
 <span class="dropdown-text">pmotools-python Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/pmotools-python/issues">
 <span class="dropdown-text">Comment or Report an issue for pmotools-python</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Contents</h2>
   
  <ul>
<li>
<a href="#pmotools-python-extract_allele_table" id="toc-pmotools-python-extract_allele_table" class="nav-link active" data-scroll-target="#pmotools-python-extract_allele_table">pmotools-python extract_allele_table</a>
  <ul>
<li><a href="#creating-output-for-moire" id="toc-creating-output-for-moire" class="nav-link" data-scroll-target="#creating-output-for-moire">Creating output for MOIRE</a></li>
  <li><a href="#creating-output-for-dcifer" id="toc-creating-output-for-dcifer" class="nav-link" data-scroll-target="#creating-output-for-dcifer">Creating output for dcifer</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Extracting allele tables using pmotools-python</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-page-left">

    
  
    
  </div>
  


</header><section id="pmotools-python-extract_allele_table" class="level1"><h1>pmotools-python extract_allele_table</h1>
<p>To extract allele table information from a PMO the command line interactive script with <code>pmotools-python extract_allele_table</code> can be used</p>
<ul>
<li><p>pmotools-python extract_allele_table</p></li>
<li>
<p><strong>Required arguments</strong></p>
<ul>
<li>
<strong>--file</strong> - the PMO file to extract from</li>
<li>
<strong>--output</strong> - the output stub of the files to be created</li>
</ul>
</li>
<li><p><strong>Optional arguments</strong></p></li>
</ul>
<p>By default only 3 fields are extracted by this extractor, 1) sampleID (library_sample_sample_name), 2) locus (target_name), and 3) allele (microhaplotype_id) with those default column names. This can be controlled by <strong>--default_base_col_names</strong> and if you supply 3 comma separated values you can change the default header.</p>
<p>You can also add to the table any values from the other portions of the PMO file by using the following arguments</p>
<ul>
<li>adding fields arguments
<ul>
<li>
<strong>--specimen_info_meta_fields</strong> - Meta Fields if any to include from the specimen table</li>
<li>
<strong>--library_sample_info_meta_fields</strong> - Meta Fields if any to include from the library_sample table</li>
<li>
<strong>--microhap_fields</strong> - additional optional fields from the detected microhaplotype object to include</li>
<li>
<strong>--representative_haps_fields</strong> - additional optional fields from the detected representative object to include</li>
</ul>
</li>
</ul>
<p>Other optional arguments have to do with the ouput file over writing and delimiter being used, use <code>-h</code> to see all arguments</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">-h</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-python extract_allele_table [-h] --file FILE
                                            [--jsonschema JSONSCHEMA]
                                            [--delim DELIM] --output OUTPUT
                                            [--overwrite]
                                            [--allele_freqs_output ALLELE_FREQS_OUTPUT]
                                            [--specimen_info_meta_fields SPECIMEN_INFO_META_FIELDS]
                                            [--library_sample_info_meta_fields LIBRARY_SAMPLE_INFO_META_FIELDS]
                                            [--microhap_fields MICROHAP_FIELDS]
                                            [--representative_haps_fields REPRESENTATIVE_HAPS_FIELDS]
                                            [--default_base_col_names DEFAULT_BASE_COL_NAMES]

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --jsonschema JSONSCHEMA
                        the jsonschema to check the PMO against
  --delim DELIM         the delimiter of the input text file, examples
                        tab,comma
  --output OUTPUT       Output allele table file name path
  --overwrite           If output file exists, overwrite it
  --allele_freqs_output ALLELE_FREQS_OUTPUT
                        if also writing out allele frequencies, write to this
                        file
  --specimen_info_meta_fields SPECIMEN_INFO_META_FIELDS
                        Meta Fields if any to include from the specimen table
  --library_sample_info_meta_fields LIBRARY_SAMPLE_INFO_META_FIELDS
                        Meta Fields if any to include from the library sample
                        table
  --microhap_fields MICROHAP_FIELDS
                        additional optional fields from the detected
                        microhaplotype object to include
  --representative_haps_fields REPRESENTATIVE_HAPS_FIELDS
                        additional optional fields from the detected
                        representative object to include
  --default_base_col_names DEFAULT_BASE_COL_NAMES
                        default base column names, must be length 3</code></pre>
</div>
</div>
<p>The python code for <code>extract_allele_table</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/src/pmotools/scripts/extractors_from_pmo/extract_allele_table.py</strong></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="im">import</span> argparse</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="im">import</span> json</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="im">import</span> os</span>
<span id="cb3-6"><a href="#cb3-6"></a></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="im">from</span> pmotools.pmo_engine.pmo_reader <span class="im">import</span> PMOReader</span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="im">from</span> pmotools.pmo_engine.pmo_checker <span class="im">import</span> PMOChecker</span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="im">from</span> pmotools.pmo_engine.pmo_processor <span class="im">import</span> PMOProcessor</span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="im">from</span> pmotools <span class="im">import</span> __version__ <span class="im">as</span> __pmotools_version__</span>
<span id="cb3-12"><a href="#cb3-12"></a></span>
<span id="cb3-13"><a href="#cb3-13"></a></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="kw">def</span> parse_args_extract_for_allele_table():</span>
<span id="cb3-15"><a href="#cb3-15"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb3-16"><a href="#cb3-16"></a>    parser.add_argument(<span class="st">"--file"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">"PMO file"</span>)</span>
<span id="cb3-17"><a href="#cb3-17"></a>    parser.add_argument(</span>
<span id="cb3-18"><a href="#cb3-18"></a>        <span class="st">"--jsonschema"</span>,</span>
<span id="cb3-19"><a href="#cb3-19"></a>        default<span class="op">=</span>os.path.join(</span>
<span id="cb3-20"><a href="#cb3-20"></a>            os.path.dirname(</span>
<span id="cb3-21"><a href="#cb3-21"></a>                os.path.dirname(os.path.dirname(os.path.abspath(<span class="va">__file__</span>)))</span>
<span id="cb3-22"><a href="#cb3-22"></a>            ),</span>
<span id="cb3-23"><a href="#cb3-23"></a>            <span class="st">"schemas/"</span>,</span>
<span id="cb3-24"><a href="#cb3-24"></a>            <span class="ss">f"portable_microhaplotype_object_v</span><span class="sc">{</span>__pmotools_version__<span class="sc">}</span><span class="ss">.schema.json"</span>,</span>
<span id="cb3-25"><a href="#cb3-25"></a>        ),</span>
<span id="cb3-26"><a href="#cb3-26"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-27"><a href="#cb3-27"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-28"><a href="#cb3-28"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"the jsonschema to check the PMO against"</span>,</span>
<span id="cb3-29"><a href="#cb3-29"></a>    )</span>
<span id="cb3-30"><a href="#cb3-30"></a></span>
<span id="cb3-31"><a href="#cb3-31"></a>    parser.add_argument(</span>
<span id="cb3-32"><a href="#cb3-32"></a>        <span class="st">"--delim"</span>,</span>
<span id="cb3-33"><a href="#cb3-33"></a>        default<span class="op">=</span><span class="st">"tab"</span>,</span>
<span id="cb3-34"><a href="#cb3-34"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-35"><a href="#cb3-35"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-36"><a href="#cb3-36"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"the delimiter of the input text file, examples tab,comma"</span>,</span>
<span id="cb3-37"><a href="#cb3-37"></a>    )</span>
<span id="cb3-38"><a href="#cb3-38"></a>    parser.add_argument(</span>
<span id="cb3-39"><a href="#cb3-39"></a>        <span class="st">"--output"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">"Output allele table file name path"</span></span>
<span id="cb3-40"><a href="#cb3-40"></a>    )</span>
<span id="cb3-41"><a href="#cb3-41"></a>    parser.add_argument(</span>
<span id="cb3-42"><a href="#cb3-42"></a>        <span class="st">"--overwrite"</span>, action<span class="op">=</span><span class="st">"store_true"</span>, <span class="bu">help</span><span class="op">=</span><span class="st">"If output file exists, overwrite it"</span></span>
<span id="cb3-43"><a href="#cb3-43"></a>    )</span>
<span id="cb3-44"><a href="#cb3-44"></a>    parser.add_argument(</span>
<span id="cb3-45"><a href="#cb3-45"></a>        <span class="st">"--allele_freqs_output"</span>,</span>
<span id="cb3-46"><a href="#cb3-46"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-47"><a href="#cb3-47"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"if also writing out allele frequencies, write to this file"</span>,</span>
<span id="cb3-48"><a href="#cb3-48"></a>    )</span>
<span id="cb3-49"><a href="#cb3-49"></a></span>
<span id="cb3-50"><a href="#cb3-50"></a>    parser.add_argument(</span>
<span id="cb3-51"><a href="#cb3-51"></a>        <span class="st">"--specimen_info_meta_fields"</span>,</span>
<span id="cb3-52"><a href="#cb3-52"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-53"><a href="#cb3-53"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-54"><a href="#cb3-54"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"Meta Fields if any to include from the specimen table"</span>,</span>
<span id="cb3-55"><a href="#cb3-55"></a>    )</span>
<span id="cb3-56"><a href="#cb3-56"></a>    parser.add_argument(</span>
<span id="cb3-57"><a href="#cb3-57"></a>        <span class="st">"--library_sample_info_meta_fields"</span>,</span>
<span id="cb3-58"><a href="#cb3-58"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-59"><a href="#cb3-59"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-60"><a href="#cb3-60"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"Meta Fields if any to include from the library sample table"</span>,</span>
<span id="cb3-61"><a href="#cb3-61"></a>    )</span>
<span id="cb3-62"><a href="#cb3-62"></a>    parser.add_argument(</span>
<span id="cb3-63"><a href="#cb3-63"></a>        <span class="st">"--microhap_fields"</span>,</span>
<span id="cb3-64"><a href="#cb3-64"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-65"><a href="#cb3-65"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-66"><a href="#cb3-66"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"additional optional fields from the detected microhaplotype object to include"</span>,</span>
<span id="cb3-67"><a href="#cb3-67"></a>    )</span>
<span id="cb3-68"><a href="#cb3-68"></a>    parser.add_argument(</span>
<span id="cb3-69"><a href="#cb3-69"></a>        <span class="st">"--representative_haps_fields"</span>,</span>
<span id="cb3-70"><a href="#cb3-70"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-71"><a href="#cb3-71"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-72"><a href="#cb3-72"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"additional optional fields from the detected representative object to include"</span>,</span>
<span id="cb3-73"><a href="#cb3-73"></a>    )</span>
<span id="cb3-74"><a href="#cb3-74"></a>    parser.add_argument(</span>
<span id="cb3-75"><a href="#cb3-75"></a>        <span class="st">"--default_base_col_names"</span>,</span>
<span id="cb3-76"><a href="#cb3-76"></a>        <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>,</span>
<span id="cb3-77"><a href="#cb3-77"></a>        required<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb3-78"><a href="#cb3-78"></a>        default<span class="op">=</span><span class="st">"library_sample_name,target_name,mhap_id"</span>,</span>
<span id="cb3-79"><a href="#cb3-79"></a>        <span class="bu">help</span><span class="op">=</span><span class="st">"default base column names, must be length 3"</span>,</span>
<span id="cb3-80"><a href="#cb3-80"></a>    )</span>
<span id="cb3-81"><a href="#cb3-81"></a></span>
<span id="cb3-82"><a href="#cb3-82"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb3-83"><a href="#cb3-83"></a></span>
<span id="cb3-84"><a href="#cb3-84"></a></span>
<span id="cb3-85"><a href="#cb3-85"></a><span class="kw">def</span> extract_for_allele_table():</span>
<span id="cb3-86"><a href="#cb3-86"></a>    args <span class="op">=</span> parse_args_extract_for_allele_table()</span>
<span id="cb3-87"><a href="#cb3-87"></a>    compressed_output <span class="op">=</span> (</span>
<span id="cb3-88"><a href="#cb3-88"></a>        <span class="st">"."</span> <span class="kw">not</span> <span class="kw">in</span> args.output <span class="kw">and</span> args.<span class="bu">file</span>.endswith(<span class="st">".gz"</span>)</span>
<span id="cb3-89"><a href="#cb3-89"></a>    ) <span class="kw">or</span> args.output.endswith(<span class="st">".gz"</span>)</span>
<span id="cb3-90"><a href="#cb3-90"></a></span>
<span id="cb3-91"><a href="#cb3-91"></a>    output_delim, output_extension <span class="op">=</span> Utils.process_delimiter_and_output_extension(</span>
<span id="cb3-92"><a href="#cb3-92"></a>        args.delim, gzip<span class="op">=</span>compressed_output</span>
<span id="cb3-93"><a href="#cb3-93"></a>    )</span>
<span id="cb3-94"><a href="#cb3-94"></a></span>
<span id="cb3-95"><a href="#cb3-95"></a>    allele_per_sample_table_out_fnp <span class="op">=</span> (</span>
<span id="cb3-96"><a href="#cb3-96"></a>        args.output</span>
<span id="cb3-97"><a href="#cb3-97"></a>        <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output</span>
<span id="cb3-98"><a href="#cb3-98"></a>        <span class="cf">else</span> Utils.appendStrAsNeeded(args.output, output_extension)</span>
<span id="cb3-99"><a href="#cb3-99"></a>    )</span>
<span id="cb3-100"><a href="#cb3-100"></a>    Utils.inputOutputFileCheck(</span>
<span id="cb3-101"><a href="#cb3-101"></a>        args.<span class="bu">file</span>, allele_per_sample_table_out_fnp, args.overwrite</span>
<span id="cb3-102"><a href="#cb3-102"></a>    )</span>
<span id="cb3-103"><a href="#cb3-103"></a></span>
<span id="cb3-104"><a href="#cb3-104"></a>    allele_freq_output <span class="op">=</span> <span class="st">""</span></span>
<span id="cb3-105"><a href="#cb3-105"></a>    <span class="cf">if</span> args.allele_freqs_output <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-106"><a href="#cb3-106"></a>        allele_freq_output <span class="op">=</span> Utils.appendStrAsNeeded(</span>
<span id="cb3-107"><a href="#cb3-107"></a>            args.allele_freqs_output, output_extension</span>
<span id="cb3-108"><a href="#cb3-108"></a>        )</span>
<span id="cb3-109"><a href="#cb3-109"></a>        Utils.inputOutputFileCheck(args.<span class="bu">file</span>, allele_freq_output, args.overwrite)</span>
<span id="cb3-110"><a href="#cb3-110"></a></span>
<span id="cb3-111"><a href="#cb3-111"></a>    pmodata <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb3-112"><a href="#cb3-112"></a>    <span class="cf">with</span> <span class="bu">open</span>(args.jsonschema, <span class="st">"r"</span>) <span class="im">as</span> f:</span>
<span id="cb3-113"><a href="#cb3-113"></a>        schema_dict <span class="op">=</span> json.load(f)</span>
<span id="cb3-114"><a href="#cb3-114"></a>        checker <span class="op">=</span> PMOChecker(schema_dict)</span>
<span id="cb3-115"><a href="#cb3-115"></a>        <span class="co"># make sure PMO is valid</span></span>
<span id="cb3-116"><a href="#cb3-116"></a>        checker.validate_pmo_json(pmodata)</span>
<span id="cb3-117"><a href="#cb3-117"></a></span>
<span id="cb3-118"><a href="#cb3-118"></a>    <span class="cf">if</span> args.specimen_info_meta_fields <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-119"><a href="#cb3-119"></a>        args.specimen_info_meta_fields <span class="op">=</span> Utils.parse_delimited_input_or_file(</span>
<span id="cb3-120"><a href="#cb3-120"></a>            args.specimen_info_meta_fields, <span class="st">","</span></span>
<span id="cb3-121"><a href="#cb3-121"></a>        )</span>
<span id="cb3-122"><a href="#cb3-122"></a>    <span class="cf">if</span> args.microhap_fields <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-123"><a href="#cb3-123"></a>        args.microhap_fields <span class="op">=</span> Utils.parse_delimited_input_or_file(</span>
<span id="cb3-124"><a href="#cb3-124"></a>            args.microhap_fields, <span class="st">","</span></span>
<span id="cb3-125"><a href="#cb3-125"></a>        )</span>
<span id="cb3-126"><a href="#cb3-126"></a>    <span class="cf">if</span> args.library_sample_info_meta_fields <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-127"><a href="#cb3-127"></a>        args.library_sample_info_meta_fields <span class="op">=</span> Utils.parse_delimited_input_or_file(</span>
<span id="cb3-128"><a href="#cb3-128"></a>            args.library_sample_info_meta_fields, <span class="st">","</span></span>
<span id="cb3-129"><a href="#cb3-129"></a>        )</span>
<span id="cb3-130"><a href="#cb3-130"></a>    <span class="cf">if</span> args.representative_haps_fields <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-131"><a href="#cb3-131"></a>        args.representative_haps_fields <span class="op">=</span> Utils.parse_delimited_input_or_file(</span>
<span id="cb3-132"><a href="#cb3-132"></a>            args.representative_haps_fields, <span class="st">","</span></span>
<span id="cb3-133"><a href="#cb3-133"></a>        )</span>
<span id="cb3-134"><a href="#cb3-134"></a></span>
<span id="cb3-135"><a href="#cb3-135"></a>    allele_table <span class="op">=</span> PMOProcessor.extract_alleles_per_sample_table(</span>
<span id="cb3-136"><a href="#cb3-136"></a>        pmodata,</span>
<span id="cb3-137"><a href="#cb3-137"></a>        additional_specimen_info_fields<span class="op">=</span>args.specimen_info_meta_fields,</span>
<span id="cb3-138"><a href="#cb3-138"></a>        additional_library_sample_info_fields<span class="op">=</span>args.library_sample_info_meta_fields,</span>
<span id="cb3-139"><a href="#cb3-139"></a>        additional_microhap_fields<span class="op">=</span>args.microhap_fields,</span>
<span id="cb3-140"><a href="#cb3-140"></a>        additional_representative_info_fields<span class="op">=</span>args.representative_haps_fields,</span>
<span id="cb3-141"><a href="#cb3-141"></a>        default_base_col_names<span class="op">=</span>args.default_base_col_names.split(<span class="st">","</span>),</span>
<span id="cb3-142"><a href="#cb3-142"></a>    )</span>
<span id="cb3-143"><a href="#cb3-143"></a>    <span class="cf">with</span> Utils.smart_open_write(allele_per_sample_table_out_fnp) <span class="im">as</span> f:</span>
<span id="cb3-144"><a href="#cb3-144"></a>        allele_table.to_csv(f, sep<span class="op">=</span>output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb3-145"><a href="#cb3-145"></a></span>
<span id="cb3-146"><a href="#cb3-146"></a>    <span class="cf">if</span> args.allele_freqs_output <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb3-147"><a href="#cb3-147"></a>        allele_freqs <span class="op">=</span> PMOProcessor.extract_allele_counts_freq_from_pmo(pmodata)</span>
<span id="cb3-148"><a href="#cb3-148"></a>        <span class="cf">with</span> Utils.smart_open_write(allele_freq_output) <span class="im">as</span> f:</span>
<span id="cb3-149"><a href="#cb3-149"></a>            allele_freqs.to_csv(f, sep<span class="op">=</span>output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb3-150"><a href="#cb3-150"></a></span>
<span id="cb3-151"><a href="#cb3-151"></a></span>
<span id="cb3-152"><a href="#cb3-152"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb3-153"><a href="#cb3-153"></a>    extract_for_allele_table()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</div>
<p>Can download example PMOs here</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/moz2018_PMO.json.gz </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># default </span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># changing default column names  </span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--default_base_col_names</span> sample,target,hapid</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Changing the output file delimiter</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--delim</span> ,</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Adding on additional columns from the specimen_infos section</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields </span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span>  </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>bioinformatics_run_name library_sample_name target_name mhap_id collection_country  collection_date
Mozambique2018-SeekDeep 8025875168  t99 3   NA  NA
Mozambique2018-SeekDeep 8025875168  t99 0   NA  NA
Mozambique2018-SeekDeep 8025875168  t97 0   NA  NA
Mozambique2018-SeekDeep 8025875168  t97 3   NA  NA
Mozambique2018-SeekDeep 8025875168  t94 1   NA  NA
Mozambique2018-SeekDeep 8025875168  t94 0   NA  NA
Mozambique2018-SeekDeep 8025875168  t90 2   NA  NA
Mozambique2018-SeekDeep 8025875168  t90 0   NA  NA
Mozambique2018-SeekDeep 8025875168  t85 0   NA  NA</code></pre>
</div>
</div>
<p>Can continue to add on more columns from other sections</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields including seq field </span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="at">--representative_haps_fields</span> seq <span class="kw">|</span> <span class="fu">head</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>bioinformatics_run_name library_sample_name target_name mhap_id collection_country  collection_date seq
Mozambique2018-SeekDeep 8025875168  t99 3   NA  NA  ATGGAAAAATGGAATATGAAGTATTAAGTGATGATAAAATAGTGTATGAAAATATACAACATGATTTATTAAAAACAATAGAAGATGGTGAAGAAATGTTAAAAGGAACTGAAAGGAAGGATAATATAGATATACTGAGGACTCCTGGAAGGGGAGAATATAATATGTGGTCTACTTCTGGACTAGGGTTCTATGAAT
Mozambique2018-SeekDeep 8025875168  t99 0   NA  NA  ATGGAAAAATGGAATATGAAGTATTAAGTGATGATAAAATAGTGTATGAAAATATACAACATGATTTATTAAAAACAATAGAAGATGATGACGAAATGTTAAAAGGAACTGAAAGGAAGGATAATATAGATATACTGAGGACTCCTGGAAGGGGAGAATATAATATGTGGTCTACTTCTGGACTAGGGTTCTATGAAT
Mozambique2018-SeekDeep 8025875168  t97 0   NA  NA  TAAACACCAGTACCATTTTTTTCTGATAAATTAATATTTTTTTGTATAACATCATATTTATCCCTTTTCGTGGTAAGTGCAGTATCCTGTTTTATTATTATATTATCGAATTCATCATGGTGTATATTTCTTTCT
Mozambique2018-SeekDeep 8025875168  t97 3   NA  NA  TAATCACCAGTACCATTTTTTTCTGATAAATTAATATTTTTTTGTATAACATCATATTTATCCCTTTTCGTGGTAAGTGCAGTATCCTGTTTTATTATTATATTATCGAATTCATCATGGTGTATATTTCTTTCA
Mozambique2018-SeekDeep 8025875168  t94 1   NA  NA  TATTAAAACTTTTTTTTCTTTCTGTAAAGTTTGTACATTATGTTTTGATGAGTTTTTATTATCTTCATAAAACTTTATATATTTATAAAAATTATTTTGTATAAAATCATTTAATAAAGGTAACATAATTTTTTTAGCTTGATTCAATTCACTACATGAATGTATAT
Mozambique2018-SeekDeep 8025875168  t94 0   NA  NA  TATTAAAACTTTTTTTTCTTTCTGTAAAGTTTGTACATTATGTTTTGATGAGTTTTGATTATCTTCATAAAACTTTATATATTTATAAAAATTATTTTGTATAAAATCATTTAATAAAGGTAACATAATTTTTTTAGCTTGATTCAATTCACTACATGAATGTATAT
Mozambique2018-SeekDeep 8025875168  t90 2   NA  NA  TTACAATGTTCTTCGCATTCGAAATTTTTTTCAGGATTACTTGAAAAGCCTTGTGGACAATTACAATATTCATATCCATGAGCATTCTTACAAACACCTTTTCCACAATTTAAAAAACATTTTTCTTCATTTAA
Mozambique2018-SeekDeep 8025875168  t90 0   NA  NA  TTACAATGTTCTTCGCATTCGAAATTTTTTTCAGGATTACTTGAAAAGCCTTCTGGACAATTACAATATTCATATCCATGAACATTCTTACAAACACCTTTTCCACAATTTAAAAAACATTTTTCTTCATTTAA
Mozambique2018-SeekDeep 8025875168  t85 0   NA  NA  AACATTTTTTTAACATCTTTACCTTTTTGACTTGGTTCTTCATCATAATTCTGTTGTTCTGCAGAATCAGCATTTACTTCAGTTTCTTCTTTATTTTGAAAAGTGTTTGATTCTACATGAGAATTGGAAGATGAACGTCTATGTTTTACTTCTGTATAACTAGTACGTTCTCCAGTATGATGAGCCTTAAGGTTTACATCTTCAGTTTCTT</code></pre>
</div>
</div>
<section id="creating-output-for-moire" class="level2"><h2 class="anchored" data-anchor-id="creating-output-for-moire">Creating output for MOIRE</h2>
<p>MOIRE is a program that can be used to estimate COI and other population estimates from a population. See it’s <a href="https://github.com/EPPIcenter/moire">github</a> for full usage.</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># default table is all moire needs </span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"example/extraction_allele_table.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">load_long_form_data</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># With data in appropriate format, run MCMC as follows</span></span>
<span><span class="va">mcmc_results</span> <span class="op">&lt;-</span> <span class="fu">moire</span><span class="fu">::</span><span class="fu">run_mcmc</span><span class="op">(</span><span class="va">data</span>, is_missing <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">is_missing</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section><section id="creating-output-for-dcifer" class="level2"><h2 class="anchored" data-anchor-id="creating-output-for-dcifer">Creating output for dcifer</h2>
<p>dcifer is a program that can estimate IBD even from mixed infections. See it’s <a href="https://github.com/EPPIcenter/dcifer">github</a> for full usage</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># default </span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--delim</span> ,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># dcifer can calculate allele frequencies if not provided or you can have extract_allele_table write them as well </span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--allele_freqs_output</span> allele_freqs_extraction <span class="at">--delim</span> ,</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dsmp</span> <span class="op">&lt;-</span> <span class="fu">readDat</span><span class="op">(</span><span class="st">"example/extraction_allele_table.csv"</span>, svar <span class="op">=</span> <span class="st">"sampleID"</span>, lvar <span class="op">=</span> <span class="st">"locus"</span>, avar <span class="op">=</span> <span class="st">"allele"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lrank</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">coi</span>   <span class="op">&lt;-</span> <span class="fu">getCOI</span><span class="op">(</span><span class="va">dsmp</span>, lrank <span class="op">=</span> <span class="va">lrank</span><span class="op">)</span></span>
<span></span>
<span><span class="va">afreq</span> <span class="op">&lt;-</span> <span class="fu">calcAfreq</span><span class="op">(</span><span class="va">dsmp</span>, <span class="va">coi</span>, tol <span class="op">=</span> <span class="fl">1e-5</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">dres0</span> <span class="op">&lt;-</span> <span class="fu">ibdDat</span><span class="op">(</span><span class="va">dsmp</span>, <span class="va">coi</span>, <span class="va">afreq</span>, pval <span class="op">=</span> <span class="cn">TRUE</span>, confint <span class="op">=</span> <span class="cn">TRUE</span>, rnull <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                alpha <span class="op">=</span> <span class="fl">0.05</span>, nr <span class="op">=</span> <span class="fl">1e3</span><span class="op">)</span>   </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb16" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Extracting allele tables using pmotools-python</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, echo=F}</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../common.R"</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu"># pmotools-python extract_allele_table</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>To extract allele table information from a PMO the command line interactive script with <span class="in">`pmotools-python extract_allele_table`</span> can be used </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>pmotools-python extract_allele_table </span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>**Required arguments**</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-file** - the PMO file to extract from </span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-output** - the output stub of the files to be created </span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>**Optional arguments** </span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>By default only 3 fields are extracted by this extractor, 1) sampleID (library_sample_sample_name), 2) locus (target_name), and 3) allele (microhaplotype_id) with those default column names. This can be controlled by **\-\-default_base_col_names** and if you supply 3 comma separated values you can change the default header. </span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>You can also add to the table any values from the other portions of the PMO file by using the following arguments </span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="ss">*  </span>adding fields arguments </span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-specimen_info_meta_fields** - Meta Fields if any to include from the specimen table </span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-library_sample_info_meta_fields** - Meta Fields if any to include from the library_sample table </span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-microhap_fields** - additional optional fields from the detected microhaplotype object to include </span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="ss">    *  </span>**\-\-representative_haps_fields** - additional optional fields from the detected representative object to include </span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>Other optional arguments have to do with the ouput file over writing and delimiter being used, use <span class="in">`-h`</span> to see all arguments </span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">-h</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`extract_allele_table`</span> script is below</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/src/pmotools/scripts/extractors_from_pmo/extract_allele_table.py</span></span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/src/pmotools/scripts/extractors_from_pmo/extract_allele_table.py</span></span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>Can download example PMOs here </span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/moz2018_PMO.json.gz </span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a><span class="co"># default </span></span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span></span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a><span class="co"># changing default column names  </span></span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--default_base_col_names</span> sample,target,hapid</span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>Changing the output file delimiter  </span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--delim</span> ,</span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>Adding on additional columns from the specimen_infos section </span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-99"><a href="#cb16-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-100"><a href="#cb16-100" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields </span></span>
<span id="cb16-101"><a href="#cb16-101" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span>  </span>
<span id="cb16-102"><a href="#cb16-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-103"><a href="#cb16-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-104"><a href="#cb16-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-105"><a href="#cb16-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, echo = F}</span></span>
<span id="cb16-106"><a href="#cb16-106" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-107"><a href="#cb16-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-108"><a href="#cb16-108" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields </span></span>
<span id="cb16-109"><a href="#cb16-109" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="dv">2</span><span class="op">&gt;</span>/dev/null <span class="kw">|</span> <span class="fu">head</span>  </span>
<span id="cb16-110"><a href="#cb16-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-111"><a href="#cb16-111" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-112"><a href="#cb16-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-113"><a href="#cb16-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-114"><a href="#cb16-114" aria-hidden="true" tabindex="-1"></a>Can continue to add on more columns from other sections </span>
<span id="cb16-115"><a href="#cb16-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-116"><a href="#cb16-116" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb16-117"><a href="#cb16-117" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-118"><a href="#cb16-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-119"><a href="#cb16-119" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields including seq field </span></span>
<span id="cb16-120"><a href="#cb16-120" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="at">--representative_haps_fields</span> seq <span class="kw">|</span> <span class="fu">head</span> </span>
<span id="cb16-121"><a href="#cb16-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-122"><a href="#cb16-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-123"><a href="#cb16-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-124"><a href="#cb16-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, echo = F}</span></span>
<span id="cb16-125"><a href="#cb16-125" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-126"><a href="#cb16-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-127"><a href="#cb16-127" aria-hidden="true" tabindex="-1"></a><span class="co"># adding other PMO fields including seq field </span></span>
<span id="cb16-128"><a href="#cb16-128" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="at">--specimen_info_meta_fields</span> collection_country,collection_date <span class="at">--representative_haps_fields</span> seq <span class="dv">2</span><span class="op">&gt;</span>/dev/null <span class="kw">|</span> <span class="fu">head</span> </span>
<span id="cb16-129"><a href="#cb16-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-130"><a href="#cb16-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-131"><a href="#cb16-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-132"><a href="#cb16-132" aria-hidden="true" tabindex="-1"></a><span class="fu">## Creating output for MOIRE </span></span>
<span id="cb16-133"><a href="#cb16-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-134"><a href="#cb16-134" aria-hidden="true" tabindex="-1"></a>MOIRE is a program that can be used to estimate COI and other population estimates from a population. See it's <span class="co">[</span><span class="ot">github</span><span class="co">](https://github.com/EPPIcenter/moire)</span> for full usage. </span>
<span id="cb16-135"><a href="#cb16-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-136"><a href="#cb16-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb16-137"><a href="#cb16-137" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb16-138"><a href="#cb16-138" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-139"><a href="#cb16-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-140"><a href="#cb16-140" aria-hidden="true" tabindex="-1"></a><span class="co"># default table is all moire needs </span></span>
<span id="cb16-141"><a href="#cb16-141" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span></span>
<span id="cb16-142"><a href="#cb16-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-143"><a href="#cb16-143" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-144"><a href="#cb16-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-145"><a href="#cb16-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = F}</span></span>
<span id="cb16-146"><a href="#cb16-146" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"example/extraction_allele_table.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb16-147"><a href="#cb16-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-148"><a href="#cb16-148" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">load_long_form_data</span>(df)</span>
<span id="cb16-149"><a href="#cb16-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-150"><a href="#cb16-150" aria-hidden="true" tabindex="-1"></a><span class="co"># With data in appropriate format, run MCMC as follows</span></span>
<span id="cb16-151"><a href="#cb16-151" aria-hidden="true" tabindex="-1"></a>mcmc_results <span class="ot">&lt;-</span> moire<span class="sc">::</span><span class="fu">run_mcmc</span>(data, <span class="at">is_missing =</span> data<span class="sc">$</span>is_missing)</span>
<span id="cb16-152"><a href="#cb16-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-153"><a href="#cb16-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-154"><a href="#cb16-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-155"><a href="#cb16-155" aria-hidden="true" tabindex="-1"></a><span class="fu">## Creating output for dcifer </span></span>
<span id="cb16-156"><a href="#cb16-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-157"><a href="#cb16-157" aria-hidden="true" tabindex="-1"></a>dcifer is a program that can estimate IBD even from mixed infections. See it's <span class="co">[</span><span class="ot">github</span><span class="co">](https://github.com/EPPIcenter/dcifer)</span> for full usage</span>
<span id="cb16-158"><a href="#cb16-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-161"><a href="#cb16-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb16-162"><a href="#cb16-162" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> example</span>
<span id="cb16-163"><a href="#cb16-163" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-164"><a href="#cb16-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-165"><a href="#cb16-165" aria-hidden="true" tabindex="-1"></a><span class="co"># default </span></span>
<span id="cb16-166"><a href="#cb16-166" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--delim</span> ,</span>
<span id="cb16-167"><a href="#cb16-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-168"><a href="#cb16-168" aria-hidden="true" tabindex="-1"></a><span class="co"># dcifer can calculate allele frequencies if not provided or you can have extract_allele_table write them as well </span></span>
<span id="cb16-169"><a href="#cb16-169" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-python</span> extract_allele_table <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> extraction <span class="at">--overwrite</span> <span class="at">--allele_freqs_output</span> allele_freqs_extraction <span class="at">--delim</span> ,</span>
<span id="cb16-170"><a href="#cb16-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-171"><a href="#cb16-171" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-172"><a href="#cb16-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-173"><a href="#cb16-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = F}</span></span>
<span id="cb16-174"><a href="#cb16-174" aria-hidden="true" tabindex="-1"></a>dsmp <span class="ot">&lt;-</span> <span class="fu">readDat</span>(<span class="st">"example/extraction_allele_table.csv"</span>, <span class="at">svar =</span> <span class="st">"sampleID"</span>, <span class="at">lvar =</span> <span class="st">"locus"</span>, <span class="at">avar =</span> <span class="st">"allele"</span>)</span>
<span id="cb16-175"><a href="#cb16-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-176"><a href="#cb16-176" aria-hidden="true" tabindex="-1"></a>lrank <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb16-177"><a href="#cb16-177" aria-hidden="true" tabindex="-1"></a>coi   <span class="ot">&lt;-</span> <span class="fu">getCOI</span>(dsmp, <span class="at">lrank =</span> lrank)</span>
<span id="cb16-178"><a href="#cb16-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-179"><a href="#cb16-179" aria-hidden="true" tabindex="-1"></a>afreq <span class="ot">&lt;-</span> <span class="fu">calcAfreq</span>(dsmp, coi, <span class="at">tol =</span> <span class="fl">1e-5</span>) </span>
<span id="cb16-180"><a href="#cb16-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-181"><a href="#cb16-181" aria-hidden="true" tabindex="-1"></a>dres0 <span class="ot">&lt;-</span> <span class="fu">ibdDat</span>(dsmp, coi, afreq, <span class="at">pval =</span> <span class="cn">TRUE</span>, <span class="at">confint =</span> <span class="cn">TRUE</span>, <span class="at">rnull =</span> <span class="dv">0</span>, </span>
<span id="cb16-182"><a href="#cb16-182" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.05</span>, <span class="at">nr =</span> <span class="fl">1e3</span>)   </span>
<span id="cb16-183"><a href="#cb16-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-184"><a href="#cb16-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-185"><a href="#cb16-185" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb16-186"><a href="#cb16-186" aria-hidden="true" tabindex="-1"></a></span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->



<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>