<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Getting basic info out of PMO using pmotools-python â€“ Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-704a3d686fdb3e6b26ef47c77620618f.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-83b1d0b3b459d34b5b0f98adfe5eb835.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>html{ scroll-behavior: smooth; }</style>
<link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">
<li>
    <a class="dropdown-item" href="../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">
<li>
    <a class="dropdown-item" href="../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">
<li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-runner.py</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/pmotools-runner.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-runner.py</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Getting sub info from PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_panel_info_from_pmo.html">
 <span class="dropdown-text">Getting panel info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/build_pmo.html">
 <span class="dropdown-text">Creating a PMO File</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">
<li>
    <a class="dropdown-item" href="../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Contents</h2>
   
  <ul>
<li>
<a href="#extract-basic-info-counts-from-pmo" id="toc-extract-basic-info-counts-from-pmo" class="nav-link active" data-scroll-target="#extract-basic-info-counts-from-pmo">Extract basic info counts from PMO</a>
  <ul>
<li><a href="#list_specimen_meta_fields" id="toc-list_specimen_meta_fields" class="nav-link" data-scroll-target="#list_specimen_meta_fields">list_specimen_meta_fields</a></li>
  <li><a href="#count_specimen_meta" id="toc-count_specimen_meta" class="nav-link" data-scroll-target="#count_specimen_meta">count_specimen_meta</a></li>
  <li><a href="#list_tar_amp_bioinformatics_info_ids" id="toc-list_tar_amp_bioinformatics_info_ids" class="nav-link" data-scroll-target="#list_tar_amp_bioinformatics_info_ids">list_tar_amp_bioinformatics_info_ids</a></li>
  <li><a href="#count_targets_per_sample" id="toc-count_targets_per_sample" class="nav-link" data-scroll-target="#count_targets_per_sample">count_targets_per_sample</a></li>
  <li><a href="#count_samples_per_target" id="toc-count_samples_per_target" class="nav-link" data-scroll-target="#count_samples_per_target">count_samples_per_target</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Getting basic info out of PMO using pmotools-python</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-page-left">

    
  
    
  </div>
  


</header><section id="extract-basic-info-counts-from-pmo" class="level1"><h1>Extract basic info counts from PMO</h1>
<p>To get simple counts of number of targets with sample counts, samples with target counts, the counts of meta fields</p>
<p>Most of these basic info extractor can be found underneath <code>extract_basic_info_from_pmo</code></p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>pmotools v1.0.0 - A suite of tools for interacting with Portable Microhaplotype Object (pmo) file format

Available functions organized by groups are
convertors_to_json
    text_meta_to_json_meta - Convert text file meta to JSON Meta
    excel_meta_to_json_meta - Convert excel file meta to JSON Meta
    microhaplotype_table_to_json_file - Convert microhaplotype table to JSON Meta
    terra_amp_output_to_json - Convert terra output table to JSON seq table

extractors_from_pmo
    extract_pmo_with_selected_meta - Extract from PMO samples and associated haplotypes with selected meta
    extract_pmo_with_select_specimen_ids - Extract from PMO specific samples from the specimens table
    extract_pmo_with_select_experiment_sample_ids - Extract from PMO specific experiment sample ids from the experiment_info table
    extract_pmo_with_select_targets - Extract from PMO specific targets
    extract_pmo_with_read_filter - Extract from PMO with a read filter
    extract_allele_table - Extract allele tables which can be as used as input to such tools as dcifer or moire

working_with_multiple_pmos
    combine_pmos - Combine multiple pmos of the same panel into a single pmo

extract_basic_info_from_pmo
    list_experiment_sample_ids_per_specimen_id - Each specimen_id can have multiple experiment_sample_ids, list out all in a PMO
    list_specimen_meta_fields - List out the specimen meta fields in the specimen_info section
    list_tar_amp_bioinformatics_info_ids - List out all the tar_amp_bioinformatics_info_ids in a PMO file
    count_specimen_meta - Count the values of specific specimen meta fields in the specimen_info section
    count_targets_per_sample - Count the number of targets per sample
    count_samples_per_target - Count the number of samples per target

extract_panel_info_from_pmo
    extract_insert_of_panels - Extract the insert of panels from a PMO
    extract_refseq_of_inserts_of_panels - Extract the ref_seq of panels from a PMO</code></pre>
</div>
</div>
<p>Getting files for examples</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/moz2018_PMO.json.gz</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="list_specimen_meta_fields" class="level2"><h2 class="anchored" data-anchor-id="list_specimen_meta_fields">list_specimen_meta_fields</h2>
<p>This will list all the meta fields within the <code>specimen_infos</code> section of a PMO file. Since not all meta fields are always present in all specimens, this will list the count of samples each field appears in and the number of total specimens</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py list_specimen_meta_fields [-h] --file FILE
                                                    [--output OUTPUT]
                                                    [--delim DELIM]
                                                    [--overwrite]

options:
  -h, --help       show this help message and exit
  --file FILE      PMO file
  --output OUTPUT  output file
  --delim DELIM    the delimiter of the output text file, examples input
                   tab,comma but can also be the actual delimiter
  --overwrite      If output file exists, overwrite it</code></pre>
</div>
</div>
<p>The python code for <code>list_specimen_meta_fields</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extract_info_from_pmo/list_specimen_meta_fields.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="im">import</span> sys</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb6-5"><a href="#cb6-5"></a></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb6-7"><a href="#cb6-7"></a></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb6-11"><a href="#cb6-11"></a></span>
<span id="cb6-12"><a href="#cb6-12"></a></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="kw">def</span> parse_args_list_specimen_meta_fields():</span>
<span id="cb6-14"><a href="#cb6-14"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb6-15"><a href="#cb6-15"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb6-16"><a href="#cb6-16"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, default<span class="op">=</span><span class="st">"STDOUT"</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'output file'</span>)</span>
<span id="cb6-17"><a href="#cb6-17"></a>    parser.add_argument(<span class="st">'--delim'</span>, default<span class="op">=</span><span class="st">"tab"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the delimiter of the output text file, examples input tab,comma but can also be the actual delimiter'</span>)</span>
<span id="cb6-18"><a href="#cb6-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb6-19"><a href="#cb6-19"></a></span>
<span id="cb6-20"><a href="#cb6-20"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb6-21"><a href="#cb6-21"></a></span>
<span id="cb6-22"><a href="#cb6-22"></a><span class="kw">def</span> list_specimen_meta_fields():</span>
<span id="cb6-23"><a href="#cb6-23"></a>    args <span class="op">=</span> parse_args_list_specimen_meta_fields()</span>
<span id="cb6-24"><a href="#cb6-24"></a></span>
<span id="cb6-25"><a href="#cb6-25"></a>    <span class="co"># check files</span></span>
<span id="cb6-26"><a href="#cb6-26"></a>    output_delim, output_extension <span class="op">=</span> Utils.process_delimiter_and_output_extension(args.delim, gzip<span class="op">=</span>args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb6-27"><a href="#cb6-27"></a>    args.output <span class="op">=</span> args.output <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> Utils.appendStrAsNeeded(args.output, output_extension)</span>
<span id="cb6-28"><a href="#cb6-28"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb6-29"><a href="#cb6-29"></a></span>
<span id="cb6-30"><a href="#cb6-30"></a>    <span class="co"># read in PMO</span></span>
<span id="cb6-31"><a href="#cb6-31"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb6-32"><a href="#cb6-32"></a></span>
<span id="cb6-33"><a href="#cb6-33"></a>    <span class="co"># count fields</span></span>
<span id="cb6-34"><a href="#cb6-34"></a>    counts_df <span class="op">=</span> PMOExtractor.count_specimen_meta_fields(pmo)</span>
<span id="cb6-35"><a href="#cb6-35"></a></span>
<span id="cb6-36"><a href="#cb6-36"></a>    <span class="co"># output</span></span>
<span id="cb6-37"><a href="#cb6-37"></a>    counts_df.to_csv(sys.stdout <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> args.output, sep <span class="op">=</span> output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb6-38"><a href="#cb6-38"></a></span>
<span id="cb6-39"><a href="#cb6-39"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb6-40"><a href="#cb6-40"></a>    list_specimen_meta_fields()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/moz2018_PMO.json.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>field   presentInSpecimensCount totalSpecimenCount
collection_country  124 124
collection_date 124 124
collector   124 124
geo_admin3  124 124
host_taxon_id   124 124
lat_lon 124 124
parasite_density    124 124
plate_col   81  124
plate_name  81  124
plate_row   81  124
project_name    124 124
samp_collect_device 124 124
samp_store_loc  124 124
samp_taxon_id   124 124
specimen_id 124 124</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> spec_fields_moz2018_PMO.tsv <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="count_specimen_meta" class="level2"><h2 class="anchored" data-anchor-id="count_specimen_meta">count_specimen_meta</h2>
<p>This will list all the meta values (and the combinations) for the meta fields within the <code>specimen_infos</code> section of a PMO file.</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py count_specimen_meta [-h] --file FILE
                                              [--output OUTPUT]
                                              [--delim DELIM] [--overwrite]
                                              --meta_fields META_FIELDS

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       output file
  --delim DELIM         the delimiter of the output text file, examples input
                        tab,comma but can also be the actual delimiter
  --overwrite           If output file exists, overwrite it
  --meta_fields META_FIELDS
                        the fields to count the subfields of, can supply
                        multiple separated by commas, e.g. --meta_fields
                        collection_country,collection_date</code></pre>
</div>
</div>
<p>The python code for <code>count_specimen_meta</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extract_info_from_pmo/count_specimen_meta.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="im">import</span> sys</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb12-5"><a href="#cb12-5"></a></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb12-7"><a href="#cb12-7"></a></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb12-11"><a href="#cb12-11"></a></span>
<span id="cb12-12"><a href="#cb12-12"></a></span>
<span id="cb12-13"><a href="#cb12-13"></a><span class="kw">def</span> parse_args_count_specimen_meta():</span>
<span id="cb12-14"><a href="#cb12-14"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb12-15"><a href="#cb12-15"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb12-16"><a href="#cb12-16"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, default<span class="op">=</span><span class="st">"STDOUT"</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'output file'</span>)</span>
<span id="cb12-17"><a href="#cb12-17"></a>    parser.add_argument(<span class="st">'--delim'</span>, default<span class="op">=</span><span class="st">"tab"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the delimiter of the output text file, examples input tab,comma but can also be the actual delimiter'</span>)</span>
<span id="cb12-18"><a href="#cb12-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action<span class="op">=</span><span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb12-19"><a href="#cb12-19"></a>    parser.add_argument(<span class="st">'--meta_fields'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the fields to count the subfields of, can supply multiple separated by commas, e.g. --meta_fields collection_country,collection_date'</span>)</span>
<span id="cb12-20"><a href="#cb12-20"></a></span>
<span id="cb12-21"><a href="#cb12-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb12-22"><a href="#cb12-22"></a></span>
<span id="cb12-23"><a href="#cb12-23"></a></span>
<span id="cb12-24"><a href="#cb12-24"></a><span class="kw">def</span> count_specimen_meta():</span>
<span id="cb12-25"><a href="#cb12-25"></a>    args <span class="op">=</span> parse_args_count_specimen_meta()</span>
<span id="cb12-26"><a href="#cb12-26"></a></span>
<span id="cb12-27"><a href="#cb12-27"></a>    <span class="co"># check files</span></span>
<span id="cb12-28"><a href="#cb12-28"></a>    output_delim, output_extension <span class="op">=</span> Utils.process_delimiter_and_output_extension(args.delim, gzip<span class="op">=</span>args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb12-29"><a href="#cb12-29"></a>    args.output <span class="op">=</span> args.output <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> Utils.appendStrAsNeeded(args.output, output_extension)</span>
<span id="cb12-30"><a href="#cb12-30"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb12-31"><a href="#cb12-31"></a></span>
<span id="cb12-32"><a href="#cb12-32"></a>    <span class="co"># process the meta_fields argument</span></span>
<span id="cb12-33"><a href="#cb12-33"></a>    meta_fields_toks <span class="op">=</span> args.meta_fields.split(<span class="st">','</span>)</span>
<span id="cb12-34"><a href="#cb12-34"></a></span>
<span id="cb12-35"><a href="#cb12-35"></a>    <span class="co"># read in PMO</span></span>
<span id="cb12-36"><a href="#cb12-36"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb12-37"><a href="#cb12-37"></a></span>
<span id="cb12-38"><a href="#cb12-38"></a>    <span class="co"># count sub-fields</span></span>
<span id="cb12-39"><a href="#cb12-39"></a>    counts_df <span class="op">=</span> PMOExtractor.count_specimen_meta_subfields(pmo, meta_fields_toks)</span>
<span id="cb12-40"><a href="#cb12-40"></a></span>
<span id="cb12-41"><a href="#cb12-41"></a>    <span class="co">#write out</span></span>
<span id="cb12-42"><a href="#cb12-42"></a>    counts_df.to_csv(sys.stdout <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> args.output, sep <span class="op">=</span> output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb12-43"><a href="#cb12-43"></a></span>
<span id="cb12-44"><a href="#cb12-44"></a></span>
<span id="cb12-45"><a href="#cb12-45"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb12-46"><a href="#cb12-46"></a>    count_specimen_meta()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>collection_country  specimensCount  specimensFreq   totalSpecimenCount
Mozambique  81  0.6532258064516129  124
NA  43  0.3467741935483871  124</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country <span class="at">--overwrite</span> <span class="at">--output</span> collection_country_count_moz2018_PMO.tsv.gz </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country,geo_admin3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>collection_country  geo_admin3  specimensCount  specimensFreq   totalSpecimenCount
Mozambique  Inhassoro   27  0.21774193548387097 124
Mozambique  Mandlakazi  28  0.22580645161290322 124
Mozambique  Namaacha    26  0.20967741935483872 124
NA  NA  43  0.3467741935483871  124</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>collection_country  collection_date specimensCount  specimensFreq   totalSpecimenCount
Bangladesh  2008    15  0.0007718828796377296   19433
Bangladesh  2009    16  0.000823341738280245    19433
Bangladesh  2012    51  0.002624401790768281    19433
Bangladesh  2015    508 0.026141100190397778    19433
Bangladesh  2016    816 0.041990428652292494    19433
Bangladesh  2017    12  0.0006175063037101837   19433
Benin   2014    41  0.002109813204343128    19433
Benin   2016    117 0.006020686461174291    19433
Brazil  1980    1   5.145885864251531e-05   19433</code></pre>
</div>
</div>
</section><section id="list_tar_amp_bioinformatics_info_ids" class="level2"><h2 class="anchored" data-anchor-id="list_tar_amp_bioinformatics_info_ids">list_tar_amp_bioinformatics_info_ids</h2>
<p>This will simply list out all the analyses (all the <code>tar_amp_bioinformatics_info_id</code>s) stored within a PMO</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py list_tar_amp_bioinformatics_info_ids
       [-h] --file FILE [--output OUTPUT] [--overwrite]

options:
  -h, --help       show this help message and exit
  --file FILE      PMO file
  --output OUTPUT  output file
  --overwrite      If output file exists, overwrite it</code></pre>
</div>
</div>
<p>The python code for <code>list_tar_amp_bioinformatics_info_ids</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extract_info_from_pmo/list_tar_amp_bioinformatics_info_ids.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb22-2"><a href="#cb22-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb22-3"><a href="#cb22-3"></a><span class="im">import</span> sys</span>
<span id="cb22-4"><a href="#cb22-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb22-5"><a href="#cb22-5"></a></span>
<span id="cb22-6"><a href="#cb22-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb22-7"><a href="#cb22-7"></a></span>
<span id="cb22-8"><a href="#cb22-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb22-9"><a href="#cb22-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb22-10"><a href="#cb22-10"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb22-11"><a href="#cb22-11"></a></span>
<span id="cb22-12"><a href="#cb22-12"></a></span>
<span id="cb22-13"><a href="#cb22-13"></a><span class="kw">def</span> parse_args_list_tar_amp_bioinformatics_info_ids():</span>
<span id="cb22-14"><a href="#cb22-14"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb22-15"><a href="#cb22-15"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb22-16"><a href="#cb22-16"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, default<span class="op">=</span><span class="st">"STDOUT"</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'output file'</span>)</span>
<span id="cb22-17"><a href="#cb22-17"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb22-18"><a href="#cb22-18"></a></span>
<span id="cb22-19"><a href="#cb22-19"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb22-20"><a href="#cb22-20"></a></span>
<span id="cb22-21"><a href="#cb22-21"></a><span class="kw">def</span> list_tar_amp_bioinformatics_info_ids():</span>
<span id="cb22-22"><a href="#cb22-22"></a>    args <span class="op">=</span> parse_args_list_tar_amp_bioinformatics_info_ids()</span>
<span id="cb22-23"><a href="#cb22-23"></a></span>
<span id="cb22-24"><a href="#cb22-24"></a>    <span class="co"># check files</span></span>
<span id="cb22-25"><a href="#cb22-25"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb22-26"><a href="#cb22-26"></a></span>
<span id="cb22-27"><a href="#cb22-27"></a>    <span class="co"># read in PMO</span></span>
<span id="cb22-28"><a href="#cb22-28"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb22-29"><a href="#cb22-29"></a></span>
<span id="cb22-30"><a href="#cb22-30"></a>    <span class="co"># extract all taramp_bioinformatics_ids</span></span>
<span id="cb22-31"><a href="#cb22-31"></a>    bioids <span class="op">=</span> pmo[<span class="st">"taramp_bioinformatics_infos"</span>].keys()</span>
<span id="cb22-32"><a href="#cb22-32"></a></span>
<span id="cb22-33"><a href="#cb22-33"></a>    <span class="co"># write</span></span>
<span id="cb22-34"><a href="#cb22-34"></a>    output_target <span class="op">=</span> sys.stdout <span class="cf">if</span> args.output <span class="op">==</span> <span class="st">"STDOUT"</span> <span class="cf">else</span> <span class="bu">open</span>(args.output, <span class="st">"w"</span>)</span>
<span id="cb22-35"><a href="#cb22-35"></a>    <span class="cf">with</span> output_target <span class="im">as</span> f:</span>
<span id="cb22-36"><a href="#cb22-36"></a>        f.write(<span class="st">"</span><span class="ch">\n</span><span class="st">"</span>.join(bioids) <span class="op">+</span> <span class="st">"</span><span class="ch">\n</span><span class="st">"</span>)</span>
<span id="cb22-37"><a href="#cb22-37"></a></span>
<span id="cb22-38"><a href="#cb22-38"></a></span>
<span id="cb22-39"><a href="#cb22-39"></a></span>
<span id="cb22-40"><a href="#cb22-40"></a></span>
<span id="cb22-41"><a href="#cb22-41"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb22-42"><a href="#cb22-42"></a>    list_tar_amp_bioinformatics_info_ids()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Mozambique2018-SeekDeep</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>PathWeaverHeome1</code></pre>
</div>
</div>
<p>This can be helpful after combining PMOs</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> combine_pmos <span class="at">--pmo_files</span> ../../format/moz2018_PMO.json.gz,../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--output</span> combined_Heome1_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">--file</span> combined_Heome1_PMO.json.gz </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>Mozambique2018-SeekDeep
PathWeaverHeome1</code></pre>
</div>
</div>
</section><section id="count_targets_per_sample" class="level2"><h2 class="anchored" data-anchor-id="count_targets_per_sample">count_targets_per_sample</h2>
<p>Count up the number targets each experimental_sample_id has. A read filter can be applied to see how targets would be kept if such a filter was applied</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py count_targets_per_sample [-h] --file FILE
                                                   [--output OUTPUT]
                                                   [--delim DELIM]
                                                   [--overwrite]
                                                   [--read_count_minimum READ_COUNT_MINIMUM]

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       output file
  --delim DELIM         the delimiter of the output text file, examples input
                        tab,comma but can also be the actual delimiter
  --overwrite           If output file exists, overwrite it
  --read_count_minimum READ_COUNT_MINIMUM
                        the minimum read count (inclusive) to be counted as
                        covered by sample</code></pre>
</div>
</div>
<p>The python code for <code>count_targets_per_sample</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extract_info_from_pmo/count_targets_per_sample.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb31-2"><a href="#cb31-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb31-3"><a href="#cb31-3"></a><span class="im">import</span> sys</span>
<span id="cb31-4"><a href="#cb31-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb31-5"><a href="#cb31-5"></a></span>
<span id="cb31-6"><a href="#cb31-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb31-7"><a href="#cb31-7"></a></span>
<span id="cb31-8"><a href="#cb31-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb31-9"><a href="#cb31-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb31-10"><a href="#cb31-10"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb31-11"><a href="#cb31-11"></a></span>
<span id="cb31-12"><a href="#cb31-12"></a></span>
<span id="cb31-13"><a href="#cb31-13"></a><span class="kw">def</span> parse_args_count_targets_per_sample():</span>
<span id="cb31-14"><a href="#cb31-14"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb31-15"><a href="#cb31-15"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb31-16"><a href="#cb31-16"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, default<span class="op">=</span><span class="st">"STDOUT"</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'output file'</span>)</span>
<span id="cb31-17"><a href="#cb31-17"></a>    parser.add_argument(<span class="st">'--delim'</span>, default<span class="op">=</span><span class="st">"tab"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the delimiter of the output text file, examples input tab,comma but can also be the actual delimiter'</span>)</span>
<span id="cb31-18"><a href="#cb31-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action<span class="op">=</span><span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb31-19"><a href="#cb31-19"></a>    parser.add_argument(<span class="st">'--read_count_minimum'</span>, default<span class="op">=</span><span class="fl">0.0</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">float</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the minimum read count (inclusive) to be counted as covered by sample'</span>)</span>
<span id="cb31-20"><a href="#cb31-20"></a></span>
<span id="cb31-21"><a href="#cb31-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb31-22"><a href="#cb31-22"></a></span>
<span id="cb31-23"><a href="#cb31-23"></a></span>
<span id="cb31-24"><a href="#cb31-24"></a><span class="kw">def</span> count_targets_per_sample():</span>
<span id="cb31-25"><a href="#cb31-25"></a>    args <span class="op">=</span> parse_args_count_targets_per_sample()</span>
<span id="cb31-26"><a href="#cb31-26"></a></span>
<span id="cb31-27"><a href="#cb31-27"></a>    <span class="co"># check files</span></span>
<span id="cb31-28"><a href="#cb31-28"></a>    output_delim, output_extension <span class="op">=</span> Utils.process_delimiter_and_output_extension(args.delim, gzip<span class="op">=</span>args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb31-29"><a href="#cb31-29"></a>    args.output <span class="op">=</span> args.output <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> Utils.appendStrAsNeeded(args.output, output_extension)</span>
<span id="cb31-30"><a href="#cb31-30"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb31-31"><a href="#cb31-31"></a></span>
<span id="cb31-32"><a href="#cb31-32"></a>    <span class="co"># read in PMO</span></span>
<span id="cb31-33"><a href="#cb31-33"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb31-34"><a href="#cb31-34"></a></span>
<span id="cb31-35"><a href="#cb31-35"></a>    <span class="co"># count</span></span>
<span id="cb31-36"><a href="#cb31-36"></a>    counts_df <span class="op">=</span> PMOExtractor.count_targets_per_sample(pmo, args.read_count_minimum)</span>
<span id="cb31-37"><a href="#cb31-37"></a></span>
<span id="cb31-38"><a href="#cb31-38"></a>    <span class="co">#write out</span></span>
<span id="cb31-39"><a href="#cb31-39"></a>    counts_df.to_csv(sys.stdout <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> args.output, sep <span class="op">=</span> output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb31-40"><a href="#cb31-40"></a></span>
<span id="cb31-41"><a href="#cb31-41"></a></span>
<span id="cb31-42"><a href="#cb31-42"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb31-43"><a href="#cb31-43"></a>    count_targets_per_sample()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>tar_amp_bioinformatics_info_id  experiment_sample_id    target_number
Mozambique2018-SeekDeep 8025874217  99
Mozambique2018-SeekDeep 8025874231  99
Mozambique2018-SeekDeep 8025874234  97
Mozambique2018-SeekDeep 8025874237  99
Mozambique2018-SeekDeep 8025874250  98
Mozambique2018-SeekDeep 8025874253  99
Mozambique2018-SeekDeep 8025874261  99
Mozambique2018-SeekDeep 8025874266  85
Mozambique2018-SeekDeep 8025874271  99</code></pre>
</div>
</div>
<p>Apply a read count minimum filter (this a total read count summed for a target and not on a haplotype level)</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">--read_count_minimum</span> 3000 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>tar_amp_bioinformatics_info_id  experiment_sample_id    target_number
Mozambique2018-SeekDeep 8025874217  99
Mozambique2018-SeekDeep 8025874231  73
Mozambique2018-SeekDeep 8025874234  93
Mozambique2018-SeekDeep 8025874237  98
Mozambique2018-SeekDeep 8025874250  68
Mozambique2018-SeekDeep 8025874253  99
Mozambique2018-SeekDeep 8025874261  98
Mozambique2018-SeekDeep 8025874266  37
Mozambique2018-SeekDeep 8025874271  98</code></pre>
</div>
</div>
</section><section id="count_samples_per_target" class="level2"><h2 class="anchored" data-anchor-id="count_samples_per_target">count_samples_per_target</h2>
<p>Count up the number of experimental_sample_ids each target has. A read filter can be applied to see how many samples a taget would have if a filter was applied</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py count_samples_per_target [-h] --file FILE
                                                   [--output OUTPUT]
                                                   [--delim DELIM]
                                                   [--overwrite]
                                                   [--read_count_minimum READ_COUNT_MINIMUM]

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       output file
  --delim DELIM         the delimiter of the output text file, examples input
                        tab,comma but can also be the actual delimiter
  --overwrite           If output file exists, overwrite it
  --read_count_minimum READ_COUNT_MINIMUM
                        the minimum read count (inclusive) to be counted as
                        covered by sample</code></pre>
</div>
</div>
<p>The python code for <code>count_samples_per_target</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extract_info_from_pmo/count_samples_per_target.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb38-2"><a href="#cb38-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb38-3"><a href="#cb38-3"></a><span class="im">import</span> sys</span>
<span id="cb38-4"><a href="#cb38-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb38-5"><a href="#cb38-5"></a></span>
<span id="cb38-6"><a href="#cb38-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb38-7"><a href="#cb38-7"></a></span>
<span id="cb38-8"><a href="#cb38-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb38-9"><a href="#cb38-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb38-10"><a href="#cb38-10"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb38-11"><a href="#cb38-11"></a></span>
<span id="cb38-12"><a href="#cb38-12"></a></span>
<span id="cb38-13"><a href="#cb38-13"></a><span class="kw">def</span> parse_args_count_samples_per_target():</span>
<span id="cb38-14"><a href="#cb38-14"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb38-15"><a href="#cb38-15"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb38-16"><a href="#cb38-16"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, default<span class="op">=</span><span class="st">"STDOUT"</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'output file'</span>)</span>
<span id="cb38-17"><a href="#cb38-17"></a>    parser.add_argument(<span class="st">'--delim'</span>, default<span class="op">=</span><span class="st">"tab"</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the delimiter of the output text file, examples input tab,comma but can also be the actual delimiter'</span>)</span>
<span id="cb38-18"><a href="#cb38-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action<span class="op">=</span><span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb38-19"><a href="#cb38-19"></a>    parser.add_argument(<span class="st">'--read_count_minimum'</span>, default<span class="op">=</span><span class="fl">0.0</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">float</span>, required<span class="op">=</span><span class="va">False</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'the minimum read count (inclusive) to be counted as covered by sample'</span>)</span>
<span id="cb38-20"><a href="#cb38-20"></a></span>
<span id="cb38-21"><a href="#cb38-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb38-22"><a href="#cb38-22"></a></span>
<span id="cb38-23"><a href="#cb38-23"></a></span>
<span id="cb38-24"><a href="#cb38-24"></a><span class="kw">def</span> count_samples_per_target():</span>
<span id="cb38-25"><a href="#cb38-25"></a>    args <span class="op">=</span> parse_args_count_samples_per_target()</span>
<span id="cb38-26"><a href="#cb38-26"></a></span>
<span id="cb38-27"><a href="#cb38-27"></a>    <span class="co"># check files</span></span>
<span id="cb38-28"><a href="#cb38-28"></a>    output_delim, output_extension <span class="op">=</span> Utils.process_delimiter_and_output_extension(args.delim, gzip<span class="op">=</span>args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb38-29"><a href="#cb38-29"></a>    args.output <span class="op">=</span> args.output <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> Utils.appendStrAsNeeded(args.output, output_extension)</span>
<span id="cb38-30"><a href="#cb38-30"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb38-31"><a href="#cb38-31"></a></span>
<span id="cb38-32"><a href="#cb38-32"></a>    <span class="co"># read in PMO</span></span>
<span id="cb38-33"><a href="#cb38-33"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb38-34"><a href="#cb38-34"></a></span>
<span id="cb38-35"><a href="#cb38-35"></a>    <span class="co"># count</span></span>
<span id="cb38-36"><a href="#cb38-36"></a>    counts_df <span class="op">=</span> PMOExtractor.count_samples_per_target(pmo, args.read_count_minimum)</span>
<span id="cb38-37"><a href="#cb38-37"></a></span>
<span id="cb38-38"><a href="#cb38-38"></a>    <span class="co">#write out</span></span>
<span id="cb38-39"><a href="#cb38-39"></a>    counts_df.to_csv(sys.stdout <span class="cf">if</span> <span class="st">"STDOUT"</span> <span class="op">==</span> args.output <span class="cf">else</span> args.output, sep <span class="op">=</span> output_delim, index<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb38-40"><a href="#cb38-40"></a></span>
<span id="cb38-41"><a href="#cb38-41"></a></span>
<span id="cb38-42"><a href="#cb38-42"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb38-43"><a href="#cb38-43"></a>    count_samples_per_target()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>tar_amp_bioinformatics_info_id  target_id   sample_number
Mozambique2018-SeekDeep t1  119
Mozambique2018-SeekDeep t10 117
Mozambique2018-SeekDeep t100    124
Mozambique2018-SeekDeep t11 120
Mozambique2018-SeekDeep t12 119
Mozambique2018-SeekDeep t13 124
Mozambique2018-SeekDeep t14 118
Mozambique2018-SeekDeep t15 119
Mozambique2018-SeekDeep t16 121</code></pre>
</div>
</div>
<p>Apply a read count minimum filter (this a total read count summed for a target and not on a haplotype level)</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">--read_count_minimum</span> 3000 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>tar_amp_bioinformatics_info_id  target_id   sample_number
Mozambique2018-SeekDeep t1  108
Mozambique2018-SeekDeep t10 107
Mozambique2018-SeekDeep t100    107
Mozambique2018-SeekDeep t11 111
Mozambique2018-SeekDeep t12 104
Mozambique2018-SeekDeep t13 105
Mozambique2018-SeekDeep t14 110
Mozambique2018-SeekDeep t15 110
Mozambique2018-SeekDeep t16 106</code></pre>
</div>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb43" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Getting basic info out of PMO using pmotools-python</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, echo=F}</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="in">source("../common.R")</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Extract basic info counts from PMO </span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>To get simple counts of number of targets with sample counts, samples with target counts, the counts of meta fields </span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>Most of these basic info extractor can be found underneath <span class="in">`extract_basic_info_from_pmo`</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, echo = F}</span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py | perl -pe 's/\e\[[0-9;]*m(?:\e\[K)?//g'</span></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>Getting files for examples </span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a><span class="in">cd example </span></span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a><span class="in">wget https://plasmogenepi.github.io/PMO_Docs/format/moz2018_PMO.json.gz</span></span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a><span class="in">wget https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span></span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-38"><a href="#cb43-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-39"><a href="#cb43-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-40"><a href="#cb43-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-41"><a href="#cb43-41" aria-hidden="true" tabindex="-1"></a><span class="fu">## list_specimen_meta_fields</span></span>
<span id="cb43-42"><a href="#cb43-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-43"><a href="#cb43-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-44"><a href="#cb43-44" aria-hidden="true" tabindex="-1"></a>This will list all the meta fields within the <span class="in">`specimen_infos`</span> section of a PMO file. Since not all meta fields are always present in all specimens, this will list the count of samples each field appears in and the number of total specimens  </span>
<span id="cb43-45"><a href="#cb43-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-48"><a href="#cb43-48" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-49"><a href="#cb43-49" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">-h</span> </span>
<span id="cb43-50"><a href="#cb43-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-51"><a href="#cb43-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-52"><a href="#cb43-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-53"><a href="#cb43-53" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`list_specimen_meta_fields`</span> script is below</span>
<span id="cb43-54"><a href="#cb43-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-57"><a href="#cb43-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb43-58"><a href="#cb43-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb43-59"><a href="#cb43-59" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb43-60"><a href="#cb43-60" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb43-61"><a href="#cb43-61" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb43-62"><a href="#cb43-62" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extract_info_from_pmo/list_specimen_meta_fields.py</span></span>
<span id="cb43-63"><a href="#cb43-63" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extract_info_from_pmo/list_specimen_meta_fields.py</span></span>
<span id="cb43-64"><a href="#cb43-64" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-65"><a href="#cb43-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-68"><a href="#cb43-68" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-69"><a href="#cb43-69" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-70"><a href="#cb43-70" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/moz2018_PMO.json.gz</span>
<span id="cb43-71"><a href="#cb43-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-72"><a href="#cb43-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-73"><a href="#cb43-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-76"><a href="#cb43-76" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-77"><a href="#cb43-77" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-78"><a href="#cb43-78" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> spec_fields_moz2018_PMO.tsv <span class="at">--overwrite</span></span>
<span id="cb43-79"><a href="#cb43-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-80"><a href="#cb43-80" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-81"><a href="#cb43-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-82"><a href="#cb43-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-83"><a href="#cb43-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-84"><a href="#cb43-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-85"><a href="#cb43-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## count_specimen_meta</span></span>
<span id="cb43-86"><a href="#cb43-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-87"><a href="#cb43-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-88"><a href="#cb43-88" aria-hidden="true" tabindex="-1"></a>This will list all the meta values (and the combinations) for the meta fields within the <span class="in">`specimen_infos`</span> section of a PMO file.</span>
<span id="cb43-89"><a href="#cb43-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-90"><a href="#cb43-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-93"><a href="#cb43-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-94"><a href="#cb43-94" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">-h</span> </span>
<span id="cb43-95"><a href="#cb43-95" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-96"><a href="#cb43-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-97"><a href="#cb43-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-98"><a href="#cb43-98" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`count_specimen_meta`</span> script is below</span>
<span id="cb43-99"><a href="#cb43-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-102"><a href="#cb43-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb43-103"><a href="#cb43-103" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb43-104"><a href="#cb43-104" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb43-105"><a href="#cb43-105" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb43-106"><a href="#cb43-106" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb43-107"><a href="#cb43-107" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extract_info_from_pmo/count_specimen_meta.py</span></span>
<span id="cb43-108"><a href="#cb43-108" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extract_info_from_pmo/count_specimen_meta.py</span></span>
<span id="cb43-109"><a href="#cb43-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-110"><a href="#cb43-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-111"><a href="#cb43-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-114"><a href="#cb43-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-115"><a href="#cb43-115" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-116"><a href="#cb43-116" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country</span>
<span id="cb43-117"><a href="#cb43-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-118"><a href="#cb43-118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-119"><a href="#cb43-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-122"><a href="#cb43-122" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-123"><a href="#cb43-123" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-124"><a href="#cb43-124" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country <span class="at">--overwrite</span> <span class="at">--output</span> collection_country_count_moz2018_PMO.tsv.gz </span>
<span id="cb43-125"><a href="#cb43-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-126"><a href="#cb43-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-127"><a href="#cb43-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-130"><a href="#cb43-130" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-131"><a href="#cb43-131" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-132"><a href="#cb43-132" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--meta_fields</span> collection_country,geo_admin3</span>
<span id="cb43-133"><a href="#cb43-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-134"><a href="#cb43-134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-135"><a href="#cb43-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-138"><a href="#cb43-138" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-139"><a href="#cb43-139" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-140"><a href="#cb43-140" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span></span>
<span id="cb43-141"><a href="#cb43-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-142"><a href="#cb43-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-143"><a href="#cb43-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-144"><a href="#cb43-144" aria-hidden="true" tabindex="-1"></a><span class="fu">## list_tar_amp_bioinformatics_info_ids </span></span>
<span id="cb43-145"><a href="#cb43-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-146"><a href="#cb43-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-147"><a href="#cb43-147" aria-hidden="true" tabindex="-1"></a>This will simply list out all the analyses (all the <span class="in">`tar_amp_bioinformatics_info_id`</span>s) stored within a PMO </span>
<span id="cb43-148"><a href="#cb43-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-149"><a href="#cb43-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-152"><a href="#cb43-152" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-153"><a href="#cb43-153" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">-h</span> </span>
<span id="cb43-154"><a href="#cb43-154" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-155"><a href="#cb43-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-156"><a href="#cb43-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-157"><a href="#cb43-157" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`list_tar_amp_bioinformatics_info_ids`</span> script is below</span>
<span id="cb43-158"><a href="#cb43-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-161"><a href="#cb43-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb43-162"><a href="#cb43-162" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb43-163"><a href="#cb43-163" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb43-164"><a href="#cb43-164" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb43-165"><a href="#cb43-165" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb43-166"><a href="#cb43-166" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extract_info_from_pmo/list_tar_amp_bioinformatics_info_ids.py</span></span>
<span id="cb43-167"><a href="#cb43-167" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extract_info_from_pmo/list_tar_amp_bioinformatics_info_ids.py</span></span>
<span id="cb43-168"><a href="#cb43-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-169"><a href="#cb43-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-170"><a href="#cb43-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-171"><a href="#cb43-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-174"><a href="#cb43-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-175"><a href="#cb43-175" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-176"><a href="#cb43-176" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  </span>
<span id="cb43-177"><a href="#cb43-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-178"><a href="#cb43-178" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-179"><a href="#cb43-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-182"><a href="#cb43-182" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-183"><a href="#cb43-183" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-184"><a href="#cb43-184" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_tar_amp_bioinformatics_info_ids <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz </span>
<span id="cb43-185"><a href="#cb43-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-186"><a href="#cb43-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-187"><a href="#cb43-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-188"><a href="#cb43-188" aria-hidden="true" tabindex="-1"></a>This can be helpful after combining PMOs </span>
<span id="cb43-189"><a href="#cb43-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-190"><a href="#cb43-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-191"><a href="#cb43-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb43-192"><a href="#cb43-192" aria-hidden="true" tabindex="-1"></a><span class="in">cd example </span></span>
<span id="cb43-193"><a href="#cb43-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-194"><a href="#cb43-194" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py combine_pmos --pmo_files ../../format/moz2018_PMO.json.gz,../../format/PathWeaverHeome1_PMO.json.gz --output combined_Heome1_PMO.json.gz --overwrite</span></span>
<span id="cb43-195"><a href="#cb43-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-196"><a href="#cb43-196" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py list_tar_amp_bioinformatics_info_ids --file combined_Heome1_PMO.json.gz </span></span>
<span id="cb43-197"><a href="#cb43-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-198"><a href="#cb43-198" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-199"><a href="#cb43-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-200"><a href="#cb43-200" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, echo = F}</span></span>
<span id="cb43-201"><a href="#cb43-201" aria-hidden="true" tabindex="-1"></a><span class="in">cd example </span></span>
<span id="cb43-202"><a href="#cb43-202" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py list_tar_amp_bioinformatics_info_ids --file combined_Heome1_PMO.json.gz </span></span>
<span id="cb43-203"><a href="#cb43-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-204"><a href="#cb43-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-205"><a href="#cb43-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-206"><a href="#cb43-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-207"><a href="#cb43-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-208"><a href="#cb43-208" aria-hidden="true" tabindex="-1"></a><span class="fu">## count_targets_per_sample </span></span>
<span id="cb43-209"><a href="#cb43-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-210"><a href="#cb43-210" aria-hidden="true" tabindex="-1"></a>Count up the number targets each experimental_sample_id has. A read filter can be applied to see how targets would be kept if such a filter was applied </span>
<span id="cb43-211"><a href="#cb43-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-214"><a href="#cb43-214" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-215"><a href="#cb43-215" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">-h</span> </span>
<span id="cb43-216"><a href="#cb43-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-217"><a href="#cb43-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-218"><a href="#cb43-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-219"><a href="#cb43-219" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`count_targets_per_sample`</span> script is below</span>
<span id="cb43-220"><a href="#cb43-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-223"><a href="#cb43-223" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb43-224"><a href="#cb43-224" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb43-225"><a href="#cb43-225" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb43-226"><a href="#cb43-226" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb43-227"><a href="#cb43-227" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb43-228"><a href="#cb43-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extract_info_from_pmo/count_targets_per_sample.py</span></span>
<span id="cb43-229"><a href="#cb43-229" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extract_info_from_pmo/count_targets_per_sample.py</span></span>
<span id="cb43-230"><a href="#cb43-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-231"><a href="#cb43-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-234"><a href="#cb43-234" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-235"><a href="#cb43-235" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-236"><a href="#cb43-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-237"><a href="#cb43-237" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span>
<span id="cb43-238"><a href="#cb43-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-239"><a href="#cb43-239" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-240"><a href="#cb43-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-241"><a href="#cb43-241" aria-hidden="true" tabindex="-1"></a>Apply a read count minimum filter (this a total read count summed for a target and not on a haplotype level) </span>
<span id="cb43-242"><a href="#cb43-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-245"><a href="#cb43-245" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-246"><a href="#cb43-246" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-247"><a href="#cb43-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-248"><a href="#cb43-248" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_targets_per_sample <span class="at">--read_count_minimum</span> 3000 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span>
<span id="cb43-249"><a href="#cb43-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-250"><a href="#cb43-250" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-251"><a href="#cb43-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-252"><a href="#cb43-252" aria-hidden="true" tabindex="-1"></a><span class="fu">## count_samples_per_target  </span></span>
<span id="cb43-253"><a href="#cb43-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-254"><a href="#cb43-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-255"><a href="#cb43-255" aria-hidden="true" tabindex="-1"></a>Count up the number of experimental_sample_ids each target has. A read filter can be applied to see how many samples a taget would have if a filter was applied </span>
<span id="cb43-256"><a href="#cb43-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-257"><a href="#cb43-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-260"><a href="#cb43-260" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-261"><a href="#cb43-261" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">-h</span> </span>
<span id="cb43-262"><a href="#cb43-262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-263"><a href="#cb43-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-264"><a href="#cb43-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-265"><a href="#cb43-265" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`count_samples_per_target`</span> script is below</span>
<span id="cb43-266"><a href="#cb43-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-269"><a href="#cb43-269" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb43-270"><a href="#cb43-270" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb43-271"><a href="#cb43-271" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb43-272"><a href="#cb43-272" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb43-273"><a href="#cb43-273" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb43-274"><a href="#cb43-274" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extract_info_from_pmo/count_samples_per_target.py</span></span>
<span id="cb43-275"><a href="#cb43-275" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extract_info_from_pmo/count_samples_per_target.py</span></span>
<span id="cb43-276"><a href="#cb43-276" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-277"><a href="#cb43-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-278"><a href="#cb43-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-279"><a href="#cb43-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-282"><a href="#cb43-282" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-283"><a href="#cb43-283" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-284"><a href="#cb43-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-285"><a href="#cb43-285" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span>
<span id="cb43-286"><a href="#cb43-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-287"><a href="#cb43-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-288"><a href="#cb43-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-289"><a href="#cb43-289" aria-hidden="true" tabindex="-1"></a>Apply a read count minimum filter (this a total read count summed for a target and not on a haplotype level) </span>
<span id="cb43-290"><a href="#cb43-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-293"><a href="#cb43-293" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb43-294"><a href="#cb43-294" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb43-295"><a href="#cb43-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-296"><a href="#cb43-296" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_samples_per_target <span class="at">--read_count_minimum</span> 3000 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz  <span class="kw">|</span> <span class="fu">head</span></span>
<span id="cb43-297"><a href="#cb43-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-298"><a href="#cb43-298" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb43-299"><a href="#cb43-299" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>