<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Subsetting from a PMO using pmotools-python â€“ Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-cee8642dd6abb73bb51e0193ec9c0780.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-83b1d0b3b459d34b5b0f98adfe5eb835.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>html{ scroll-behavior: smooth; }</style>
<link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">
<li>
    <a class="dropdown-item" href="../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">
<li>
    <a class="dropdown-item" href="../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">
<li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-runner.py</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/pmotools-runner.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-runner.py</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li>
    <a class="dropdown-item" href="../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">
<li>
    <a class="dropdown-item" href="../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Contents</h2>
   
  <ul>
<li>
<a href="#subsetting" id="toc-subsetting" class="nav-link active" data-scroll-target="#subsetting">Subsetting</a>
  <ul>
<li><a href="#subsetting-by-specific-targets" id="toc-subsetting-by-specific-targets" class="nav-link" data-scroll-target="#subsetting-by-specific-targets">Subsetting by specific targets</a></li>
  <li><a href="#subsetting-by-specific-specimen_ids" id="toc-subsetting-by-specific-specimen_ids" class="nav-link" data-scroll-target="#subsetting-by-specific-specimen_ids">Subsetting by specific specimen_ids</a></li>
  <li><a href="#subsetting-by-specific-experiment_sample_ids" id="toc-subsetting-by-specific-experiment_sample_ids" class="nav-link" data-scroll-target="#subsetting-by-specific-experiment_sample_ids">Subsetting by specific experiment_sample_ids</a></li>
  <li><a href="#subsetting-by-samples-within-specific-metafields" id="toc-subsetting-by-samples-within-specific-metafields" class="nav-link" data-scroll-target="#subsetting-by-samples-within-specific-metafields">Subsetting by samples within specific metafields</a></li>
  <li><a href="#piping-together-extraction" id="toc-piping-together-extraction" class="nav-link" data-scroll-target="#piping-together-extraction">Piping together extraction</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Subsetting from a PMO using pmotools-python</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-page-left">

    
  
    
  </div>
  


</header><section id="subsetting" class="level1"><h1>Subsetting</h1>
<p>There may be some instances were you want to subset a much larger PMO file into a smaller PMO file to focus only one a set of samples and/or targets. There are several ways of doing this.</p>
<section id="subsetting-by-specific-targets" class="level2"><h2 class="anchored" data-anchor-id="subsetting-by-specific-targets">Subsetting by specific targets</h2>
<p>Can subset to only specific targets by using <code>pmotools-runner.py extract_pmo_with_select_targets</code></p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">-h</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py extract_pmo_with_select_targets [-h] --file FILE
                                                          --output OUTPUT
                                                          [--overwrite]
                                                          [--verbose]
                                                          --targets TARGETS

options:
  -h, --help         show this help message and exit
  --file FILE        PMO file
  --output OUTPUT    Output json file path
  --overwrite        If output file exists, overwrite it
  --verbose          write out various messages about extraction
  --targets TARGETS  Can either comma separated target_ids, or a plain text
                     file where each line is a target_ids</code></pre>
</div>
</div>
<p>The python code for <code>extract_pmo_with_select_targets</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_targets.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="im">import</span> sys</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb3-5"><a href="#cb3-5"></a></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="im">from</span> pmotools.pmo_utils.PMOWriter <span class="im">import</span> PMOWriter</span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb3-12"><a href="#cb3-12"></a></span>
<span id="cb3-13"><a href="#cb3-13"></a></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="kw">def</span> parse_args_extract_pmo_with_select_targets():</span>
<span id="cb3-15"><a href="#cb3-15"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb3-16"><a href="#cb3-16"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb3-17"><a href="#cb3-17"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Output json file path'</span>)</span>
<span id="cb3-18"><a href="#cb3-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb3-19"><a href="#cb3-19"></a>    parser.add_argument(<span class="st">'--verbose'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'write out various messages about extraction'</span>)</span>
<span id="cb3-20"><a href="#cb3-20"></a>    parser.add_argument(<span class="st">'--targets'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Can either comma separated target_ids, or a plain text file where each line is a target_ids'</span>)</span>
<span id="cb3-21"><a href="#cb3-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb3-22"><a href="#cb3-22"></a></span>
<span id="cb3-23"><a href="#cb3-23"></a><span class="kw">def</span> extract_pmo_with_select_targets():</span>
<span id="cb3-24"><a href="#cb3-24"></a>    args <span class="op">=</span> parse_args_extract_pmo_with_select_targets()</span>
<span id="cb3-25"><a href="#cb3-25"></a></span>
<span id="cb3-26"><a href="#cb3-26"></a>    <span class="co"># check files</span></span>
<span id="cb3-27"><a href="#cb3-27"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb3-28"><a href="#cb3-28"></a></span>
<span id="cb3-29"><a href="#cb3-29"></a>    <span class="co"># parse target ids</span></span>
<span id="cb3-30"><a href="#cb3-30"></a>    all_target_ids <span class="op">=</span> Utils.parse_delimited_input_or_file(args.targets)</span>
<span id="cb3-31"><a href="#cb3-31"></a></span>
<span id="cb3-32"><a href="#cb3-32"></a>    <span class="co"># read in pmo</span></span>
<span id="cb3-33"><a href="#cb3-33"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb3-34"><a href="#cb3-34"></a></span>
<span id="cb3-35"><a href="#cb3-35"></a>    <span class="co"># extract</span></span>
<span id="cb3-36"><a href="#cb3-36"></a>    pmo_out <span class="op">=</span> PMOExtractor.extract_from_pmo_select_targets(pmo, all_target_ids)</span>
<span id="cb3-37"><a href="#cb3-37"></a></span>
<span id="cb3-38"><a href="#cb3-38"></a>    <span class="co"># write out the extracted</span></span>
<span id="cb3-39"><a href="#cb3-39"></a>    args.output <span class="op">=</span> PMOWriter.add_pmo_extension_as_needed(args.output, args.<span class="bu">file</span>.endswith(<span class="st">'.gz'</span>) <span class="kw">or</span> args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb3-40"><a href="#cb3-40"></a>    PMOWriter.write_out_pmo(pmo_out, args.output, args.overwrite)</span>
<span id="cb3-41"><a href="#cb3-41"></a></span>
<span id="cb3-42"><a href="#cb3-42"></a></span>
<span id="cb3-43"><a href="#cb3-43"></a></span>
<span id="cb3-44"><a href="#cb3-44"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb3-45"><a href="#cb3-45"></a>    extract_pmo_with_select_targets()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<p>You can extract by supplies the desired targets with comma separated values on the command line</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>You can also provide a single column file where each line is a desired target</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"t1\nt20\nt31"</span> <span class="op">&gt;</span> select_targets.txt </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> select_targets.txt  <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"t1\nt20\nt31"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> STDIN <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="subsetting-by-specific-specimen_ids" class="level2"><h2 class="anchored" data-anchor-id="subsetting-by-specific-specimen_ids">Subsetting by specific specimen_ids</h2>
<p>You can subset to just to just select specimen_id, each specimen can have several experiments associated with it, by supplying the specimen_id all associated experiments will also be pulled</p>
<p>Similar to above you can supply the specimen_ids either as comma separated values or in a plain text file where each line is a specimen_id</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">-h</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py extract_pmo_with_select_specimen_ids
       [-h] --file FILE --output OUTPUT [--overwrite] [--verbose]
       --specimen_ids SPECIMEN_IDS

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       Output json file path
  --overwrite           If output file exists, overwrite it
  --verbose             write out various messages about extraction
  --specimen_ids SPECIMEN_IDS
                        Can either comma separated specimen_ids, or a plain
                        text file where each line is a specimen_id</code></pre>
</div>
</div>
<p>The python code for <code>extract_pmo_with_select_specimen_ids</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_specimen_ids.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="im">import</span> sys</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb8-5"><a href="#cb8-5"></a></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb8-7"><a href="#cb8-7"></a></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="im">from</span> pmotools.pmo_utils.PMOWriter <span class="im">import</span> PMOWriter</span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb8-12"><a href="#cb8-12"></a></span>
<span id="cb8-13"><a href="#cb8-13"></a></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="kw">def</span> parse_args_extract_pmo_with_select_specimen_ids():</span>
<span id="cb8-15"><a href="#cb8-15"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb8-16"><a href="#cb8-16"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb8-17"><a href="#cb8-17"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Output json file path'</span>)</span>
<span id="cb8-18"><a href="#cb8-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb8-19"><a href="#cb8-19"></a>    parser.add_argument(<span class="st">'--verbose'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'write out various messages about extraction'</span>)</span>
<span id="cb8-20"><a href="#cb8-20"></a>    parser.add_argument(<span class="st">'--specimen_ids'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Can either comma separated specimen_ids, or a plain text file where each line is a specimen_id'</span>)</span>
<span id="cb8-21"><a href="#cb8-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb8-22"><a href="#cb8-22"></a></span>
<span id="cb8-23"><a href="#cb8-23"></a><span class="kw">def</span> extract_pmo_with_select_specimen_ids():</span>
<span id="cb8-24"><a href="#cb8-24"></a>    args <span class="op">=</span> parse_args_extract_pmo_with_select_specimen_ids()</span>
<span id="cb8-25"><a href="#cb8-25"></a></span>
<span id="cb8-26"><a href="#cb8-26"></a>    <span class="co"># check files</span></span>
<span id="cb8-27"><a href="#cb8-27"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb8-28"><a href="#cb8-28"></a></span>
<span id="cb8-29"><a href="#cb8-29"></a>    <span class="co"># parse specimen ids</span></span>
<span id="cb8-30"><a href="#cb8-30"></a>    all_specimen_ids <span class="op">=</span> Utils.parse_delimited_input_or_file(args.specimen_ids)</span>
<span id="cb8-31"><a href="#cb8-31"></a></span>
<span id="cb8-32"><a href="#cb8-32"></a>    <span class="co"># read in pmo</span></span>
<span id="cb8-33"><a href="#cb8-33"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb8-34"><a href="#cb8-34"></a></span>
<span id="cb8-35"><a href="#cb8-35"></a>    <span class="co"># extract</span></span>
<span id="cb8-36"><a href="#cb8-36"></a>    pmo_out <span class="op">=</span> PMOExtractor.extract_from_pmo_select_specimen_ids(pmo, all_specimen_ids)</span>
<span id="cb8-37"><a href="#cb8-37"></a></span>
<span id="cb8-38"><a href="#cb8-38"></a>    <span class="co"># write out the extracted</span></span>
<span id="cb8-39"><a href="#cb8-39"></a>    args.output <span class="op">=</span> PMOWriter.add_pmo_extension_as_needed(args.output, args.<span class="bu">file</span>.endswith(<span class="st">'.gz'</span>) <span class="kw">or</span> args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb8-40"><a href="#cb8-40"></a>    PMOWriter.write_out_pmo(pmo_out, args.output, args.overwrite)</span>
<span id="cb8-41"><a href="#cb8-41"></a></span>
<span id="cb8-42"><a href="#cb8-42"></a></span>
<span id="cb8-43"><a href="#cb8-43"></a></span>
<span id="cb8-44"><a href="#cb8-44"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb8-45"><a href="#cb8-45"></a>    extract_pmo_with_select_specimen_ids()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> 8025874217,8025875146,8034209589 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025874217\n8025875146\n8034209589"</span> <span class="op">&gt;</span> select_specimen_ids.txt </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> select_specimen_ids.txt <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025874217\n8025875146\n8034209589"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="subsetting-by-specific-experiment_sample_ids" class="level2"><h2 class="anchored" data-anchor-id="subsetting-by-specific-experiment_sample_ids">Subsetting by specific experiment_sample_ids</h2>
<p>If you want just specific experiment_sample_id you can supply those instead too</p>
<p>Similar to above you can supply the experiment_sample_ids either as comma separated values or in a plain text file where each line is a experiment_sample_id or from standard in (STDIN)</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">-h</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py extract_pmo_with_select_experiment_sample_ids
       [-h] --file FILE --output OUTPUT [--overwrite] [--verbose]
       --experiment_sample_ids EXPERIMENT_SAMPLE_IDS

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       Output json file path
  --overwrite           If output file exists, overwrite it
  --verbose             write out various messages about extraction
  --experiment_sample_ids EXPERIMENT_SAMPLE_IDS
                        Can either comma separated experiment_sample_ids, or a
                        plain text file where each line is a
                        experiment_sample_id</code></pre>
</div>
</div>
<p>The python code for <code>extract_pmo_with_select_experiment_sample_ids</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_experiment_sample_ids.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="im">import</span> sys</span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb12-5"><a href="#cb12-5"></a></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb12-7"><a href="#cb12-7"></a></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="im">from</span> pmotools.pmo_utils.PMOWriter <span class="im">import</span> PMOWriter</span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb12-12"><a href="#cb12-12"></a></span>
<span id="cb12-13"><a href="#cb12-13"></a></span>
<span id="cb12-14"><a href="#cb12-14"></a><span class="kw">def</span> parse_args_extract_pmo_with_select_experiment_sample_ids():</span>
<span id="cb12-15"><a href="#cb12-15"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb12-16"><a href="#cb12-16"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb12-17"><a href="#cb12-17"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Output json file path'</span>)</span>
<span id="cb12-18"><a href="#cb12-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb12-19"><a href="#cb12-19"></a>    parser.add_argument(<span class="st">'--verbose'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'write out various messages about extraction'</span>)</span>
<span id="cb12-20"><a href="#cb12-20"></a>    parser.add_argument(<span class="st">'--experiment_sample_ids'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Can either comma separated experiment_sample_ids, or a plain text file where each line is a experiment_sample_id'</span>)</span>
<span id="cb12-21"><a href="#cb12-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb12-22"><a href="#cb12-22"></a></span>
<span id="cb12-23"><a href="#cb12-23"></a><span class="kw">def</span> extract_pmo_with_select_experiment_sample_ids():</span>
<span id="cb12-24"><a href="#cb12-24"></a>    args <span class="op">=</span> parse_args_extract_pmo_with_select_experiment_sample_ids()</span>
<span id="cb12-25"><a href="#cb12-25"></a></span>
<span id="cb12-26"><a href="#cb12-26"></a>    <span class="co"># check files</span></span>
<span id="cb12-27"><a href="#cb12-27"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb12-28"><a href="#cb12-28"></a></span>
<span id="cb12-29"><a href="#cb12-29"></a>    <span class="co"># parse specimen ids</span></span>
<span id="cb12-30"><a href="#cb12-30"></a>    all_experiment_sample_ids <span class="op">=</span> Utils.parse_delimited_input_or_file(args.experiment_sample_ids)</span>
<span id="cb12-31"><a href="#cb12-31"></a></span>
<span id="cb12-32"><a href="#cb12-32"></a>    <span class="co"># read in pmo</span></span>
<span id="cb12-33"><a href="#cb12-33"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb12-34"><a href="#cb12-34"></a></span>
<span id="cb12-35"><a href="#cb12-35"></a>    <span class="co"># extract</span></span>
<span id="cb12-36"><a href="#cb12-36"></a>    pmo_out <span class="op">=</span> PMOExtractor.extract_from_pmo_select_experiment_sample_ids(pmo, all_experiment_sample_ids)</span>
<span id="cb12-37"><a href="#cb12-37"></a></span>
<span id="cb12-38"><a href="#cb12-38"></a>    <span class="co"># write out the extracted</span></span>
<span id="cb12-39"><a href="#cb12-39"></a>    args.output <span class="op">=</span> PMOWriter.add_pmo_extension_as_needed(args.output, args.<span class="bu">file</span>.endswith(<span class="st">'.gz'</span>) <span class="kw">or</span> args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb12-40"><a href="#cb12-40"></a>    PMOWriter.write_out_pmo(pmo_out, args.output, args.overwrite)</span>
<span id="cb12-41"><a href="#cb12-41"></a></span>
<span id="cb12-42"><a href="#cb12-42"></a></span>
<span id="cb12-43"><a href="#cb12-43"></a></span>
<span id="cb12-44"><a href="#cb12-44"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb12-45"><a href="#cb12-45"></a>    extract_pmo_with_select_experiment_sample_ids()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> 8025875029,8034209834,8034209115 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="op">&gt;</span> select_experiment_sample_ids.txt </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> select_experiment_sample_ids.txt <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="subsetting-by-samples-within-specific-metafields" class="level2"><h2 class="anchored" data-anchor-id="subsetting-by-samples-within-specific-metafields">Subsetting by samples within specific metafields</h2>
<p>If you want to get specific samples that match certain meta fields like specific collection_country or collection_date you can use ``</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta <span class="at">-h</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>usage: pmotools-runner.py extract_pmo_with_selected_meta [-h] --file FILE
                                                         --output OUTPUT
                                                         [--overwrite]
                                                         [--verbose]
                                                         --metaFieldsValues
                                                         METAFIELDSVALUES

options:
  -h, --help            show this help message and exit
  --file FILE           PMO file
  --output OUTPUT       Output json file path
  --overwrite           If output file exists, overwrite it
  --verbose             write out various messages about extraction
  --metaFieldsValues METAFIELDSVALUES
                        Meta Fields to include, should either be a table with
                        columns field, values (and optionally group) or
                        supplied command line as
                        field1=value1,value2,value3:field2=value1,value2</code></pre>
</div>
</div>
<p>The python code for <code>extract_pmo_with_selected_meta</code> script is below</p>
<div class="cell" data-layout-align="center">
<div class="code-with-filename">
<details class="code-fold"><summary>Code</summary><div class="code-with-filename-file">
<pre><strong>pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_selected_meta.py</strong></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource python number-lines code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">#!/usr/bin/env python3</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="im">import</span> os, argparse, json</span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="im">import</span> sys</span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="im">from</span> collections <span class="im">import</span> defaultdict</span>
<span id="cb16-5"><a href="#cb16-5"></a></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb16-7"><a href="#cb16-7"></a></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="im">from</span> pmotools.pmo_utils.PMOExtractor <span class="im">import</span> PMOExtractor</span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="im">from</span> pmotools.pmo_utils.PMOReader <span class="im">import</span> PMOReader</span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="im">from</span> pmotools.pmo_utils.PMOWriter <span class="im">import</span> PMOWriter</span>
<span id="cb16-11"><a href="#cb16-11"></a><span class="im">from</span> pmotools.utils.small_utils <span class="im">import</span> Utils</span>
<span id="cb16-12"><a href="#cb16-12"></a></span>
<span id="cb16-13"><a href="#cb16-13"></a></span>
<span id="cb16-14"><a href="#cb16-14"></a><span class="kw">def</span> parse_args_extract_pmo_with_selected_meta():</span>
<span id="cb16-15"><a href="#cb16-15"></a>    parser <span class="op">=</span> argparse.ArgumentParser()</span>
<span id="cb16-16"><a href="#cb16-16"></a>    parser.add_argument(<span class="st">'--file'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'PMO file'</span>)</span>
<span id="cb16-17"><a href="#cb16-17"></a>    parser.add_argument(<span class="st">'--output'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Output json file path'</span>)</span>
<span id="cb16-18"><a href="#cb16-18"></a>    parser.add_argument(<span class="st">'--overwrite'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'If output file exists, overwrite it'</span>)</span>
<span id="cb16-19"><a href="#cb16-19"></a>    parser.add_argument(<span class="st">'--verbose'</span>, action <span class="op">=</span> <span class="st">'store_true'</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'write out various messages about extraction'</span>)</span>
<span id="cb16-20"><a href="#cb16-20"></a>    parser.add_argument(<span class="st">'--metaFieldsValues'</span>, <span class="bu">type</span><span class="op">=</span><span class="bu">str</span>, required<span class="op">=</span><span class="va">True</span>, <span class="bu">help</span><span class="op">=</span><span class="st">'Meta Fields to include, should either be a table with columns field, values (and optionally group) or supplied command line as field1=value1,value2,value3:field2=value1,value2'</span>)</span>
<span id="cb16-21"><a href="#cb16-21"></a>    <span class="cf">return</span> parser.parse_args()</span>
<span id="cb16-22"><a href="#cb16-22"></a></span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="kw">def</span> extract_pmo_with_selected_meta():</span>
<span id="cb16-24"><a href="#cb16-24"></a>    args <span class="op">=</span> parse_args_extract_pmo_with_selected_meta()</span>
<span id="cb16-25"><a href="#cb16-25"></a></span>
<span id="cb16-26"><a href="#cb16-26"></a>    <span class="co"># check files</span></span>
<span id="cb16-27"><a href="#cb16-27"></a>    Utils.inputOutputFileCheck(args.<span class="bu">file</span>, args.output, args.overwrite)</span>
<span id="cb16-28"><a href="#cb16-28"></a></span>
<span id="cb16-29"><a href="#cb16-29"></a>    <span class="co"># read in pmo</span></span>
<span id="cb16-30"><a href="#cb16-30"></a>    pmo <span class="op">=</span> PMOReader.read_in_pmo(args.<span class="bu">file</span>)</span>
<span id="cb16-31"><a href="#cb16-31"></a></span>
<span id="cb16-32"><a href="#cb16-32"></a>    <span class="co"># extract out of PMO</span></span>
<span id="cb16-33"><a href="#cb16-33"></a>    pmo_out, group_counts <span class="op">=</span> PMOExtractor.extract_from_pmo_samples_with_meta_groupings(pmo, args.metaFieldsValues)</span>
<span id="cb16-34"><a href="#cb16-34"></a></span>
<span id="cb16-35"><a href="#cb16-35"></a>    <span class="co"># write out the extracted</span></span>
<span id="cb16-36"><a href="#cb16-36"></a>    args.output <span class="op">=</span> PMOWriter.add_pmo_extension_as_needed(args.output, args.<span class="bu">file</span>.endswith(<span class="st">'.gz'</span>) <span class="kw">or</span> args.output.endswith(<span class="st">".gz"</span>))</span>
<span id="cb16-37"><a href="#cb16-37"></a>    PMOWriter.write_out_pmo(pmo_out, args.output, args.overwrite)</span>
<span id="cb16-38"><a href="#cb16-38"></a></span>
<span id="cb16-39"><a href="#cb16-39"></a>    <span class="cf">if</span> args.verbose:</span>
<span id="cb16-40"><a href="#cb16-40"></a>        sys.stdout.write(<span class="st">"Extracted the following number of specimens per group:"</span> <span class="op">+</span> <span class="st">"</span><span class="ch">\n</span><span class="st">"</span>)</span>
<span id="cb16-41"><a href="#cb16-41"></a>        group_counts.to_csv(sys.stdout, sep <span class="op">=</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>, index <span class="op">=</span> <span class="va">True</span>)</span>
<span id="cb16-42"><a href="#cb16-42"></a></span>
<span id="cb16-43"><a href="#cb16-43"></a><span class="cf">if</span> <span class="va">__name__</span> <span class="op">==</span> <span class="st">"__main__"</span>:</span>
<span id="cb16-44"><a href="#cb16-44"></a>    extract_pmo_with_selected_meta()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div>
<p><code>pmotools-runner.py extract_pmo_with_selected_meta</code> is written to allow the extraction on multiple intersecting meta field requirments that can be either supplied in a file or with delimited on the command line</p>
<p>You may also want to know what current meta fields are present and how many samples in each. This can be done with <a href="../pmotools-python-usages/getting_basic_info_from_pmo.html#list_specimen_meta_fields"><code>pmotools-runner.py list_specimen_meta_fields</code></a> and <code>pmotools-runner.py count_specimen_meta</code></p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>field   presentInSpecimensCount totalSpecimenCount
collection_country  19433   19433
collection_date 19433   19433
collector   19433   19433
geo_admin3  19433   19433
geo_continent   19433   19433
geo_region  19433   19433
geo_subRegion   19433   19433
host_taxon_id   19433   19433
project_name    19433   19433
samp_collect_device 19433   19433
samp_store_loc  19433   19433
samp_taxon_id   19433   19433
specimen_id 19433   19433</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz  <span class="at">--meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span> <span class="at">-20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>collection_country  collection_date specimensCount  specimensFreq   totalSpecimenCount
Bangladesh  2008    15  0.0007718828796377296   19433
Bangladesh  2009    16  0.000823341738280245    19433
Bangladesh  2012    51  0.002624401790768281    19433
Bangladesh  2015    508 0.026141100190397778    19433
Bangladesh  2016    816 0.041990428652292494    19433
Bangladesh  2017    12  0.0006175063037101837   19433
Benin   2014    41  0.002109813204343128    19433
Benin   2016    117 0.006020686461174291    19433
Brazil  1980    1   5.145885864251531e-05   19433
Brazil  2016    13  0.000668965162352699    19433
Brazil  2017    5   0.00025729429321257654  19433
Brazil  NA  1   5.145885864251531e-05   19433
Burkina Faso    2008    58  0.002984613801265888    19433
Cambodia    1993    6   0.00030875315185509186  19433
Cambodia    2007    26  0.001337930324705398    19433
Cambodia    2008    50  0.0025729429321257654   19433
Cambodia    2009    66  0.0033962846704060105   19433
Cambodia    2010    182 0.009365512272937786    19433
Cambodia    2011    441 0.02269335666134925 19433</code></pre>
</div>
</div>
<p>Extracting on matching 1 meta field, below will extract just the samples that have collection_country=Bangladesh</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh"</span> <span class="at">--output</span> Bangladesh_moz2018_PMO.json.gz   <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>If you want to see how many samples were extracted can use <code>--verbose</code></p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh"</span> <span class="at">--output</span> Bangladesh_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  count
0   Bangladesh  1418</code></pre>
</div>
</div>
<p>Collecting more than 1 matching field separate by comma, for example to extract both Bangladesh,Benin</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh,Benin"</span> <span class="at">--output</span> Bangladesh_Benin_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  count
0   Bangladesh,Benin    1576</code></pre>
</div>
</div>
<p>Can add more extraction criteria meta, for example to extract samples with collection_country of Bangladesh or Benin and with collection_date of 2016</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh,Benin:collection_date=2016"</span> <span class="at">--output</span> Bangladesh_Benin_2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  collection_date count
0   Bangladesh,Benin    2016    933</code></pre>
</div>
</div>
<p>To get more specific you can group meta field extraction criteria , for example if you wanted samples from Bangladesh from year 2015 but wanted Benin from year 2016 you can separate by a ;</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh:collection_date=2015;collection_country=Benin:collection_date=2016"</span> <span class="at">--output</span> Bangladesh2015_Benin2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  collection_date count
0   Bangladesh  2015    508
1   Benin   2016    117</code></pre>
</div>
</div>
<p>Rather than supplying with the command line a file can be created</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"group\tfield\tvalues"</span> <span class="op">&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Bangladesh2015\tcollection_country\tBangladesh"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Bangladesh2015\tcollection_date\t2015"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Benin2016\tcollection_country\tBenin"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Benin2016\tcollection_date\t2016"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> Bangladesh2015_Benin2016_extractionCriteria.tsv <span class="at">--output</span> Bangladesh2015_Benin2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  collection_date count
Bangladesh2015  Bangladesh  2015    508
Benin2016   Benin   2016    117</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Mozambique:geo_admin3=Inhassoro;collection_country=Mozambique:geo_admin3=Mandlakazi,Namaacha"</span> <span class="at">--output</span> Mozambique_moz2018_PMO.json.gz  <span class="at">--verbose</span>  <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Extracted the following number of specimens per group:
group   collection_country  geo_admin3  count
0   Mozambique  Inhassoro   27
1   Mozambique  Mandlakazi,Namaacha 54</code></pre>
</div>
</div>
</section><section id="piping-together-extraction" class="level2"><h2 class="anchored" data-anchor-id="piping-together-extraction">Piping together extraction</h2>
<p>The extraction methods also allow for STDOUT and STDIN piping for example</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> t1_t20_t31_8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Can also pipe into other <code>pmotools-runner.py</code> functions like extracting allele tables</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_allele_table <span class="at">--file</span> STDIN <span class="at">--bioid</span> Mozambique2018-SeekDeep <span class="at">--output</span> alleles_data_t1_t20_t31_8025875029_8034209834_8034209115_moz2018_PMO.tsv.gz <span class="at">--overwrite</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Can pipe final output to STDOUT as well for further processing</p>
<div class="cell" data-layout-align="center">
<details open="" class="code-fold"><summary>Code</summary><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_allele_table <span class="at">--file</span> STDIN <span class="at">--bioid</span> Mozambique2018-SeekDeep <span class="at">--output</span> STDOUT  <span class="at">--specimen_info_meta_fields</span> specimen_id,collection_country</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>sampleID    locus   allele  specimen_id collection_country
8025875029  t1  t1.0    8025875029  Mozambique
8025875029  t1  t1.1    8025875029  Mozambique
8025875029  t20 t20.3   8025875029  Mozambique
8025875029  t20 t20.5   8025875029  Mozambique
8025875029  t20 t20.4   8025875029  Mozambique
8025875029  t31 t31.1   8025875029  Mozambique
8025875029  t31 t31.3   8025875029  Mozambique
8034209115  t1  t1.2    8034209115  Mozambique
8034209115  t20 t20.4   8034209115  Mozambique
8034209115  t20 t20.1   8034209115  Mozambique
8034209115  t31 t31.1   8034209115  Mozambique
8034209115  t31 t31.3   8034209115  Mozambique
8034209834  t1  t1.0    8034209834  Mozambique
8034209834  t20 t20.1   8034209834  Mozambique
8034209834  t20 t20.0   8034209834  Mozambique
8034209834  t31 t31.2   8034209834  Mozambique
8034209834  t31 t31.0   8034209834  Mozambique</code></pre>
</div>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb39" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Subsetting from a PMO using pmotools-python</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, echo=F}</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="in">source("../common.R")</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Subsetting </span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>There may be some instances were you want to subset a much larger PMO file into a smaller PMO file to focus only one a set of samples and/or targets. There are several ways of doing this. </span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="fu">## Subsetting by specific targets  </span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>Can subset to only specific targets by using <span class="in">`pmotools-runner.py extract_pmo_with_select_targets`</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">-h</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`extract_pmo_with_select_targets`</span> script is below</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_targets.py</span></span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_targets.py</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>You can extract by supplies the desired targets with comma separated values on the command line </span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a>You can also provide a single column file where each line is a desired target </span>
<span id="cb39-51"><a href="#cb39-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-52"><a href="#cb39-52" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-53"><a href="#cb39-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-54"><a href="#cb39-54" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"t1\nt20\nt31"</span> <span class="op">&gt;</span> select_targets.txt </span>
<span id="cb39-55"><a href="#cb39-55" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> select_targets.txt  <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-56"><a href="#cb39-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-57"><a href="#cb39-57" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"t1\nt20\nt31"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--targets</span> STDIN <span class="at">--output</span> t1_t20_t31_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-58"><a href="#cb39-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-59"><a href="#cb39-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-60"><a href="#cb39-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-61"><a href="#cb39-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-62"><a href="#cb39-62" aria-hidden="true" tabindex="-1"></a><span class="fu">## Subsetting by specific specimen_ids </span></span>
<span id="cb39-63"><a href="#cb39-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-64"><a href="#cb39-64" aria-hidden="true" tabindex="-1"></a>You can subset to just to just select specimen_id, each specimen can have several experiments associated with it, by supplying the specimen_id all associated experiments will also be pulled </span>
<span id="cb39-65"><a href="#cb39-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-66"><a href="#cb39-66" aria-hidden="true" tabindex="-1"></a>Similar to above you can supply the specimen_ids either as comma separated values or in a plain text file where each line is a specimen_id</span>
<span id="cb39-67"><a href="#cb39-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-70"><a href="#cb39-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-71"><a href="#cb39-71" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">-h</span></span>
<span id="cb39-72"><a href="#cb39-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-73"><a href="#cb39-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-74"><a href="#cb39-74" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`extract_pmo_with_select_specimen_ids`</span> script is below</span>
<span id="cb39-75"><a href="#cb39-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-78"><a href="#cb39-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb39-79"><a href="#cb39-79" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb39-80"><a href="#cb39-80" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb39-81"><a href="#cb39-81" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb39-82"><a href="#cb39-82" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb39-83"><a href="#cb39-83" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_specimen_ids.py</span></span>
<span id="cb39-84"><a href="#cb39-84" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_specimen_ids.py</span></span>
<span id="cb39-85"><a href="#cb39-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-86"><a href="#cb39-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-89"><a href="#cb39-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-90"><a href="#cb39-90" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-91"><a href="#cb39-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-92"><a href="#cb39-92" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> 8025874217,8025875146,8034209589 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-93"><a href="#cb39-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-94"><a href="#cb39-94" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025874217\n8025875146\n8034209589"</span> <span class="op">&gt;</span> select_specimen_ids.txt </span>
<span id="cb39-95"><a href="#cb39-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-96"><a href="#cb39-96" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> select_specimen_ids.txt <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-97"><a href="#cb39-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-98"><a href="#cb39-98" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025874217\n8025875146\n8034209589"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_specimen_ids <span class="at">--specimen_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025874217_8025875146_8034209589_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-99"><a href="#cb39-99" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-100"><a href="#cb39-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-101"><a href="#cb39-101" aria-hidden="true" tabindex="-1"></a><span class="fu">## Subsetting by specific experiment_sample_ids </span></span>
<span id="cb39-102"><a href="#cb39-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-103"><a href="#cb39-103" aria-hidden="true" tabindex="-1"></a>If you want just specific experiment_sample_id you can supply those instead too  </span>
<span id="cb39-104"><a href="#cb39-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-105"><a href="#cb39-105" aria-hidden="true" tabindex="-1"></a>Similar to above you can supply the experiment_sample_ids either as comma separated values or in a plain text file where each line is a experiment_sample_id or from standard in (STDIN)</span>
<span id="cb39-106"><a href="#cb39-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-109"><a href="#cb39-109" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-110"><a href="#cb39-110" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">-h</span></span>
<span id="cb39-111"><a href="#cb39-111" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-112"><a href="#cb39-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-113"><a href="#cb39-113" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`extract_pmo_with_select_experiment_sample_ids`</span> script is below</span>
<span id="cb39-114"><a href="#cb39-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-117"><a href="#cb39-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb39-118"><a href="#cb39-118" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb39-119"><a href="#cb39-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb39-120"><a href="#cb39-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb39-121"><a href="#cb39-121" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb39-122"><a href="#cb39-122" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_experiment_sample_ids.py</span></span>
<span id="cb39-123"><a href="#cb39-123" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_select_experiment_sample_ids.py</span></span>
<span id="cb39-124"><a href="#cb39-124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-125"><a href="#cb39-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-128"><a href="#cb39-128" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-129"><a href="#cb39-129" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-130"><a href="#cb39-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-131"><a href="#cb39-131" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> 8025875029,8034209834,8034209115 <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-132"><a href="#cb39-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-133"><a href="#cb39-133" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="op">&gt;</span> select_experiment_sample_ids.txt </span>
<span id="cb39-134"><a href="#cb39-134" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-135"><a href="#cb39-135" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> select_experiment_sample_ids.txt <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-136"><a href="#cb39-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-137"><a href="#cb39-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-138"><a href="#cb39-138" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> 8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-139"><a href="#cb39-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-140"><a href="#cb39-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-141"><a href="#cb39-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-142"><a href="#cb39-142" aria-hidden="true" tabindex="-1"></a><span class="fu">## Subsetting by samples within specific metafields  </span></span>
<span id="cb39-143"><a href="#cb39-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-144"><a href="#cb39-144" aria-hidden="true" tabindex="-1"></a>If you want to get specific samples that match certain meta fields like specific collection_country or collection_date you can use ``</span>
<span id="cb39-145"><a href="#cb39-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-148"><a href="#cb39-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-149"><a href="#cb39-149" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta <span class="at">-h</span> </span>
<span id="cb39-150"><a href="#cb39-150" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-151"><a href="#cb39-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-152"><a href="#cb39-152" aria-hidden="true" tabindex="-1"></a>The python code for <span class="in">`extract_pmo_with_selected_meta`</span> script is below</span>
<span id="cb39-153"><a href="#cb39-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-156"><a href="#cb39-156" aria-hidden="true" tabindex="-1"></a><span class="in">```{python}</span></span>
<span id="cb39-157"><a href="#cb39-157" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb39-158"><a href="#cb39-158" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb39-159"><a href="#cb39-159" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb39-160"><a href="#cb39-160" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-line-numbers: true</span></span>
<span id="cb39-161"><a href="#cb39-161" aria-hidden="true" tabindex="-1"></a><span class="co">#| filename: pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_selected_meta.py</span></span>
<span id="cb39-162"><a href="#cb39-162" aria-hidden="true" tabindex="-1"></a><span class="co">#| file: ../pmotools-python/scripts/extractors_from_pmo/extract_pmo_with_selected_meta.py</span></span>
<span id="cb39-163"><a href="#cb39-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-164"><a href="#cb39-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-165"><a href="#cb39-165" aria-hidden="true" tabindex="-1"></a><span class="in">`pmotools-runner.py extract_pmo_with_selected_meta`</span> is written to allow the extraction on multiple intersecting meta field requirments that can be either supplied in a file or with delimited on the command line </span>
<span id="cb39-166"><a href="#cb39-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-167"><a href="#cb39-167" aria-hidden="true" tabindex="-1"></a>You may also want to know what current meta fields are present and how many samples in each. This can be done with <span class="co">[</span><span class="ot">`pmotools-runner.py list_specimen_meta_fields`</span><span class="co">](getting_basic_info_from_pmo.qmd#list_specimen_meta_fields)</span> and <span class="in">`pmotools-runner.py count_specimen_meta`</span></span>
<span id="cb39-168"><a href="#cb39-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-169"><a href="#cb39-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb39-170"><a href="#cb39-170" aria-hidden="true" tabindex="-1"></a><span class="in">cd example </span></span>
<span id="cb39-171"><a href="#cb39-171" aria-hidden="true" tabindex="-1"></a><span class="in">wget https://plasmogenepi.github.io/PMO_Docs/format/PathWeaverHeome1_PMO.json.gz</span></span>
<span id="cb39-172"><a href="#cb39-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-173"><a href="#cb39-173" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-174"><a href="#cb39-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-177"><a href="#cb39-177" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-178"><a href="#cb39-178" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-179"><a href="#cb39-179" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> list_specimen_meta_fields <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz</span>
<span id="cb39-180"><a href="#cb39-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-181"><a href="#cb39-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-182"><a href="#cb39-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-185"><a href="#cb39-185" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-186"><a href="#cb39-186" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-187"><a href="#cb39-187" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> count_specimen_meta <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz  <span class="at">--meta_fields</span> collection_country,collection_date <span class="kw">|</span> <span class="fu">head</span> <span class="at">-20</span></span>
<span id="cb39-188"><a href="#cb39-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-189"><a href="#cb39-189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-190"><a href="#cb39-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-191"><a href="#cb39-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-192"><a href="#cb39-192" aria-hidden="true" tabindex="-1"></a>Extracting on matching 1 meta field, below will extract just the samples that have collection_country=Bangladesh</span>
<span id="cb39-193"><a href="#cb39-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-194"><a href="#cb39-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval = F}</span></span>
<span id="cb39-195"><a href="#cb39-195" aria-hidden="true" tabindex="-1"></a><span class="in">cd example </span></span>
<span id="cb39-196"><a href="#cb39-196" aria-hidden="true" tabindex="-1"></a><span class="in">pmotools-runner.py extract_pmo_with_selected_meta  --file ../../format/PathWeaverHeome1_PMO.json.gz --metaFieldsValues "collection_country=Bangladesh" --output Bangladesh_moz2018_PMO.json.gz   --overwrite</span></span>
<span id="cb39-197"><a href="#cb39-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-198"><a href="#cb39-198" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-199"><a href="#cb39-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-200"><a href="#cb39-200" aria-hidden="true" tabindex="-1"></a>If you want to see how many samples were extracted can use <span class="in">`--verbose`</span></span>
<span id="cb39-201"><a href="#cb39-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-204"><a href="#cb39-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-205"><a href="#cb39-205" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-206"><a href="#cb39-206" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh"</span> <span class="at">--output</span> Bangladesh_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span></span>
<span id="cb39-207"><a href="#cb39-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-208"><a href="#cb39-208" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-209"><a href="#cb39-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-210"><a href="#cb39-210" aria-hidden="true" tabindex="-1"></a>Collecting more than 1 matching field separate by comma, for example to extract both Bangladesh,Benin</span>
<span id="cb39-211"><a href="#cb39-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-214"><a href="#cb39-214" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-215"><a href="#cb39-215" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-216"><a href="#cb39-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-217"><a href="#cb39-217" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh,Benin"</span> <span class="at">--output</span> Bangladesh_Benin_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span>
<span id="cb39-218"><a href="#cb39-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-219"><a href="#cb39-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-220"><a href="#cb39-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-221"><a href="#cb39-221" aria-hidden="true" tabindex="-1"></a>Can add more extraction criteria meta, for example to extract samples with collection_country of Bangladesh or Benin and with collection_date of 2016 </span>
<span id="cb39-222"><a href="#cb39-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-225"><a href="#cb39-225" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-226"><a href="#cb39-226" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-227"><a href="#cb39-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-228"><a href="#cb39-228" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh,Benin:collection_date=2016"</span> <span class="at">--output</span> Bangladesh_Benin_2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span>
<span id="cb39-229"><a href="#cb39-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-230"><a href="#cb39-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-231"><a href="#cb39-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-232"><a href="#cb39-232" aria-hidden="true" tabindex="-1"></a>To get more specific you can group meta field extraction criteria , for example if you wanted samples from Bangladesh from year 2015 but wanted Benin from year 2016 you can separate by a ; </span>
<span id="cb39-233"><a href="#cb39-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-236"><a href="#cb39-236" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-237"><a href="#cb39-237" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-238"><a href="#cb39-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-239"><a href="#cb39-239" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Bangladesh:collection_date=2015;collection_country=Benin:collection_date=2016"</span> <span class="at">--output</span> Bangladesh2015_Benin2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span>
<span id="cb39-240"><a href="#cb39-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-241"><a href="#cb39-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-242"><a href="#cb39-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-243"><a href="#cb39-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-244"><a href="#cb39-244" aria-hidden="true" tabindex="-1"></a>Rather than supplying with the command line a file can be created </span>
<span id="cb39-245"><a href="#cb39-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-248"><a href="#cb39-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-249"><a href="#cb39-249" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-250"><a href="#cb39-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-251"><a href="#cb39-251" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"group\tfield\tvalues"</span> <span class="op">&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb39-252"><a href="#cb39-252" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Bangladesh2015\tcollection_country\tBangladesh"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb39-253"><a href="#cb39-253" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Bangladesh2015\tcollection_date\t2015"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb39-254"><a href="#cb39-254" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Benin2016\tcollection_country\tBenin"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb39-255"><a href="#cb39-255" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"Benin2016\tcollection_date\t2016"</span> <span class="op">&gt;&gt;</span> Bangladesh2015_Benin2016_extractionCriteria.tsv </span>
<span id="cb39-256"><a href="#cb39-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-257"><a href="#cb39-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-258"><a href="#cb39-258" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/PathWeaverHeome1_PMO.json.gz <span class="at">--metaFieldsValues</span> Bangladesh2015_Benin2016_extractionCriteria.tsv <span class="at">--output</span> Bangladesh2015_Benin2016_moz2018_PMO.json.gz   <span class="at">--overwrite</span> <span class="at">--verbose</span> </span>
<span id="cb39-259"><a href="#cb39-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-260"><a href="#cb39-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-261"><a href="#cb39-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-262"><a href="#cb39-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-263"><a href="#cb39-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-266"><a href="#cb39-266" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-267"><a href="#cb39-267" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-268"><a href="#cb39-268" aria-hidden="true" tabindex="-1"></a><span class="ex">pmotools-runner.py</span> extract_pmo_with_selected_meta  <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--metaFieldsValues</span> <span class="st">"collection_country=Mozambique:geo_admin3=Inhassoro;collection_country=Mozambique:geo_admin3=Mandlakazi,Namaacha"</span> <span class="at">--output</span> Mozambique_moz2018_PMO.json.gz  <span class="at">--verbose</span>  <span class="at">--overwrite</span></span>
<span id="cb39-269"><a href="#cb39-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-270"><a href="#cb39-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-271"><a href="#cb39-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-272"><a href="#cb39-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-273"><a href="#cb39-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-274"><a href="#cb39-274" aria-hidden="true" tabindex="-1"></a><span class="fu">## Piping together extraction </span></span>
<span id="cb39-275"><a href="#cb39-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-276"><a href="#cb39-276" aria-hidden="true" tabindex="-1"></a>The extraction methods also allow for STDOUT and STDIN piping for example  </span>
<span id="cb39-279"><a href="#cb39-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-280"><a href="#cb39-280" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-281"><a href="#cb39-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-282"><a href="#cb39-282" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> t1_t20_t31_8025875029_8034209834_8034209115_moz2018_PMO.json.gz <span class="at">--overwrite</span></span>
<span id="cb39-283"><a href="#cb39-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-284"><a href="#cb39-284" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-285"><a href="#cb39-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-286"><a href="#cb39-286" aria-hidden="true" tabindex="-1"></a>Can also pipe into other <span class="in">`pmotools-runner.py`</span> functions like extracting allele tables </span>
<span id="cb39-287"><a href="#cb39-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-290"><a href="#cb39-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-291"><a href="#cb39-291" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-292"><a href="#cb39-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-293"><a href="#cb39-293" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_allele_table <span class="at">--file</span> STDIN <span class="at">--bioid</span> Mozambique2018-SeekDeep <span class="at">--output</span> alleles_data_t1_t20_t31_8025875029_8034209834_8034209115_moz2018_PMO.tsv.gz <span class="at">--overwrite</span></span>
<span id="cb39-294"><a href="#cb39-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-295"><a href="#cb39-295" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-296"><a href="#cb39-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-297"><a href="#cb39-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-298"><a href="#cb39-298" aria-hidden="true" tabindex="-1"></a>Can pipe final output to STDOUT as well for further processing </span>
<span id="cb39-299"><a href="#cb39-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-302"><a href="#cb39-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb39-303"><a href="#cb39-303" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> example </span>
<span id="cb39-304"><a href="#cb39-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-305"><a href="#cb39-305" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="at">-e</span> <span class="st">"8025875029\n8034209834\n8034209115"</span> <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_experiment_sample_ids <span class="at">--experiment_sample_ids</span> STDIN <span class="at">--file</span> ../../format/moz2018_PMO.json.gz <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_pmo_with_select_targets <span class="at">--file</span> STDIN <span class="at">--targets</span> t1,t20,t31  <span class="at">--output</span> STDOUT <span class="kw">|</span> <span class="ex">pmotools-runner.py</span> extract_allele_table <span class="at">--file</span> STDIN <span class="at">--bioid</span> Mozambique2018-SeekDeep <span class="at">--output</span> STDOUT  <span class="at">--specimen_info_meta_fields</span> specimen_id,collection_country</span>
<span id="cb39-306"><a href="#cb39-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-307"><a href="#cb39-307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb39-308"><a href="#cb39-308" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>