{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "553323d7-da1b-4308-b7b3-c5d35237ba4f",
   "metadata": {},
   "source": [
    "# Public Genomic Surveillance Data of *Plasmodium falciparum* from Five Countries\n",
    "\n",
    "This notebook demonstrates how publicly available genomic surveillance data from four countries (Eswatini, Namibia, South Africa, and Zambia) are reformatted into a Portable Microhaplotype Object (PMO) file for downstream analysis.  \n",
    "\n",
    "Data sources: Aranda-DÃ­az et al. (2025); Eloff et al. (2025); Nhlengethwa et al. (2025); Raman et al. (2025)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "569c31af-973b-4236-a383-cd0b2c143265",
   "metadata": {},
   "source": [
    "## Setup \n",
    "\n",
    "Here we import the necessary libraries, including those from the **pmotools** package that we will use to build the PMO."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "aad36565-2e3a-4be1-aeed-0959c86c7299",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a9321897-bcd0-46ea-b702-8ed0a99c740f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pmotools.pmo_builder.panel_information_to_pmo import panel_info_table_to_pmo, merge_panel_info_dicts\n",
    "from pmotools.pmo_builder.metatable_to_pmo import library_sample_info_table_to_pmo, specimen_info_table_to_pmo\n",
    "from pmotools.pmo_builder.mhap_table_to_pmo import mhap_table_to_pmo\n",
    "from pmotools.pmo_builder.merge_to_pmo import merge_to_pmo\n",
    "from pmotools.pmo_engine.pmo_writer import * \n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "445d376a-9295-44c0-8381-547b5db201b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pmotools.pmo_builder import panel_information_to_pmo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34db73de-bde0-4249-bfac-ef9ad620fee9",
   "metadata": {},
   "source": [
    "## Read in Data \n",
    "\n\n",
    "To build a PMO we need to assemble several pieces of information. This notebook walks through that process step-by-step, including:\n\n",
    "* Project Information\n",
    "* Specimen Information\n",
    "* Panel Information\n",
    "* Library Sample Information\n",
    "* Sequencing Information\n",
    "* Bioinformatics Information\n",
    "* Microhaplotype Information\n",
    "  \n",
    "We use pmotools to construct each section and then merge them at the end to create the final PMO.\n",
    "\n",
    "For any of the functions being used you can see more details using something like the example below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7f69a6d1-6f5e-4790-974d-a24bf01d17a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function merge_to_pmo in module pmotools.pmo_builder.merge_to_pmo:\n",
      "\n",
      "merge_to_pmo(specimen_info: list, library_sample_info: list, sequencing_info: list, panel_info: dict, mhap_info: dict, bioinfo_method_info: list, bioinfo_run_info: list, project_info: list, read_counts_by_stage_info: list | None = None)\n",
      "    Merge components into PMO, replacing names with indeces.\n",
      "    \n",
      "    :param specimen_info (list): a list of all the specimens within this project\n",
      "    :param library_sample_info (list) : a list of library samples within this project\n",
      "    :param sequencing_info (list) : a list of sequencing info for this project\n",
      "    :param panel_info (list) : a dictionary containing the panel and target information for this project\n",
      "    :param mhap_info (list) : a dictionary containing the microhaplotypes within this project, both detected and representative\n",
      "    :param bioinfo_method_info (list) : the bioinformatics pipeline/methods used to generated the amplicon analysis for this project\n",
      "    :param bioinfo_run_info (list) : the runtime info for the bioinformatics pipeline used to generated the amplicon analysis for this project\n",
      "    :param project_info (list) : the information about the projects stored in this PMO\n",
      "    :param read_counts_by_stage_info (Optional[list]) : the read counts by stage information for this project\n",
      "    \n",
      "    :return: a json formatted PMO string.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(merge_to_pmo)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d43b6489-0364-4c6a-a569-3915cac43a73",
   "metadata": {},
   "source": [
    "First we need to read in the data as it was stored for the project. In this case, there were three separate tables\n\n",
    "* [allele_data.txt.gz](allele_data.txt.gz) : this file contains the microhaplotype information\n",
    "* [specimen_info.tsv](specimen_info.tsv) : this file contains details of the speciment collected within the project\n",
    "* [library_sample_info.tsv](library_sample_info.tsv) : this file contains the information on the sequencing performed on the specimens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "88cd8c2c-5250-4876-9eb0-8ec2bb1b21f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "mhap_info_df = pd.read_csv('allele_data.txt.gz', sep='\\t')\n",
    "specimen_info_df = pd.read_csv('specimen_info.tsv', sep='\\t')\n",
    "library_sample_info = pd.read_csv('library_sample_info.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2413062-b8f2-4e24-9a03-770dbe6274d9",
   "metadata": {},
   "source": [
    "Below, we summarize the data by examining how many specimens were sequenced in replicate and therefore have multiple library samples associated with them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e44ebbf3-07eb-4486-a5d8-6a011b5b06bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of specimens: 2025\n",
      "Number of library samples: 2592\n",
      "\n",
      "Replicate summary:\n",
      "   n_replicates  count\n",
      "0             1   1503\n",
      "1             2    480\n",
      "2             3     39\n",
      "3             4      3\n",
      "Number of specimens sequenced in replicate: 522\n"
     ]
    }
   ],
   "source": [
    "print(\"Number of specimens:\",specimen_info_df['form participant_id'].nunique())\n",
    "print(\"Number of library samples:\",library_sample_info.SampleID.nunique())\n",
    "print(\"\\nReplicate summary:\")\n",
    "replicate_summary = library_sample_info.groupby('specimen_id').SampleID.size().value_counts().reset_index().rename(columns={'SampleID':'n_replicates'})\n",
    "print(replicate_summary)\n",
    "print(\"Number of specimens sequenced in replicate:\",replicate_summary[replicate_summary.n_replicates>1]['count'].sum())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f06c63c9-45b7-4808-8149-51abc0a810f8",
   "metadata": {},
   "source": [
    "## Project Information "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f080cbc-40cf-430a-9c81-cdceb2fcf540",
   "metadata": {},
   "source": [
    "In this section we enter information about the project. More information about the fields to include can be found [here](https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#projectinfo)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1a51e500-bbe3-4d78-bbd8-db741290110f",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_project_info = [{\n",
    "    \"project_name\":\"RegGenE8\",\n",
    "    \"project_description\":\"P. falciparum malaria surveillance in the Elimination 8.\",\n",
    "    \"project_collector_chief_scientist\":\"Jennifer Smith\",\n",
    "    \"project_type\":\"cross-sectional\"\n",
    "}]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94f6e549-0b11-4088-9976-f4e93f6bdeb9",
   "metadata": {},
   "source": [
    "## Specimen Information \n",
    "Now we transform the specimen table to comply with the PMO format. More information about the fields in this table can be found [here](https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#specimeninfo)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9cd150ec-5374-45d1-9807-d3d4d04ff084",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>form participant_id</th>\n",
       "      <th>facility_district</th>\n",
       "      <th>facility_province</th>\n",
       "      <th>facility_name</th>\n",
       "      <th>form date_diagnosis</th>\n",
       "      <th>host_taxon_id</th>\n",
       "      <th>specimen_taxon_id</th>\n",
       "      <th>project_name</th>\n",
       "      <th>country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>650004</td>\n",
       "      <td>Sithobela</td>\n",
       "      <td>Lubombo</td>\n",
       "      <td>SITHOBELA RURAL HEALTH CENTER</td>\n",
       "      <td>2023-03-30</td>\n",
       "      <td>9606</td>\n",
       "      <td>5833</td>\n",
       "      <td>RegGenE8</td>\n",
       "      <td>Eswatini</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>650005</td>\n",
       "      <td>Sithobela</td>\n",
       "      <td>Lubombo</td>\n",
       "      <td>SITHOBELA RURAL HEALTH CENTER</td>\n",
       "      <td>2023-03-22</td>\n",
       "      <td>9606</td>\n",
       "      <td>5833</td>\n",
       "      <td>RegGenE8</td>\n",
       "      <td>Eswatini</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>650007</td>\n",
       "      <td>Sithobela</td>\n",
       "      <td>Lubombo</td>\n",
       "      <td>SITHOBELA RURAL HEALTH CENTER</td>\n",
       "      <td>2023-03-30</td>\n",
       "      <td>9606</td>\n",
       "      <td>5833</td>\n",
       "      <td>RegGenE8</td>\n",
       "      <td>Eswatini</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>650008</td>\n",
       "      <td>Sithobela</td>\n",
       "      <td>Lubombo</td>\n",
       "      <td>SITHOBELA RURAL HEALTH CENTER</td>\n",
       "      <td>2023-04-03</td>\n",
       "      <td>9606</td>\n",
       "      <td>5833</td>\n",
       "      <td>RegGenE8</td>\n",
       "      <td>Eswatini</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>650010</td>\n",
       "      <td>Sithobela</td>\n",
       "      <td>Lubombo</td>\n",
       "      <td>SITHOBELA RURAL HEALTH CENTER</td>\n",
       "      <td>2023-04-04</td>\n",
       "      <td>9606</td>\n",
       "      <td>5833</td>\n",
       "      <td>RegGenE8</td>\n",
       "      <td>Eswatini</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  form participant_id facility_district facility_province  \\\n",
       "0              650004         Sithobela           Lubombo   \n",
       "1              650005         Sithobela           Lubombo   \n",
       "2              650007         Sithobela           Lubombo   \n",
       "3              650008         Sithobela           Lubombo   \n",
       "4              650010         Sithobela           Lubombo   \n",
       "\n",
       "                   facility_name form date_diagnosis  host_taxon_id  \\\n",
       "0  SITHOBELA RURAL HEALTH CENTER          2023-03-30           9606   \n",
       "1  SITHOBELA RURAL HEALTH CENTER          2023-03-22           9606   \n",
       "2  SITHOBELA RURAL HEALTH CENTER          2023-03-30           9606   \n",
       "3  SITHOBELA RURAL HEALTH CENTER          2023-04-03           9606   \n",
       "4  SITHOBELA RURAL HEALTH CENTER          2023-04-04           9606   \n",
       "\n",
       "   specimen_taxon_id project_name   country  \n",
       "0               5833     RegGenE8  Eswatini  \n",
       "1               5833     RegGenE8  Eswatini  \n",
       "2               5833     RegGenE8  Eswatini  \n",
       "3               5833     RegGenE8  Eswatini  \n",
       "4               5833     RegGenE8  Eswatini  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "specimen_info_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19120e3c-3446-41a7-a8f1-b7c3b60a561e",
   "metadata": {},
   "source": [
    "Because a specimen could contain multiple taxon IDs, the specimen_taxon_id is stored as a list inside of PMO. Therefore, we need to convert the values in this column to a list now. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "25842a76-c10c-4226-b0cb-24a059eec9ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "specimen_info_df.specimen_taxon_id=[[taxon_id]for taxon_id in (specimen_info_df.specimen_taxon_id)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c9d1211a-65aa-4288-bb8f-11d789cb1312",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_spec_info = specimen_info_table_to_pmo(\n",
    "                            specimen_info_df, \n",
    "                            specimen_name_col='form participant_id',\n",
    "                            specimen_taxon_id_col='specimen_taxon_id', \n",
    "                            host_taxon_id_col='host_taxon_id', \n",
    "                            collection_date_col='form date_diagnosis',\n",
    "                            collection_country_col='country',\n",
    "                            project_name_col='project_name',\n",
    "                            geo_admin1_col='facility_province',\n",
    "                            geo_admin2_col='facility_district',\n",
    "                            geo_admin3_col='facility_name'\n",
    "                           )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7a37e18-3ac6-47ba-b6f3-b86e21318c8e",
   "metadata": {},
   "source": [
    "## Panel Information\n",
    "The panel information describes which panel was used to sequence the samples. At the time this data was generated, only the drug resistance data was publicly available, so the dataset has been subset accordingly, and the panel information tables include only those targets. In this project, two different pool combinations from the [MAD4HatTeR panel](https://doi.org/10.1038/s41598-025-94716-5) were used. Because they all share the same reference genome, we first generate the genome information here:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0c2f3c12-b80a-4555-b775-fa6a5b2eee88",
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_info_dict = {\n",
    "        'name':'3D7',\n",
    "        'genome_version':'65',\n",
    "        'taxon_id':[5833],\n",
    "        'url':'https://plasmodb.org/a/service/raw-files/release-65/Pfalciparum3D7/fasta/data/PlasmoDB-65_Pfalciparum3D7_Genome.fasta'\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "68539e94-87f5-477a-98c6-547aaeaed3de",
   "metadata": {},
   "source": [
    "Now we generate the panel information using the `panel_info_table_to_pmo` for each of the panels\n\n*  [a52_pools.tsv](a52_pools.tsv)\n* [ab2_pools.tsv](ab2_pools.tsv) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "26a5bcdb-bd23-4c33-9e91-37ccfb62789a",
   "metadata": {},
   "outputs": [],
   "source": [
    "a52panel_info = pd.read_csv('a52_pools.tsv', sep='\\t')\n",
    "ab2panel_info = pd.read_csv('ab2_pools.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c2c3883e-bab5-4939-8054-4dc5a4cf626a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A52\n",
    "a52_panel_info = panel_info_table_to_pmo(\n",
    "    a52panel_info,\n",
    "    'A52',\n",
    "    genome_info_dict, \n",
    "    target_name_col='amplicon',\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "01bf0832-0a1e-45d4-917e-9fda32796754",
   "metadata": {},
   "outputs": [],
   "source": [
    "# AB2\n",
    "ab2_panel_info = panel_info_table_to_pmo(\n",
    "    ab2panel_info,\n",
    "    'AB2',\n",
    "    genome_info_dict, \n",
    "    target_name_col='amplicon',\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32204d14-5d49-4f24-a0c7-ce3019d5e7db",
   "metadata": {},
   "source": [
    "Now we have all of the panels generated seperately, we can merge them all together using the `merge_panel_info_dicts` function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "db2fc8dd-cb1a-458d-bbc4-6feede1e53f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "panels = [a52_panel_info, ab2_panel_info]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "915949c1-49fc-4fbd-9d33-35b8967ba3ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_panel_info = merge_panel_info_dicts(panels)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d21a263-bef0-4402-ba05-b5b971526ee9",
   "metadata": {},
   "source": [
    "## Library Sample Information \n",
    "Here we compile the information for all library samples from all sequencing or amplification runs of the specimens in this project. More information can be found [here](https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#librarysampleinfo)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8aebf32e-7934-4e18-95a8-8bfc0f33c781",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>SSPOOL</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>Pools</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MADH100_ES355_8071121168_A52_S365</td>\n",
       "      <td>SSPOOL19</td>\n",
       "      <td>650004</td>\n",
       "      <td>A52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MADH100_ES418_8071120168_A52_S369</td>\n",
       "      <td>SSPOOL19</td>\n",
       "      <td>650005</td>\n",
       "      <td>A52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MADH052_ES433_8071097924_A52_S69</td>\n",
       "      <td>SSPOOL13</td>\n",
       "      <td>650007</td>\n",
       "      <td>A52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MADH052_ES041_8071088821_A52_S18</td>\n",
       "      <td>SSPOOL13</td>\n",
       "      <td>650008</td>\n",
       "      <td>A52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MADH084_ES041_8071088821_A52_S290</td>\n",
       "      <td>SSPOOL16</td>\n",
       "      <td>650008</td>\n",
       "      <td>A52</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            SampleID    SSPOOL specimen_id Pools\n",
       "0  MADH100_ES355_8071121168_A52_S365  SSPOOL19      650004   A52\n",
       "1  MADH100_ES418_8071120168_A52_S369  SSPOOL19      650005   A52\n",
       "2   MADH052_ES433_8071097924_A52_S69  SSPOOL13      650007   A52\n",
       "3   MADH052_ES041_8071088821_A52_S18  SSPOOL13      650008   A52\n",
       "4  MADH084_ES041_8071088821_A52_S290  SSPOOL16      650008   A52"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "library_sample_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "12b7f25a-eb29-4e55-9ab6-7f5e7d2738db",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_library_sample = library_sample_info_table_to_pmo(\n",
    "    library_sample_info, \n",
    "    library_sample_name_col='SampleID', \n",
    "    sequencing_info_name_col='SSPOOL', \n",
    "    specimen_name_col='specimen_id', \n",
    "    panel_name_col='Pools',\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd8dd3e4-053f-4568-8248-be8d61b66eeb",
   "metadata": {},
   "source": [
    "## Sequencing Information"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e7d5902-9590-4ce8-90f5-3c384e4abb03",
   "metadata": {},
   "source": [
    "Now we put together the sequencing information for every sequencing run that was carried out on the library samples. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4774b226-c0ce-4fc2-8b1f-2f1f5fddd1b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "seq_platform='Illumina'\n",
    "seq_instrument_model = 'MiSeq'\n",
    "library_layout='150 paired-end reads'\n",
    "library_strategy='AMPLICON'\n",
    "library_source='GENOMIC'\n",
    "library_selection='PCR'\n",
    "seq_center='NICD'\n",
    "\n",
    "pmo_seq_info = []\n",
    "for seq_run in library_sample_info.SSPOOL.unique(): \n",
    "    seq_info_dict = {\n",
    "        'sequencing_info_name':(seq_run), \n",
    "        'seq_platform':seq_platform, \n",
    "        'seq_instrument_model':seq_instrument_model, \n",
    "        'library_layout':library_layout, \n",
    "        'library_strategy':library_strategy,\n",
    "        'library_source':library_source,\n",
    "        'library_selection':library_selection,\n",
    "        'seq_center':seq_center,\n",
    "    }\n",
    "    pmo_seq_info.append(seq_info_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f021a0ba-ec9f-4634-96f2-ff4ce28fa598",
   "metadata": {},
   "source": [
    "## Bioinformatics Method\n",
    "The [Mad4hatter pipeline](https://github.com/EPPIcenter/mad4hatter) was used to analyse all of this data. Below we link to this pipeline and provide some details about the programs used within the pipeline. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2c5b8003-0531-43ba-9aa3-ec20bdb0b227",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_bioinfo_method = [\n",
    "    {\n",
    "        \"methods\": [\n",
    "            {\"program\": \"DADA2\", \"program_version\": \"3.17\"},\n",
    "            {\"program\": \"cutadapt\", \"program_version\": \"4.4\"},\n",
    "        ],\n",
    "        \"pipeline\": {\n",
    "            \"program\": \"Mad4hatter\",\n",
    "            \"program_version\": \"v0.2.1\",\n",
    "            \"program_url\": \"https://github.com/EPPIcenter/mad4hatter\",\n",
    "        },\n",
    "    }\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2315fc3f-6528-440a-8437-5602f5d35c79",
   "metadata": {},
   "source": [
    "## Bioinformatics Runs\n",
    "\n",
    "Now we put together information on each of the bioinformatics runs that were used to generate the microhaplotypes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1ada65c9-3596-43f6-b075-17ce16004b7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_bioinfo_runs = []\n",
    "for run in library_sample_info.SSPOOL.unique(): \n",
    "    bioinfo_run = {\n",
    "        'bioinformatics_run_name':(run),\n",
    "        'bioinformatics_methods_id':0\n",
    "    }\n",
    "    pmo_bioinfo_runs.append(bioinfo_run)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1c9b656-797f-483e-8c0b-0f7a6394690f",
   "metadata": {},
   "source": [
    "## Microhaplotype Info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "587eaa25-7339-4b56-8768-2d0ac8e540d2",
   "metadata": {},
   "source": [
    "Finally, we put together the microhaplotype sections of the pmo."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8b751992-cc99-43fa-a997-c3f5988eaf3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Locus</th>\n",
       "      <th>ASV</th>\n",
       "      <th>Reads</th>\n",
       "      <th>Allele</th>\n",
       "      <th>PseudoCIGAR</th>\n",
       "      <th>SSPOOL</th>\n",
       "      <th>keep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MADH001_MPN17383_A52_S75_L001</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B</td>\n",
       "      <td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>\n",
       "      <td>12</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B.1</td>\n",
       "      <td>14+9N42+17N112+8N</td>\n",
       "      <td>SSPOOL1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MADH001_MPN17383_A52_S75_L001</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B</td>\n",
       "      <td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>\n",
       "      <td>4154</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B.2</td>\n",
       "      <td>14+9N42+17N88C112+8N</td>\n",
       "      <td>SSPOOL1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MADH001_MPN17383_A52_S75_L001</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B</td>\n",
       "      <td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>\n",
       "      <td>2056</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B.7</td>\n",
       "      <td>14+9N42+17N79A112+8N</td>\n",
       "      <td>SSPOOL1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MADH001_MPN17383_A52_S75_L001</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B</td>\n",
       "      <td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>\n",
       "      <td>3</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B.8</td>\n",
       "      <td>14+9N42+17N79A112+8N139D=TGATCCACTTTATGATAATAT...</td>\n",
       "      <td>SSPOOL1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MADH001_KZN6765_A52_S11_L001</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B</td>\n",
       "      <td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>\n",
       "      <td>593</td>\n",
       "      <td>Pf3D7_01_v3-194742-194973-1B.1</td>\n",
       "      <td>14+9N42+17N112+8N</td>\n",
       "      <td>SSPOOL1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        SampleID                         Locus  \\\n",
       "0  MADH001_MPN17383_A52_S75_L001  Pf3D7_01_v3-194742-194973-1B   \n",
       "1  MADH001_MPN17383_A52_S75_L001  Pf3D7_01_v3-194742-194973-1B   \n",
       "2  MADH001_MPN17383_A52_S75_L001  Pf3D7_01_v3-194742-194973-1B   \n",
       "3  MADH001_MPN17383_A52_S75_L001  Pf3D7_01_v3-194742-194973-1B   \n",
       "4   MADH001_KZN6765_A52_S11_L001  Pf3D7_01_v3-194742-194973-1B   \n",
       "\n",
       "                                                 ASV  Reads  \\\n",
       "0  TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...     12   \n",
       "1  TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...   4154   \n",
       "2  TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...   2056   \n",
       "3  TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...      3   \n",
       "4  TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...    593   \n",
       "\n",
       "                           Allele  \\\n",
       "0  Pf3D7_01_v3-194742-194973-1B.1   \n",
       "1  Pf3D7_01_v3-194742-194973-1B.2   \n",
       "2  Pf3D7_01_v3-194742-194973-1B.7   \n",
       "3  Pf3D7_01_v3-194742-194973-1B.8   \n",
       "4  Pf3D7_01_v3-194742-194973-1B.1   \n",
       "\n",
       "                                         PseudoCIGAR   SSPOOL  keep  \n",
       "0                                  14+9N42+17N112+8N  SSPOOL1  True  \n",
       "1                               14+9N42+17N88C112+8N  SSPOOL1  True  \n",
       "2                               14+9N42+17N79A112+8N  SSPOOL1  True  \n",
       "3  14+9N42+17N79A112+8N139D=TGATCCACTTTATGATAATAT...  SSPOOL1  True  \n",
       "4                                  14+9N42+17N112+8N  SSPOOL1  True  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mhap_info_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "87de8841-8105-40e2-8a7e-fede7083ad31",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmo_mhaps = mhap_table_to_pmo(\n",
    "                       microhaplotype_table=mhap_info_df, \n",
    "                       bioinformatics_run_name='SSPOOL',\n",
    "                       library_sample_name_col='SampleID',\n",
    "                       target_name_col='Locus',\n",
    "                       seq_col='ASV',\n",
    "                       reads_col='Reads')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7aefab41-1ac8-4731-b0c0-dd23efe03800",
   "metadata": {},
   "source": [
    "## Merge"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78367184-6846-4c84-890d-53e8e6fe442e",
   "metadata": {},
   "source": [
    "Now that we have all of the sections together, we can merge these together to form a complete PMO. If there are issues with the data this will be reported with descriptive errors. To further validate the final PMO you generate see the [validating pmos page](https://plasmogenepi.github.io/PMO_Docs/pmotools-python-usages/validating_pmos.html)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bacd7182-3837-4a5d-b738-26e40a1c5db3",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset1_pmo = merge_to_pmo(\n",
    "    pmo_spec_info,\n",
    "    pmo_library_sample,\n",
    "    pmo_seq_info,\n",
    "    pmo_panel_info,\n",
    "    pmo_mhaps,\n",
    "    pmo_bioinfo_method,\n",
    "    pmo_bioinfo_runs,\n",
    "    pmo_project_info\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "478c6f0c-eb8e-4d10-a06e-eeb95683a280",
   "metadata": {},
   "source": [
    "We can now write this to a file that can be shared. Notice the extension '.gz' added to the end of the filename, this will automatically compress the output file. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "335138c4-6e88-4b44-ac6d-75dc7d36e913",
   "metadata": {},
   "outputs": [],
   "source": [
    "pmowriter = PMOWriter()\n",
    "pmowriter.write_out_pmo(dataset1_pmo, \"dataset1_pmo.json.gz\", overwrite=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e66a3bfe-62b0-46f3-88b3-466684476be2",
   "metadata": {},
   "source": [
    "We can also validate our PMO, [dataset1_pmo.json.gz](dataset1_pmo.json.gz), and make sure everything is correct according to the schema. If there are any issues these will be output from the following command."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b9cf9ab2-2e46-4905-94a4-9be4516e1be0",
   "metadata": {},
   "outputs": [],
   "source": [
    "!pmotools-python validate_pmo --pmo dataset1_pmo.json.gz"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pmotools",
   "language": "python",
   "name": "pmotools"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
