<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Public Genomic Surveillance Data of Plasmodium falciparum from Five Countries – Portable Microhaplotype Object (PMO)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-49a824ff064150fa59632921ab93e189.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-6a2c4b793c0becaeb94cc5bf146c1484.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Portable Microhaplotype Object (PMO)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-format-info" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Format Info</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-format-info">    
        <li class="dropdown-header"><span class="menuTitle">Overview of Format</span></li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="../../format/FormatOverview.html">
 <span class="dropdown-text">PMO fields overview</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../PMO_Ecosystem/Pmo_in_the_ecosystem.html">
 <span class="dropdown-text">PMO within a Data Analysis Ecosystem</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">History of Format Development</span></li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="../../format/DevelopmentOfFormat.html">
 <span class="dropdown-text">History of how PMO Format was derived</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Overview of Format For Bioinformaticians</span></li>
        <li><hr class="dropdown-divider"></li>
        <li>
    <a class="dropdown-item" href="../../format/FormatExample.html">
 <span class="dropdown-text">PMO Examples</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../format/FormatOverviewAdvanced.html">
 <span class="dropdown-text">Format Overview For Developers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tools-installation" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Tools Installation</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tools-installation">    
        <li>
    <a class="dropdown-item" href="../../Tools_Installation/pmotools-python_installation.html">
 <span class="dropdown-text">pmotools-python installation</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-pmotools-python-usages" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">pmotools-python usages</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-pmotools-python-usages">    
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">Python interface</span></li>
        <li class="dropdown-header"><span class="menuTitle">Building a PMO</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages-notebooks/building_a_pmo/dataset1_pmo_generation.html">
 <span class="dropdown-text">Example of how to build a PMO</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Interacting with a PMO file</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages-notebooks/getting_basic_info_on_pmo_file.html">
 <span class="dropdown-text">Getting basic info out of a PMO</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Command line interface</span></li>
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-python</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/pmotools-python.html">
 <span class="dropdown-text">Command line interface to pmotools-python with pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Extracting out of PMO</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/extracting_allele_tables.html">
 <span class="dropdown-text">Extracting allele tables using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Subset PMO</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/subsetting_from_PMO.html">
 <span class="dropdown-text">Subsetting from a PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Getting sub info from PMO</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/getting_basic_info_from_pmo.html">
 <span class="dropdown-text">Getting basic info out of PMO using pmotools-python</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/extracting_panel_info_from_pmo.html">
 <span class="dropdown-text">Getting panel info out of PMO using pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Handling Multiple PMOs</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/handling_multiple_pmos.html">
 <span class="dropdown-text">Handling multiple PMOs pmotools-python</span></a>
  </li>  
        <li class="dropdown-header"><span class="menuTitle">Validating PMO files</span></li>
        <li>
    <a class="dropdown-item" href="../../pmotools-python-usages/validating_pmos.html">
 <span class="dropdown-text">Validating PMOs pmotools-python</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-resources" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Resources</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-resources">    
        <li>
    <a class="dropdown-item" href="../../References.html">
 <span class="dropdown-text">References</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li class="dropdown-header"><span class="menuTitle">Documentation</span></li>
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/PMO_Docs">
 <span class="dropdown-text">Documentation Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/PMO_Docs/issues">
 <span class="dropdown-text">Comment or Report an issue for Documentation</span></a>
  </li>  
        <li><hr class="dropdown-divider"></li>
        <li class="dropdown-header"><span class="menuTitle">pmotools-python</span></li>
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/pmotools-python/tree/develop">
 <span class="dropdown-text">pmotools-python Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/PlasmoGenEpi/pmotools-python/issues">
 <span class="dropdown-text">Comment or Report an issue for pmotools-python</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in Data</a></li>
  <li><a href="#project-information" id="toc-project-information" class="nav-link" data-scroll-target="#project-information">Project Information</a></li>
  <li><a href="#specimen-information" id="toc-specimen-information" class="nav-link" data-scroll-target="#specimen-information">Specimen Information</a></li>
  <li><a href="#panel-information" id="toc-panel-information" class="nav-link" data-scroll-target="#panel-information">Panel Information</a></li>
  <li><a href="#library-sample-information" id="toc-library-sample-information" class="nav-link" data-scroll-target="#library-sample-information">Library Sample Information</a></li>
  <li><a href="#sequencing-information" id="toc-sequencing-information" class="nav-link" data-scroll-target="#sequencing-information">Sequencing Information</a></li>
  <li><a href="#bioinformatics-method" id="toc-bioinformatics-method" class="nav-link" data-scroll-target="#bioinformatics-method">Bioinformatics Method</a></li>
  <li><a href="#bioinformatics-runs" id="toc-bioinformatics-runs" class="nav-link" data-scroll-target="#bioinformatics-runs">Bioinformatics Runs</a></li>
  <li><a href="#microhaplotype-info" id="toc-microhaplotype-info" class="nav-link" data-scroll-target="#microhaplotype-info">Microhaplotype Info</a></li>
  <li><a href="#merge" id="toc-merge" class="nav-link" data-scroll-target="#merge">Merge</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Public Genomic Surveillance Data of <em>Plasmodium falciparum</em> from Five Countries</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li></ul></div></div>
</div>



<div class="quarto-title-meta column-page-left">

    
  
    
  </div>
  


</header>


<p>This notebook demonstrates how publicly available genomic surveillance data from four countries (Eswatini, Namibia, South Africa, and Zambia) are reformatted into a Portable Microhaplotype Object (PMO) file for downstream analysis.</p>
<p>Data sources: Aranda-Díaz et al.&nbsp;(2025); Eloff et al.&nbsp;(2025); Nhlengethwa et al.&nbsp;(2025); Raman et al.&nbsp;(2025).</p>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<p>Here we import the necessary libraries, including those from the <strong>pmotools</strong> package that we will use to build the PMO.</p>
<div id="aad36565-2e3a-4be1-aeed-0959c86c7299" class="cell" data-execution_count="1">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="a9321897-bcd0-46ea-b702-8ed0a99c740f" class="cell" data-execution_count="2">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_builder.panel_information_to_pmo <span class="im">import</span> panel_info_table_to_pmo, merge_panel_info_dicts</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_builder.metatable_to_pmo <span class="im">import</span> library_sample_info_table_to_pmo, specimen_info_table_to_pmo</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_builder.mhap_table_to_pmo <span class="im">import</span> mhap_table_to_pmo</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_builder.merge_to_pmo <span class="im">import</span> merge_to_pmo</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_engine.pmo_writer <span class="im">import</span> <span class="op">*</span> </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="445d376a-9295-44c0-8381-547b5db201b2" class="cell" data-execution_count="3">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pmotools.pmo_builder <span class="im">import</span> panel_information_to_pmo</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="read-in-data" class="level2">
<h2 class="anchored" data-anchor-id="read-in-data">Read in Data</h2>
<p>To build a PMO we need to assemble several pieces of information. This notebook walks through that process step-by-step, including:</p>
<ul>
<li>Project Information</li>
<li>Specimen Information</li>
<li>Panel Information</li>
<li>Library Sample Information</li>
<li>Sequencing Information</li>
<li>Bioinformatics Information</li>
<li>Microhaplotype Information</li>
</ul>
<p>We use pmotools to construct each section and then merge them at the end to create the final PMO.</p>
<p>For any of the functions being used you can see more details using something like the example below:</p>
<div id="7f69a6d1-6f5e-4790-974d-a24bf01d17a0" class="cell" data-execution_count="4">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="bu">help</span>(merge_to_pmo)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Help on function merge_to_pmo in module pmotools.pmo_builder.merge_to_pmo:

merge_to_pmo(specimen_info: list, library_sample_info: list, sequencing_info: list, panel_info: dict, mhap_info: dict, bioinfo_method_info: list, bioinfo_run_info: list, project_info: list, read_counts_by_stage_info: list | None = None)
    Merge components into PMO, replacing names with indeces.
    
    :param specimen_info (list): a list of all the specimens within this project
    :param library_sample_info (list) : a list of library samples within this project
    :param sequencing_info (list) : a list of sequencing info for this project
    :param panel_info (list) : a dictionary containing the panel and target information for this project
    :param mhap_info (list) : a dictionary containing the microhaplotypes within this project, both detected and representative
    :param bioinfo_method_info (list) : the bioinformatics pipeline/methods used to generated the amplicon analysis for this project
    :param bioinfo_run_info (list) : the runtime info for the bioinformatics pipeline used to generated the amplicon analysis for this project
    :param project_info (list) : the information about the projects stored in this PMO
    :param read_counts_by_stage_info (Optional[list]) : the read counts by stage information for this project
    
    :return: a json formatted PMO string.
</code></pre>
</div>
</div>
<p>First we need to read in the data as it was stored for the project. In this case, there were three separate tables</p>
<ul>
<li><a href="allele_data.txt.gz">allele_data.txt.gz</a> : this file contains the microhaplotype information</li>
<li><a href="specimen_info.tsv">specimen_info.tsv</a> : this file contains details of the speciment collected within the project</li>
<li><a href="library_sample_info.tsv">library_sample_info.tsv</a> : this file contains the information on the sequencing performed on the specimens</li>
</ul>
<div id="88cd8c2c-5250-4876-9eb0-8ec2bb1b21f5" class="cell" data-execution_count="5">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>mhap_info_df <span class="op">=</span> pd.read_csv(<span class="st">'allele_data.txt.gz'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>specimen_info_df <span class="op">=</span> pd.read_csv(<span class="st">'specimen_info.tsv'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>library_sample_info <span class="op">=</span> pd.read_csv(<span class="st">'library_sample_info.tsv'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Below, we summarize the data by examining how many specimens were sequenced in replicate and therefore have multiple library samples associated with them.</p>
<div id="e44ebbf3-07eb-4486-a5d8-6a011b5b06bf" class="cell" data-execution_count="6">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Number of specimens:"</span>,specimen_info_df[<span class="st">'form participant_id'</span>].nunique())</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Number of library samples:"</span>,library_sample_info.SampleID.nunique())</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"</span><span class="ch">\n</span><span class="st">Replicate summary:"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>replicate_summary <span class="op">=</span> library_sample_info.groupby(<span class="st">'specimen_id'</span>).SampleID.size().value_counts().reset_index().rename(columns<span class="op">=</span>{<span class="st">'SampleID'</span>:<span class="st">'n_replicates'</span>})</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(replicate_summary)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Number of specimens sequenced in replicate:"</span>,replicate_summary[replicate_summary.n_replicates<span class="op">&gt;</span><span class="dv">1</span>][<span class="st">'count'</span>].<span class="bu">sum</span>())</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of specimens: 2025
Number of library samples: 2592

Replicate summary:
   n_replicates  count
0             1   1503
1             2    480
2             3     39
3             4      3
Number of specimens sequenced in replicate: 522</code></pre>
</div>
</div>
</section>
<section id="project-information" class="level2">
<h2 class="anchored" data-anchor-id="project-information">Project Information</h2>
<p>In this section we enter information about the project. More information about the fields to include can be found <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#projectinfo">here</a>.</p>
<div id="1a51e500-bbe3-4d78-bbd8-db741290110f" class="cell" data-execution_count="7">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pmo_project_info <span class="op">=</span> [{</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"project_name"</span>:<span class="st">"RegGenE8"</span>,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"project_description"</span>:<span class="st">"P. falciparum malaria surveillance in the Elimination 8."</span>,</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"project_collector_chief_scientist"</span>:<span class="st">"Jennifer Smith"</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"project_type"</span>:<span class="st">"cross-sectional"</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>}]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="specimen-information" class="level2">
<h2 class="anchored" data-anchor-id="specimen-information">Specimen Information</h2>
<p>Now we transform the specimen table to comply with the PMO format. More information about the fields in this table can be found <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#specimeninfo">here</a>.</p>
<div id="9cd150ec-5374-45d1-9807-d3d4d04ff084" class="cell" data-execution_count="8">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>specimen_info_df.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="8">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">form participant_id</th>
<th data-quarto-table-cell-role="th">facility_district</th>
<th data-quarto-table-cell-role="th">facility_province</th>
<th data-quarto-table-cell-role="th">facility_name</th>
<th data-quarto-table-cell-role="th">form date_diagnosis</th>
<th data-quarto-table-cell-role="th">host_taxon_id</th>
<th data-quarto-table-cell-role="th">specimen_taxon_id</th>
<th data-quarto-table-cell-role="th">project_name</th>
<th data-quarto-table-cell-role="th">country</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>650004</td>
<td>Sithobela</td>
<td>Lubombo</td>
<td>SITHOBELA RURAL HEALTH CENTER</td>
<td>2023-03-30</td>
<td>9606</td>
<td>5833</td>
<td>RegGenE8</td>
<td>Eswatini</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>650005</td>
<td>Sithobela</td>
<td>Lubombo</td>
<td>SITHOBELA RURAL HEALTH CENTER</td>
<td>2023-03-22</td>
<td>9606</td>
<td>5833</td>
<td>RegGenE8</td>
<td>Eswatini</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>650007</td>
<td>Sithobela</td>
<td>Lubombo</td>
<td>SITHOBELA RURAL HEALTH CENTER</td>
<td>2023-03-30</td>
<td>9606</td>
<td>5833</td>
<td>RegGenE8</td>
<td>Eswatini</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>650008</td>
<td>Sithobela</td>
<td>Lubombo</td>
<td>SITHOBELA RURAL HEALTH CENTER</td>
<td>2023-04-03</td>
<td>9606</td>
<td>5833</td>
<td>RegGenE8</td>
<td>Eswatini</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>650010</td>
<td>Sithobela</td>
<td>Lubombo</td>
<td>SITHOBELA RURAL HEALTH CENTER</td>
<td>2023-04-04</td>
<td>9606</td>
<td>5833</td>
<td>RegGenE8</td>
<td>Eswatini</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Because a specimen could contain multiple taxon IDs, the specimen_taxon_id is stored as a list inside of PMO. Therefore, we need to convert the values in this column to a list now.</p>
<div id="25842a76-c10c-4226-b0cb-24a059eec9ab" class="cell" data-execution_count="9">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>specimen_info_df.specimen_taxon_id<span class="op">=</span>[[taxon_id]<span class="cf">for</span> taxon_id <span class="kw">in</span> (specimen_info_df.specimen_taxon_id)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="c9d1211a-65aa-4288-bb8f-11d789cb1312" class="cell" data-execution_count="10">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>pmo_spec_info <span class="op">=</span> specimen_info_table_to_pmo(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                            specimen_info_df, </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                            specimen_name_col<span class="op">=</span><span class="st">'form participant_id'</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                            specimen_taxon_id_col<span class="op">=</span><span class="st">'specimen_taxon_id'</span>, </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                            host_taxon_id_col<span class="op">=</span><span class="st">'host_taxon_id'</span>, </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                            collection_date_col<span class="op">=</span><span class="st">'form date_diagnosis'</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                            collection_country_col<span class="op">=</span><span class="st">'country'</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                            project_name_col<span class="op">=</span><span class="st">'project_name'</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                            geo_admin1_col<span class="op">=</span><span class="st">'facility_province'</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                            geo_admin2_col<span class="op">=</span><span class="st">'facility_district'</span>,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                            geo_admin3_col<span class="op">=</span><span class="st">'facility_name'</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                           )</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="panel-information" class="level2">
<h2 class="anchored" data-anchor-id="panel-information">Panel Information</h2>
<p>The panel information describes which panel was used to sequence the samples. At the time this data was generated, only the drug resistance data was publicly available, so the dataset has been subset accordingly, and the panel information tables include only those targets. In this project, two different pool combinations from the <a href="https://doi.org/10.1038/s41598-025-94716-5">MAD4HatTeR panel</a> were used. Because they all share the same reference genome, we first generate the genome information here:</p>
<div id="0c2f3c12-b80a-4555-b775-fa6a5b2eee88" class="cell" data-execution_count="11">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>genome_info_dict <span class="op">=</span> {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">'name'</span>:<span class="st">'3D7'</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">'genome_version'</span>:<span class="st">'65'</span>,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">'taxon_id'</span>:[<span class="dv">5833</span>],</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">'url'</span>:<span class="st">'https://plasmodb.org/a/service/raw-files/release-65/Pfalciparum3D7/fasta/data/PlasmoDB-65_Pfalciparum3D7_Genome.fasta'</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Now we generate the panel information using the <code>panel_info_table_to_pmo</code> for each of the panels</p>
<ul>
<li><a href="a52_pools.tsv">a52_pools.tsv</a></li>
<li><a href="ab2_pools.tsv">ab2_pools.tsv</a></li>
</ul>
<div id="26a5bcdb-bd23-4c33-9e91-37ccfb62789a" class="cell" data-execution_count="12">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>a52panel_info <span class="op">=</span> pd.read_csv(<span class="st">'a52_pools.tsv'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>ab2panel_info <span class="op">=</span> pd.read_csv(<span class="st">'ab2_pools.tsv'</span>, sep<span class="op">=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="c2c3883e-bab5-4939-8054-4dc5a4cf626a" class="cell" data-execution_count="13">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># A52</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>a52_panel_info <span class="op">=</span> panel_info_table_to_pmo(</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    a52panel_info,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'A52'</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>    genome_info_dict, </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>    target_name_col<span class="op">=</span><span class="st">'amplicon'</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="01bf0832-0a1e-45d4-917e-9fda32796754" class="cell" data-execution_count="14">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AB2</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>ab2_panel_info <span class="op">=</span> panel_info_table_to_pmo(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    ab2panel_info,</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'AB2'</span>,</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    genome_info_dict, </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    target_name_col<span class="op">=</span><span class="st">'amplicon'</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Now we have all of the panels generated seperately, we can merge them all together using the <code>merge_panel_info_dicts</code> function.</p>
<div id="db2fc8dd-cb1a-458d-bbc4-6feede1e53f5" class="cell" data-execution_count="15">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>panels <span class="op">=</span> [a52_panel_info, ab2_panel_info]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div id="915949c1-49fc-4fbd-9d33-35b8967ba3ef" class="cell" data-execution_count="16">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>pmo_panel_info <span class="op">=</span> merge_panel_info_dicts(panels)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="library-sample-information" class="level2">
<h2 class="anchored" data-anchor-id="library-sample-information">Library Sample Information</h2>
<p>Here we compile the information for all library samples from all sequencing or amplification runs of the specimens in this project. More information can be found <a href="https://plasmogenepi.github.io/PMO_Docs/format/FormatOverviewAdvanced.html#librarysampleinfo">here</a>.</p>
<div id="8aebf32e-7934-4e18-95a8-8bfc0f33c781" class="cell" data-execution_count="17">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>library_sample_info.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="17">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">SampleID</th>
<th data-quarto-table-cell-role="th">SSPOOL</th>
<th data-quarto-table-cell-role="th">specimen_id</th>
<th data-quarto-table-cell-role="th">Pools</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>MADH100_ES355_8071121168_A52_S365</td>
<td>SSPOOL19</td>
<td>650004</td>
<td>A52</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>MADH100_ES418_8071120168_A52_S369</td>
<td>SSPOOL19</td>
<td>650005</td>
<td>A52</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>MADH052_ES433_8071097924_A52_S69</td>
<td>SSPOOL13</td>
<td>650007</td>
<td>A52</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>MADH052_ES041_8071088821_A52_S18</td>
<td>SSPOOL13</td>
<td>650008</td>
<td>A52</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>MADH084_ES041_8071088821_A52_S290</td>
<td>SSPOOL16</td>
<td>650008</td>
<td>A52</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="12b7f25a-eb29-4e55-9ab6-7f5e7d2738db" class="cell" data-execution_count="18">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>pmo_library_sample <span class="op">=</span> library_sample_info_table_to_pmo(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    library_sample_info, </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    library_sample_name_col<span class="op">=</span><span class="st">'SampleID'</span>, </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    sequencing_info_name_col<span class="op">=</span><span class="st">'SSPOOL'</span>, </span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    specimen_name_col<span class="op">=</span><span class="st">'specimen_id'</span>, </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    panel_name_col<span class="op">=</span><span class="st">'Pools'</span>,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="sequencing-information" class="level2">
<h2 class="anchored" data-anchor-id="sequencing-information">Sequencing Information</h2>
<p>Now we put together the sequencing information for every sequencing run that was carried out on the library samples.</p>
<div id="4774b226-c0ce-4fc2-8b1f-2f1f5fddd1b3" class="cell" data-execution_count="19">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>seq_platform<span class="op">=</span><span class="st">'Illumina'</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>seq_instrument_model <span class="op">=</span> <span class="st">'MiSeq'</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>library_layout<span class="op">=</span><span class="st">'150 paired-end reads'</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>library_strategy<span class="op">=</span><span class="st">'AMPLICON'</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>library_source<span class="op">=</span><span class="st">'GENOMIC'</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>library_selection<span class="op">=</span><span class="st">'PCR'</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>seq_center<span class="op">=</span><span class="st">'NICD'</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>pmo_seq_info <span class="op">=</span> []</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> seq_run <span class="kw">in</span> library_sample_info.SSPOOL.unique(): </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>    seq_info_dict <span class="op">=</span> {</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">'sequencing_info_name'</span>:(seq_run), </span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">'seq_platform'</span>:seq_platform, </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">'seq_instrument_model'</span>:seq_instrument_model, </span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">'library_layout'</span>:library_layout, </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">'library_strategy'</span>:library_strategy,</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">'library_source'</span>:library_source,</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">'library_selection'</span>:library_selection,</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">'seq_center'</span>:seq_center,</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    pmo_seq_info.append(seq_info_dict)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="bioinformatics-method" class="level2">
<h2 class="anchored" data-anchor-id="bioinformatics-method">Bioinformatics Method</h2>
<p>The <a href="https://github.com/EPPIcenter/mad4hatter">Mad4hatter pipeline</a> was used to analyse all of this data. Below we link to this pipeline and provide some details about the programs used within the pipeline.</p>
<div id="2c5b8003-0531-43ba-9aa3-ec20bdb0b227" class="cell" data-execution_count="20">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>pmo_bioinfo_method <span class="op">=</span> [</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"methods"</span>: [</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>            {<span class="st">"program"</span>: <span class="st">"DADA2"</span>, <span class="st">"program_version"</span>: <span class="st">"3.17"</span>},</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>            {<span class="st">"program"</span>: <span class="st">"cutadapt"</span>, <span class="st">"program_version"</span>: <span class="st">"4.4"</span>},</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>        ],</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pipeline"</span>: {</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">"program"</span>: <span class="st">"Mad4hatter"</span>,</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">"program_version"</span>: <span class="st">"v0.2.1"</span>,</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>            <span class="st">"program_url"</span>: <span class="st">"https://github.com/EPPIcenter/mad4hatter"</span>,</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="bioinformatics-runs" class="level2">
<h2 class="anchored" data-anchor-id="bioinformatics-runs">Bioinformatics Runs</h2>
<p>Now we put together information on each of the bioinformatics runs that were used to generate the microhaplotypes.</p>
<div id="1ada65c9-3596-43f6-b075-17ce16004b7d" class="cell" data-execution_count="21">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>pmo_bioinfo_runs <span class="op">=</span> []</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> run <span class="kw">in</span> library_sample_info.SSPOOL.unique(): </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    bioinfo_run <span class="op">=</span> {</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">'bioinformatics_run_name'</span>:(run),</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">'bioinformatics_methods_id'</span>:<span class="dv">0</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    pmo_bioinfo_runs.append(bioinfo_run)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="microhaplotype-info" class="level2">
<h2 class="anchored" data-anchor-id="microhaplotype-info">Microhaplotype Info</h2>
<p>Finally, we put together the microhaplotype sections of the pmo.</p>
<div id="8b751992-cc99-43fa-a997-c3f5988eaf3f" class="cell" data-execution_count="22">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>mhap_info_df.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
<div class="cell-output cell-output-display" data-execution_count="22">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">SampleID</th>
<th data-quarto-table-cell-role="th">Locus</th>
<th data-quarto-table-cell-role="th">ASV</th>
<th data-quarto-table-cell-role="th">Reads</th>
<th data-quarto-table-cell-role="th">Allele</th>
<th data-quarto-table-cell-role="th">PseudoCIGAR</th>
<th data-quarto-table-cell-role="th">SSPOOL</th>
<th data-quarto-table-cell-role="th">keep</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>MADH001_MPN17383_A52_S75_L001</td>
<td>Pf3D7_01_v3-194742-194973-1B</td>
<td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>
<td>12</td>
<td>Pf3D7_01_v3-194742-194973-1B.1</td>
<td>14+9N42+17N112+8N</td>
<td>SSPOOL1</td>
<td>True</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>MADH001_MPN17383_A52_S75_L001</td>
<td>Pf3D7_01_v3-194742-194973-1B</td>
<td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>
<td>4154</td>
<td>Pf3D7_01_v3-194742-194973-1B.2</td>
<td>14+9N42+17N88C112+8N</td>
<td>SSPOOL1</td>
<td>True</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>MADH001_MPN17383_A52_S75_L001</td>
<td>Pf3D7_01_v3-194742-194973-1B</td>
<td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>
<td>2056</td>
<td>Pf3D7_01_v3-194742-194973-1B.7</td>
<td>14+9N42+17N79A112+8N</td>
<td>SSPOOL1</td>
<td>True</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>MADH001_MPN17383_A52_S75_L001</td>
<td>Pf3D7_01_v3-194742-194973-1B</td>
<td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>
<td>3</td>
<td>Pf3D7_01_v3-194742-194973-1B.8</td>
<td>14+9N42+17N79A112+8N139D=TGATCCACTTTATGATAATAT...</td>
<td>SSPOOL1</td>
<td>True</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>MADH001_KZN6765_A52_S11_L001</td>
<td>Pf3D7_01_v3-194742-194973-1B</td>
<td>TACCTATAAAAATGAAAAAAATAAAGAAGAAGATAAATATGGAAAA...</td>
<td>593</td>
<td>Pf3D7_01_v3-194742-194973-1B.1</td>
<td>14+9N42+17N112+8N</td>
<td>SSPOOL1</td>
<td>True</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="87de8841-8105-40e2-8a7e-fede7083ad31" class="cell" data-execution_count="23">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>pmo_mhaps <span class="op">=</span> mhap_table_to_pmo(</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                       microhaplotype_table<span class="op">=</span>mhap_info_df, </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                       bioinformatics_run_name<span class="op">=</span><span class="st">'SSPOOL'</span>,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                       library_sample_name_col<span class="op">=</span><span class="st">'SampleID'</span>,</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                       target_name_col<span class="op">=</span><span class="st">'Locus'</span>,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>                       seq_col<span class="op">=</span><span class="st">'ASV'</span>,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                       reads_col<span class="op">=</span><span class="st">'Reads'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section>
<section id="merge" class="level2">
<h2 class="anchored" data-anchor-id="merge">Merge</h2>
<p>Now that we have all of the sections together, we can merge these together to form a complete PMO. If there are issues with the data this will be reported with descriptive errors. To further validate the final PMO you generate see the <a href="https://plasmogenepi.github.io/PMO_Docs/pmotools-python-usages/validating_pmos.html">validating pmos page</a>.</p>
<div id="bacd7182-3837-4a5d-b738-26e40a1c5db3" class="cell" data-execution_count="24">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>dataset1_pmo <span class="op">=</span> merge_to_pmo(</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>    pmo_spec_info,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    pmo_library_sample,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    pmo_seq_info,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    pmo_panel_info,</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    pmo_mhaps,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    pmo_bioinfo_method,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    pmo_bioinfo_runs,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>    pmo_project_info</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>We can now write this to a file that can be shared. Notice the extension ‘.gz’ added to the end of the filename, this will automatically compress the output file.</p>
<div id="335138c4-6e88-4b44-ac6d-75dc7d36e913" class="cell" data-execution_count="25">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>pmowriter <span class="op">=</span> PMOWriter()</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>pmowriter.write_out_pmo(dataset1_pmo, <span class="st">"dataset1_pmo.json.gz"</span>, overwrite<span class="op">=</span><span class="va">True</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>We can also validate our PMO, <a href="dataset1_pmo.json.gz">dataset1_pmo.json.gz</a>, and make sure everything is correct according to the schema. If there are any issues these will be output from the following command.</p>
<div id="b9cf9ab2-2e46-4905-94a4-9be4516e1be0" class="cell" data-execution_count="26">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>pmotools<span class="op">-</span>python validate_pmo <span class="op">--</span>pmo dataset1_pmo.json.gz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/PlasmoGenEpi\.github\.io\/PMO_Docs\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>